--- phonon-kde-4.3.4.orig/phonon/CMakeLists.txt
+++ phonon-kde-4.3.4/phonon/CMakeLists.txt
@@ -14,6 +14,16 @@ else(KDE4_DISABLE_MULTIMEDIA)
    macro_log_feature(_phonon_version_ok "Phonon" "Phonon library" "svn.kde.org/home/kde/trunk/kdesupport/phonon" TRUE "${PHONON_MIN_VERSION}" "STRONGLY RECOMMENDED: Required for playing audio and video throughout KDE")
 
    FIND_PACKAGE(Alsa)
+
+   option(WITH_ALSA "Use ALSA, if found" ON)
+   if(NOT WITH_ALSA)
+      message(STATUS "Disabling ALSA support, by request")
+      set(ALSA_FOUND)
+      set(ASOUND_LIBRARY)
+      set(HAVE_LIBASOUND2)
+      set(ALSA_INCLUDES)
+   endif(NOT WITH_ALSA)
+
    ALSA_CONFIGURE_FILE(${CMAKE_CURRENT_BINARY_DIR}/config-alsa.h)
 
    if(NOT ALSA_FOUND)
