diff -Naur a/services/fileindexer/indexer/ffmpegextractor.cpp b/services/fileindexer/indexer/ffmpegextractor.cpp
--- a/services/fileindexer/indexer/ffmpegextractor.cpp	2013-06-10 20:50:24.000000000 +0200
+++ b/services/fileindexer/indexer/ffmpegextractor.cpp	2013-06-23 20:36:27.336844353 +0200
@@ -158,12 +158,12 @@
                 subRes.addProperty( NFO::height(), codec->height );
 
                 int aspectRatio = codec->sample_aspect_ratio.num;
-                int frameRate = stream->r_frame_rate.num;
+                int frameRate = stream->avg_frame_rate.num;
 
                 if( codec->sample_aspect_ratio.den )
                     aspectRatio /= codec->sample_aspect_ratio.den;
-                if( stream->r_frame_rate.den )
-                    frameRate /= stream->r_frame_rate.den;
+                if( stream->avg_frame_rate.den )
+                    frameRate /= stream->avg_frame_rate.den;
 
                 subRes.addProperty( NFO::aspectRatio(), aspectRatio );
                 subRes.addProperty( NFO::frameRate(), frameRate );
