--- examples/Makefile.am.orig	2010-09-20 23:42:07.380545155 +0200
+++ examples/Makefile.am	2010-09-20 23:43:31.231899979 +0200
@@ -18,7 +18,7 @@
 	$(LIBINDICATE_CFLAGS)
 
 indicate_and_crash_LDADD = \
-	../libindicate/libindicate.la \
+	$(top_srcdir)/libindicate/libindicate.la \
 	$(LIBINDICATE_LIBS)
 
 indicate_alot_SOURCES = \
@@ -30,7 +30,7 @@
 	$(LIBINDICATE_CFLAGS)
 
 indicate_alot_LDADD = \
-	../libindicate/libindicate.la \
+	$(top_srcdir)/libindicate/libindicate.la \
 	$(LIBINDICATE_LIBS)
 
 listen_and_print_SOURCES = \
@@ -43,8 +43,8 @@
 	$(LIBINDICATEGTK_CFLAGS)
 
 listen_and_print_LDADD = \
-	../libindicate/libindicate.la \
-	../libindicate-gtk/libindicate-gtk.la \
+	$(top_srcdir)/libindicate/libindicate.la \
+	$(top_srcdir)/libindicate-gtk/libindicate-gtk.la \
 	$(LIBINDICATE_LIBS) \
 	$(LIBINDICATEGTK_LIBS)
 
@@ -58,8 +58,8 @@
 	$(LIBINDICATEGTK_CFLAGS)
 
 im_client_LDADD = \
-	../libindicate/libindicate.la \
-	../libindicate-gtk/libindicate-gtk.la \
+	$(top_srcdir)/libindicate/libindicate.la \
+	$(top_srcdir)/libindicate-gtk/libindicate-gtk.la \
 	$(LIBINDICATE_LIBS) \
 	$(LIBINDICATEGTK_LIBS)
 
@@ -72,7 +72,7 @@
 	$(LIBINDICATE_CFLAGS)
 
 show_hide_server_LDADD = \
-	../libindicate/libindicate.la \
+	$(top_srcdir)/libindicate/libindicate.la \
 	$(LIBINDICATE_LIBS)
 
 python_examples = \
