--- configure.ac.orig	2010-09-20 23:15:36.692289201 +0200
+++ configure.ac	2010-09-20 23:26:30.159014141 +0200
@@ -171,8 +171,18 @@
 # GTK Doc 
 ###########################
 
+AC_ARG_ENABLE([doc],
+	AC_HELP_STRING([--disable-doc], [Disable doc generation]),
+	[enable_doc=$enableval], [enable_doc=auto])
+AM_CONDITIONAL([WANT_DOC], [test "x$enable_doc" != "xno"])
+
 GTK_DOC_CHECK([1.4])
+AS_IF([test "x$enable_doc" != "xno"],[
 GNOME_DOC_INIT
+],[
+# The initialisation is still needed for certain conditionals to be defined, just make it not output any error
+GNOME_DOC_INIT(,,[:])
+])
 GNOME_COMPILE_WARNINGS
 
 ##############################
@@ -271,2 +281,7 @@
-	GObject Introspection:  $enable_introspection
-])
+	GObject Introspection:  $enable_introspection
+])
+
+AM_COND_IF([HAVE_GNOME_DOC_UTILS],
+	[AC_MSG_NOTICE([	Documentation:          yes])],
+	[AC_MSG_NOTICE([	Documentation:          no])]
+)
--- Makefile.am.orig	2010-09-20 23:12:10.442903362 +0200
+++ Makefile.am	2010-09-20 23:27:12.557785756 +0200
@@ -1,2 +1,6 @@
 
+if WANT_DOC
+DOC_SUBDIR = docs
+endif
+
 SUBDIRS = \
@@ -8,3 +12,3 @@
 	tests \
-	docs
+	$(DOC_SUBDIR)
 
