--- configure.ac.orig	2010-09-20 23:15:36.692289201 +0200
+++ configure.ac	2010-09-20 23:17:50.380918143 +0200
@@ -66,3 +66,9 @@
 
+AC_ARG_ENABLE([gtk],
+	AC_HELP_STRING([--disable-gtk], [Disable libindicate-gtk library]),
+	[enable_gtk=$enableval], [enable_gtk=auto])
+AM_CONDITIONAL([WANT_LIBINDICATEGTK], [test "x$enable_gtk" != "xno"])
+
+AS_IF([test "x$enable_gtk" != "xno"],[
 GDK_PIXBUF_REQUIRED_VERSION=2.12
 
@@ -74,4 +80,7 @@
-                                  dbus-glib-1 >= $DBUS_REQUIRED_VERSION)
+                                  dbus-glib-1 >= $DBUS_REQUIRED_VERSION,
+                                  [have_gtk=yes]
+)
 AC_SUBST(LIBINDICATEGTK_CFLAGS)
 AC_SUBST(LIBINDICATEGTK_LIBS)
+])
 
@@ -271,2 +280,7 @@
-	GObject Introspection:  $enable_introspection
-])
+	GObject Introspection:  $enable_introspection
+])
+
+AS_IF([test "x$have_gtk" = "xyes"],
+	[AC_MSG_NOTICE([	Gtk:                    yes])],
+	[AC_MSG_NOTICE([	Gtk:                    no])]
+)
--- Makefile.am.orig	2010-09-20 23:12:10.442903362 +0200
+++ Makefile.am	2010-09-20 23:17:35.141406450 +0200
@@ -1,5 +1,9 @@
 
+if WANT_LIBINDICATEGTK
+LIBINDICATEGTK_SUBDIR = libindicate-gtk
+endif
+
 SUBDIRS = \
 	libindicate \
-	libindicate-gtk \
+	$(LIBINDICATEGTK_SUBDIR) \
 	bindings \
