--- bindings/python/Makefile.am.orig	2010-12-01 11:02:01.089105608 +0100
+++ bindings/python/Makefile.am	2010-12-01 11:02:32.619126150 +0100
@@ -21,7 +21,7 @@
 indicatedir = $(pkgpyexecdir)/indicate
 indicate_LTLIBRARIES = _indicate.la
 
-_indicate_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_indicate -lpyglib-2.0-python2.6 -lpython2.6
+_indicate_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_indicate -lpyglib-2.0-python@PYTHON_VERSION@ -lpython@PYTHON_VERSION@
 _indicate_la_LIBADD = $(INDICATE_PYTHON_LIBS)  -L$(top_builddir)/libindicate/.libs/ -lindicate -L$(top_builddir)/libindicate-gtk/.libs -lindicate-gtk $(LIBINDICATEGTK_LIBS)
 _indicate_la_SOURCES = indicatemodule.c
 nodist__indicate_la_SOURCES = indicate.c
