--- configure.ac.orig	2011-01-29 21:38:44.667942630 +0100
+++ configure.ac	2011-01-29 21:42:03.892863493 +0100
@@ -153,15 +153,16 @@
-AM_PATH_PYTHON(2.3.5)
-AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(could not find Python headers)])
-
+PYTHON_REQUIRED=2.3.5
 PYGTK_REQUIRED=2.14.0
 PYGOBJECT_REQUIRED=0.22
 
-PKG_CHECK_MODULES(INDICATE_PYTHON, [pygtk-2.0 >= $PYGTK_REQUIRED
-                   pygobject-2.0 >= $PYGOBJECT_REQUIRED],
-                   [have_python=yes]
+AM_PATH_PYTHON([$PYTHON_REQUIRED])
+AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(could not find Python headers)])
+
+PKG_CHECK_MODULES([INDICATE_PYTHON], [python-$PYTHON_VERSION >= $PYTHON_REQUIRED
+                                      pygtk-2.0 >= $PYGTK_REQUIRED
+                                      pygobject-2.0 >= $PYGOBJECT_REQUIRED],
+                                     [have_python=yes]
 )
 
-INDICATE_PYTHON_LIBS+=$(python$PYTHON_VERSION-config --libs)
 # libpyglib-2.0-python* doesn't have a pkgconfig unfortunately
 INDICATE_PYTHON_LIBS+=" -lpyglib-2.0-python$PYTHON_VERSION"
 
