--- a/CMakeLists.txt	2015-01-21 01:15:05.438460864 +0100
+++ b/CMakeLists.txt	2015-01-21 01:17:08.896283297 +0100
@@ -2,10 +2,11 @@
 
 cmake_minimum_required(VERSION 2.8.12)
 
+set(QT_MIN_VERSION "5.2.0")
 find_package(ECM 0.0.9 REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR} ${CMAKE_SOURCE_DIR}/cmake)
 
-find_package(Qt5 5.2 REQUIRED COMPONENTS Quick Test)
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Quick)
 find_package(KF5 REQUIRED COMPONENTS I18n KIO Notifications ConfigWidgets DBusAddons KCMUtils IconThemes)
 find_package(Qca-qt5 2.1.0 REQUIRED)
 
@@ -32,6 +33,9 @@
 add_subdirectory(cli)
 add_subdirectory(fileitemactionplugin)
 
-add_subdirectory(tests)
+if(BUILD_TESTING)
+    find_package(Qt5Test ${QT_MIN_VERSION} REQUIRED)
+    add_subdirectory(tests)
+endif()
 
 install(PROGRAMS kdeconnect-non-plasma.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
