diff --git a/plugins/mousepad/CMakeLists.txt b/plugins/mousepad/CMakeLists.txt
index 7f14139e..a1daaa16 100644
--- a/plugins/mousepad/CMakeLists.txt
+++ b/plugins/mousepad/CMakeLists.txt
@@ -1,5 +1,5 @@
 if(UNIX)
-    find_package(KF5 ${KF5_MIN_VERSION} QUIET COMPONENTS Wayland)
+    find_package(KF5Wayland ${KF5_MIN_VERSION} QUIET)
 
     find_package(LibFakeKey QUIET)
     set_package_properties(LibFakeKey PROPERTIES DESCRIPTION "fake key events"
