diff --git a/kcms/keyboard/xkb_rules.cpp b/kcms/keyboard/xkb_rules.cpp
index 9fd0324..094f94f 100644
--- a/kcms/keyboard/xkb_rules.cpp
+++ b/kcms/keyboard/xkb_rules.cpp
@@ -160,29 +160,7 @@ QString Rules::getRulesName()
 
 QString Rules::findXkbDir()
 {
-	QString xkbParentDir;
-
-	QString base(XLIBDIR);
-	if( base.count('/') >= 3 ) {
-		// .../usr/lib/X11 -> /usr/share/X11/xkb vs .../usr/X11/lib -> /usr/X11/share/X11/xkb
-		QString delta = base.endsWith("X11") ? "/../../share/X11" : "/../share/X11";
-		QDir baseDir(base + delta);
-		if( baseDir.exists() ) {
-			xkbParentDir = baseDir.absolutePath();
-		}
-		else {
-			QDir baseDir(base + "/X11");	// .../usr/X11/lib/X11/xkb (old XFree)
-			if( baseDir.exists() ) {
-				xkbParentDir = baseDir.absolutePath();
-			}
-		}
-	}
-
-	if( xkbParentDir.isEmpty() ) {
-		xkbParentDir = "/usr/share/X11";
-	}
-
-	return xkbParentDir + "/xkb";
+	return QStringLiteral(XKBDIR);
 }
 
 static QString findXkbRulesFile()
diff --git a/ConfigureChecks.cmake b/ConfigureChecks.cmake
index 4fc4bce..e35ae11 100644
--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -1,16 +1,9 @@
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${CMAKE_MODULE_PATH} )
 
-find_program(some_x_program NAMES iceauth xrdb xterm)
-if (NOT some_x_program)
-    set(some_x_program /usr/bin/xrdb)
-    message(WARNING "Warning: Could not determine X binary directory. Assuming /usr/bin.")
-endif ()
-get_filename_component(proto_xbindir "${some_x_program}" PATH)
-get_filename_component(XBINDIR "${proto_xbindir}" ABSOLUTE)
-get_filename_component(xrootdir "${XBINDIR}" PATH)
-set(XLIBDIR "${xrootdir}/lib/X11")
-if (NOT EXISTS "${XLIBDIR}")
-    message(FATAL_ERROR "Couldn't find X11 prefix XLIBDIR: ${XLIBDIR}")
+find_file(XKBDIR "share/X11/xkb/")
+set(XLIBDIR "${xrootdir}/share/X11")
+if (NOT EXISTS "${XKBDIR}")
+    message(FATAL_ERROR "Couldn't find XKB location: ${XKBDIR}")
 endif()
 set(KWIN_BIN "kwin_x11" CACHE STRING "Name of the KWin binary")
 
diff --git a/config-workspace.h.cmake b/config-workspace.h.cmake
index eda8996..d0b48b9 100644
--- a/config-workspace.h.cmake
+++ b/config-workspace.h.cmake
@@ -140,11 +140,8 @@
 /* Define to 1 if you can safely include both <sys/time.h> and <time.h>. */
 #cmakedefine TIME_WITH_SYS_TIME 1
 
-/* X binaries directory */
-#cmakedefine XBINDIR "${XBINDIR}"
-
-/* X libraries directory */
-#cmakedefine XLIBDIR "${XLIBDIR}"
+/* xkb resources directory */
+#cmakedefine XKBDIR "${XKBDIR}"
 
 /* PackageKit-Qt has been found */
 #cmakedefine PackageKitQt5_FOUND 1
