--- a/CMakeLists.txt	2015-01-21 01:59:46.589187275 +0100
+++ b/CMakeLists.txt	2015-01-21 02:00:47.260569833 +0100
@@ -25,7 +25,6 @@
     Widgets
     X11Extras
     Svg
-    Test
     Concurrent
 )
 
--- a/kcms/lookandfeel/CMakeLists.txt	2015-01-21 01:59:46.643187615 +0100
+++ b/kcms/lookandfeel/CMakeLists.txt	2015-01-21 02:05:09.909255405 +0100
@@ -44,6 +44,9 @@
 install(FILES kcm_lookandfeel.desktop DESTINATION ${SERVICES_INSTALL_DIR})
 install(TARGETS kcm_lookandfeel DESTINATION ${PLUGIN_INSTALL_DIR})
 
-add_subdirectory(autotests)
+if(BUILD_TESTING)
+   find_package(Qt5Test ${QT_MIN_VERSION} CONFIG REQUIRED)
+   add_subdirectory(autotests)
+endif()
 
 plasma_install_package(package kcm_lookandfeel kcms kcm_lookandfeel)
--- a/kcms/keyboard/CMakeLists.txt	2015-01-21 01:59:46.629187527 +0100
+++ b/kcms/keyboard/CMakeLists.txt	2015-01-21 02:08:24.103523804 +0100
@@ -183,5 +183,7 @@
 
 
 # Unit tests
-
-add_subdirectory( tests )
+if(BUILD_TESTING)
+    find_package(Qt5Test ${QT_MIN_VERSION} CONFIG REQUIRED)
+    add_subdirectory( tests )
+endif()
