--- idjc-0.8.7.orig/Makefile.am	2011-09-10 13:21:48.000000000 +0000
+++ idjc-0.8.7/Makefile.am	2012-09-20 19:49:17.990368400 +0000
@@ -4,7 +4,6 @@
 EXTRA_SCRIPTS=
 nodist_bin_SCRIPTS=${PACKAGE_NAME} idjcctrl
 dist_noinst_SCRIPTS=idjc.in.in idjcctrl_.py env-up ffmpeg_avcodec.c
-nodist_doc_DATA=AUTHORS.gz NEWS.gz README.gz ChangeLog.gz
 docdir=${prefix}/share/doc/${PACKAGE_NAME}-${PACKAGE_VERSION}
 dist_noinst_DATA=idjc.desktop.in.in
 nodist_appmenu_DATA=${PACKAGE_NAME}.desktop
@@ -21,20 +20,8 @@
 ${PACKAGE_NAME}.desktop: idjc.desktop.in
 	@sed -e 's|$${prefix}|${prefix}|' -e 's|$${exec_prefix}|${exec_prefix}|' $? >$@
 
-AUTHORS.gz: AUTHORS
-	@ gzip -c $? >$@
-	 
-NEWS.gz: NEWS
-	@ gzip -c $? >$@
-	
-README.gz: README
-	@ gzip -c $? >$@
-	
-ChangeLog.gz: ChangeLog
-	@ gzip -c $? >$@
-
 clean-local:
-	rm -f ChangeLog.gz README.gz NEWS.gz AUTHORS.gz ${PACKAGE_NAME}.desktop idjcctrl ${PACKAGE_NAME}
+	rm -f ${PACKAGE_NAME}.desktop idjcctrl ${PACKAGE_NAME}
 
 # Too lazy to manually claw this out from the source tree.
 dtdist:
