--- idjc-0.8.7.orig/idjc.desktop.in.in	2011-12-18 18:43:36.000000000 +0000
+++ idjc-0.8.7/idjc.desktop.in.in	2012-09-20 19:57:17.840381251 +0000
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Categories=GNOME;Network;
+Categories=GTK;GNOME;Network;
 Comment=Create your live radio show or podcast
 Comment[en_GB]=Create your live radio show or podcast
 Comment[it]=Crea la tua live radio o podcast
@@ -12,15 +12,10 @@
 GenericName[de]=Grafischer Shoutcast/Icecast Client
 GenericName[fr]=Client graphique shoutcast/icecast
 Icon=${prefix}/share/pixmaps/@PACKAGE_NAME@.png
-MimeType=
 Name=Internet DJ Console
-Path=
 StartupNotify=true
 Terminal=false
-TerminalOptions=
 Type=Application
 Version=1.0
-X-DCOP-ServiceType=
 X-KDE-SubstituteUID=false
-X-KDE-Username=
 X-Ubuntu-Gettext-Domain=idjc
