--- nsplugins/viewer/scripting.h.orig	2010-12-26 01:50:16.704000278 +0100
+++ nsplugins/viewer/scripting.h	2010-12-26 01:57:20.581000171 +0100
@@ -20,6 +20,9 @@
 
 */
 
+#ifndef VIEWER_SCRIPTING_H
+#define VIEWER_SCRIPTING_H
+
 #include "nsplugin.h"
 #include "sdk/npruntime.h"
 #include <kparts/browserextension.h>
@@ -93,5 +96,6 @@
 
 } // namespace kdeNsPluginViewer
 
-// kate: indent-width 4; replace-tabs on; tab-width 4; space-indent on;
+#endif
 
+// kate: indent-width 4; replace-tabs on; tab-width 4; space-indent on;
