--- kdeui/util/fixx11h.h	2009/03/04 21:28:02	935354
+++ kdeui/util/fixx11h.h	2009/03/27 14:43:00	945494
@@ -89,11 +89,16 @@
 #ifdef Bool
 #ifndef FIXX11H_Bool
 #define FIXX11H_Bool
-typedef Bool XBool;
-#undef Bool
-typedef XBool Bool;
+#ifdef _XTYPEDEF_BOOL /* Xdefs.h has typedef'ed Bool already */
+ #undef Bool
+#else
+ typedef Bool XBool;
+ #undef Bool
+ typedef XBool Bool;
+#endif
 #endif
 #undef Bool
+#define _XTYPEDEF_BOOL
 #endif
 // template <---

