--- kdeui/util/kkeyserver_x11.cpp	2012-02-25 18:53:49.480771132 -0200
+++ kdeui/util/kkeyserver_x11.cpp	2012-02-25 18:20:38.912403430 -0200
@@ -355,7 +355,7 @@
     { Qt::Key_MediaRecord, XF86XK_AudioRecord },
     { Qt::Key_LaunchMail, XF86XK_Mail },
     { Qt::Key_Launch0,    XF86XK_MyComputer },
-    { Qt::Key_Calculator,    XF86XK_Calculator },
+    { Qt::Key_Launch1,    XF86XK_Calculator },
     { Qt::Key_Memo,    XF86XK_Memo },
     { Qt::Key_ToDoList,    XF86XK_ToDoList },
     { Qt::Key_Calendar,    XF86XK_Calendar },
