diff -up kdelibs-4.4.80/kdeui/notifications/kstatusnotifieritem.cpp.icon-name-qt47 kdelibs-4.4.80/kdeui/notifications/kstatusnotifieritem.cpp
--- kdelibs-4.4.80/kdeui/notifications/kstatusnotifieritem.cpp.icon-name-qt47	2010-05-16 22:04:05.000000000 +0200
+++ kdelibs-4.4.80/kdeui/notifications/kstatusnotifieritem.cpp	2010-05-21 22:50:44.819211510 +0200
@@ -65,14 +65,7 @@ protected:
     virtual QString iconNameForAction(QAction *action)
     {
         KIcon icon(action->icon());
-#if QT_VERSION >= 0x040700
-        // QIcon::name() is in the 4.7 git branch, but it is not in 4.7 TP.
-        // If you get a build error here, you need to update your pre-release
-        // of Qt 4.7.
-        return icon.isNull() ? QString() : icon.name();
-#else
         return QString();
-#endif
     }
 };
 
