Desc:	Add needed include for mmap() on Solaris.
Author:	Daniel Vergien

--- kdelibs-4.3.2/kdecore/sycoca/ksycoca.cpp.orig	2009-10-14 12:12:45.697358506 +0200
+++ kdelibs-4.3.2/kdecore/sycoca/ksycoca.cpp	2009-10-14 12:13:16.082152232 +0200
@@ -46,6 +46,10 @@
 
 #include "ksycocadevices_p.h"
 
+/* needed on solaris 10 for mmap */
+
+#include <sys/mman.h>
+
 // TODO: remove mmap() from kdewin32 and use QFile::mmap() when needed
 #ifdef Q_WS_WIN
 #undef HAVE_MMAP
