--- kdelibs-4.7.1/khtml/imload/decoders/pngloader.cpp
+++ kdelibs-4.7.1/khtml/imload/decoders/pngloader.cpp
@@ -198,7 +198,7 @@
     {
         if (interlaced)
         {
-            Q_ASSERT(pngReadStruct->row_info.pixel_depth <= depth * 8);
+            //Q_ASSERT(pngReadStruct->row_info.pixel_depth <= depth * 8);
             requestScanline(rowNum, scanlineBuf);
             png_progressive_combine_row(pngReadStruct, scanlineBuf, data);
             notifyScanline(pass + 1, scanlineBuf);
