--- plasma/popupapplet.cpp	2009-04-09 15:15:12.000000000 +0200
+++ plasma/popupapplet.cpp	2009-04-09 16:59:39.000000000 +0200
@@ -275,11 +275,11 @@
                 //stuff out of your Dialog (extenders). Monitor WindowDeactivate events so we can
                 //emulate the same kind of behavior as Qt::Popup (close when you click somewhere
                 //else.
-                //Use Qt::Tool otherwise dialogs get shown over screensaver (bug #179924).
-                Qt::WindowFlags wflags = Qt::Tool | Qt::FramelessWindowHint | Qt::WindowStaysOnTopHint;
+                Qt::WindowFlags wflags = Qt::FramelessWindowHint | Qt::WindowStaysOnTopHint;
 
                 if (passive) {
-                    wflags |= Qt::X11BypassWindowManagerHint;
+                    //Use Qt::Tool otherwise dialogs get shown over screensaver (bug #179924).
+                    wflags |= Qt::Tool | Qt::X11BypassWindowManagerHint;
                 }
 
                 dialog->setWindowFlags(wflags);
@@ -413,10 +413,11 @@
     if (d->dialog) {
         Qt::WindowFlags wflags = d->dialog->windowFlags();
 
+        Qt::WindowFlags passiveFlags = Qt::Tool | Qt::X11BypassWindowManagerHint;
         if (d->passive) {
-            wflags |= Qt::X11BypassWindowManagerHint;
+            wflags |= passiveFlags;
         } else {
-            wflags &= ~Qt::X11BypassWindowManagerHint;
+            wflags &= ~passiveFlags;
         }
 
         d->dialog->setWindowFlags(wflags);
