From 5926264d13900efec70aaac434b7b67f9d8e6837 Mon Sep 17 00:00:00 2001
From: Lukas Tinkl <lukas@kde.org>
Date: Tue, 19 Jun 2012 18:27:04 +0200
Subject: [PATCH 5/5] correctly report power saving status

---
 solid/solid/powermanagement.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/solid/solid/powermanagement.cpp b/solid/solid/powermanagement.cpp
index 8dc8535..faadea4 100644
--- a/solid/solid/powermanagement.cpp
+++ b/solid/solid/powermanagement.cpp
@@ -73,7 +73,7 @@ Solid::PowerManagement::Notifier::Notifier()
 
 bool Solid::PowerManagement::appShouldConserveResources()
 {
-    return globalPowerManager->powerSaveStatus;
+    return globalPowerManager->managerIface.GetPowerSaveStatus();
 }
 
 QSet<Solid::PowerManagement::SleepState> Solid::PowerManagement::supportedSleepStates()
-- 
1.7.9.2

