https://bugs.kde.org/show_bug.cgi?id=276151

--- cmake/modules/PythonMacros.cmake
+++ cmake/modules/PythonMacros.cmake
@@ -26,54 +26,56 @@
   install(FILES ${SOURCE_FILE} DESTINATION ${DESTINATION_DIR})
 
   # Byte compile and install the .pyc file.
-  get_filename_component(_absfilename ${SOURCE_FILE} ABSOLUTE)
-  get_filename_component(_filename ${SOURCE_FILE} NAME)
-  get_filename_component(_filenamebase ${SOURCE_FILE} NAME_WE)
-  get_filename_component(_basepath ${SOURCE_FILE} PATH)
-
-  if(WIN32)
-    # remove drive letter
-    string(REGEX REPLACE "^[a-zA-Z]:/" "/" _basepath "${_basepath}")
-  endif(WIN32)
-
-  set(_bin_py ${CMAKE_CURRENT_BINARY_DIR}/${_basepath}/${_filename})
-
-  # Python 3.2 changed the pyc file location
-  if(PYTHON_VERSION_STRING VERSION_GREATER 3.1)
-    # To get the right version for suffix
-    set(_bin_pyc "${CMAKE_CURRENT_BINARY_DIR}/${_basepath}/__pycache__/${_filenamebase}.cpython-${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}.pyc")
-    set(_py_install_dir "${DESTINATION_DIR}/__pycache__/")
-  else()
-    set(_bin_pyc "${CMAKE_CURRENT_BINARY_DIR}/${_basepath}/${_filenamebase}.pyc")
-    set(_py_install_dir "${DESTINATION_DIR}")
-  endif()
-
-  FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${_basepath})
-
-  # Setting because it will be displayed later, in compile_python_files
-  set(_message "Byte-compiling ${_bin_py} to ${_bin_pyc}")
-
-  get_filename_component(_abs_bin_py ${_bin_py} ABSOLUTE)
-  if(_abs_bin_py STREQUAL _absfilename)    # Don't copy the file onto itself.
-    add_custom_command(
-      TARGET compile_python_files
-      COMMAND "${CMAKE_COMMAND}" -E echo "${_message}"
-      COMMAND "${PYTHON_EXECUTABLE}" "${_python_compile_py}" "${_bin_py}"
-      DEPENDS "${_absfilename}"
-    )
-  else()
-    add_custom_command(
-      TARGET compile_python_files
-      COMMAND "${CMAKE_COMMAND}" -E echo "${_message}"
-      COMMAND "${CMAKE_COMMAND}" -E copy "${_absfilename}" "${_bin_py}"
-      COMMAND "${PYTHON_EXECUTABLE}" "${_python_compile_py}" "${_bin_py}"
-      DEPENDS "${_absfilename}"
-    )
-  endif()
-
-  install(FILES ${_bin_pyc} DESTINATION "${_py_install_dir}")
-  unset(_py_install_dir)
-  unset(_message)
+  if("$ENV{PYTHONDONTWRITEBYTECODE}" STREQUAL "")
+    get_filename_component(_absfilename ${SOURCE_FILE} ABSOLUTE)
+    get_filename_component(_filename ${SOURCE_FILE} NAME)
+    get_filename_component(_filenamebase ${SOURCE_FILE} NAME_WE)
+    get_filename_component(_basepath ${SOURCE_FILE} PATH)
+
+    if(WIN32)
+      # remove drive letter
+      string(REGEX REPLACE "^[a-zA-Z]:/" "/" _basepath "${_basepath}")
+    endif(WIN32)
+
+    set(_bin_py ${CMAKE_CURRENT_BINARY_DIR}/${_basepath}/${_filename})
+
+    # Python 3.2 changed the pyc file location
+    if(PYTHON_VERSION_STRING VERSION_GREATER 3.1)
+      # To get the right version for suffix
+      set(_bin_pyc "${CMAKE_CURRENT_BINARY_DIR}/${_basepath}/__pycache__/${_filenamebase}.cpython-${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}.pyc")
+      set(_py_install_dir "${DESTINATION_DIR}/__pycache__/")
+    else()
+      set(_bin_pyc "${CMAKE_CURRENT_BINARY_DIR}/${_basepath}/${_filenamebase}.pyc")
+      set(_py_install_dir "${DESTINATION_DIR}")
+    endif()
+
+    FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${_basepath})
+
+    # Setting because it will be displayed later, in compile_python_files
+    set(_message "Byte-compiling ${_bin_py} to ${_bin_pyc}")
+
+    get_filename_component(_abs_bin_py ${_bin_py} ABSOLUTE)
+    if(_abs_bin_py STREQUAL _absfilename)    # Don't copy the file onto itself.
+      add_custom_command(
+        TARGET compile_python_files
+        COMMAND "${CMAKE_COMMAND}" -E echo "${_message}"
+        COMMAND "${PYTHON_EXECUTABLE}" "${_python_compile_py}" "${_bin_py}"
+        DEPENDS "${_absfilename}"
+      )
+    else()
+      add_custom_command(
+        TARGET compile_python_files
+        COMMAND "${CMAKE_COMMAND}" -E echo "${_message}"
+        COMMAND "${CMAKE_COMMAND}" -E copy "${_absfilename}" "${_bin_py}"
+        COMMAND "${PYTHON_EXECUTABLE}" "${_python_compile_py}" "${_bin_py}"
+        DEPENDS "${_absfilename}"
+      )
+    endif()
+
+    install(FILES ${_bin_pyc} DESTINATION "${_py_install_dir}")
+    unset(_py_install_dir)
+    unset(_message)
 
+  endif("$ENV{PYTHONDONTWRITEBYTECODE}" STREQUAL "")
 endmacro(PYTHON_INSTALL)
 
