diff -urNp kdelibs-4.6.1.org/kparts/CMakeLists.txt kdelibs-4.6.1/kparts/CMakeLists.txt
--- kdelibs-4.6.1.org/kparts/CMakeLists.txt	2011-02-25 23:53:31.000000000 +0200
+++ kdelibs-4.6.1/kparts/CMakeLists.txt	2011-04-02 00:41:54.884508059 +0300
@@ -2,7 +2,10 @@ project(kparts)
 
 add_subdirectory( tests )
 
-include_directories(${KDE4_KIO_INCLUDES} ${kparts_BINARY_DIR} ${nepomuk_SOURCE_DIR} ${nepomuk_SOURCE_DIR}/core ${nepomuk_SOURCE_DIR}/types)
+include_directories(${KDE4_KIO_INCLUDES} ${kparts_BINARY_DIR})
+if(HAVE_NEPOMUK)
+include_directories(${nepomuk_SOURCE_DIR} ${nepomuk_SOURCE_DIR}/core ${nepomuk_SOURCE_DIR}/types)
+endif(HAVE_NEPOMUK)
 if(Soprano_FOUND)
 include_directories(${SOPRANO_INCLUDE_DIR})
 endif(Soprano_FOUND)
