Index: kdecore/io/ksavefile.cpp
===================================================================
--- trunk/KDE/kdelibs/kdecore/io/ksavefile.cpp	(revision 863691)
+++ trunk/KDE/kdelibs/kdecore/io/ksavefile.cpp	(revision 863692)
@@ -27,6 +27,7 @@
 
 #include <QtCore/QDir>
 #include <QProcess>
+#include <QTemporaryFile>
 
 #include <kconfig.h>
 #include <kde_file.h>
@@ -34,7 +35,6 @@
 #include <kstandarddirs.h>
 #include <kconfiggroup.h>
 #include <kcomponentdata.h>
-#include "ktemporaryfile.h"
 
 class KSaveFile::Private
 {
@@ -102,7 +102,7 @@
     }
 
     //Create our temporary file
-    KTemporaryFile tempFile(d->componentData);
+    QTemporaryFile tempFile;
     tempFile.setAutoRemove(false);
     tempFile.setFileTemplate(d->realFileName + "XXXXXX.new");
     if (!tempFile.open()) {
