Index: kdelibs/cmake/modules/create_exe_symlink.cmake
===================================================================
--- kdelibs/cmake/modules/create_exe_symlink.cmake	(revision 964573)
+++ kdelibs/cmake/modules/create_exe_symlink.cmake	(working copy)
@@ -11,6 +11,6 @@
 GET_FILENAME_COMPONENT(abs_target ${TARGET} ABSOLUTE)
 IF(UNIX)
     EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E create_symlink ${abs_target} ${abs_link_name})
-    EXECUTE_PROCESS(COMMAND chmod a+x ${abs_target})
+    EXECUTE_PROCESS(COMMAND chmod a+x $ENV{DESTDIR}/${abs_target})
 ENDIF(UNIX)
 # FIXME: WIN32 support