diff -U 3 -H -d -r -N -- todo_plasmoid/CMakeLists.txt todo_plasmoid_KDE42/CMakeLists.txt
--- todo_plasmoid/CMakeLists.txt	2008-10-14 19:52:45.000000000 +0200
+++ todo_plasmoid_KDE42/CMakeLists.txt	2009-02-03 14:10:03.000000000 +0100
@@ -2,7 +2,6 @@
 
 # search packages used by KDE
 find_package(KDE4 REQUIRED)  
-find_package(Plasma REQUIRED)
 include (KDE4Defaults)       
 include (MacroLibrary)       
 include(MacroOptionalAddSubdirectory)
diff -U 3 -H -d -r -N -- todo_plasmoid/src/applet/CMakeLists.txt todo_plasmoid_KDE42/src/applet/CMakeLists.txt
--- todo_plasmoid/src/applet/CMakeLists.txt	2008-10-14 19:52:45.000000000 +0200
+++ todo_plasmoid_KDE42/src/applet/CMakeLists.txt	2009-02-03 14:10:21.000000000 +0100
@@ -15,7 +15,7 @@
 kde4_add_ui_files(todoapplet_SRCS todoappletconfig.ui)
 
 kde4_add_plugin(todoapplet ${todoapplet_SRCS})
-target_link_libraries(todoapplet ${PLASMA_LIBS} ${KDE4_KDEUI_LIBS})
+target_link_libraries(todoapplet ${KDE4_PLASMA_LIBS} ${KDE4_KDEUI_LIBS})
 
 install(TARGETS todoapplet DESTINATION ${PLUGIN_INSTALL_DIR})
 
diff -U 3 -H -d -r -N -- todo_plasmoid/src/applet/todoapplet.cpp todo_plasmoid_KDE42/src/applet/todoapplet.cpp
--- todo_plasmoid/src/applet/todoapplet.cpp	2008-11-17 15:26:59.000000000 +0100
+++ todo_plasmoid_KDE42/src/applet/todoapplet.cpp	2009-02-03 14:12:28.000000000 +0100
@@ -38,7 +38,7 @@
 
 // plasma headers
 #include <Plasma/Dialog>
-#include <Plasma/Icon>
+#include <plasma/widgets/iconwidget.h>
 #include <Plasma/Theme>
 
 K_EXPORT_PLASMA_APPLET(todoapplet, TodoApplet)
@@ -209,7 +209,7 @@
             m_dialog->setLayout(dialogLayout);
 
             // panel icon
-            m_icon = new Plasma::Icon(KIcon("view-pim-tasks"), QString(), this);
+            m_icon = new Plasma::IconWidget(KIcon("view-pim-tasks"), QString(), this);
             m_layout->addItem(m_icon);
             connect(m_icon, SIGNAL(clicked()), SLOT(showDialog()));
 
diff -U 3 -H -d -r -N -- todo_plasmoid/src/applet/todoapplet.h todo_plasmoid_KDE42/src/applet/todoapplet.h
--- todo_plasmoid/src/applet/todoapplet.h	2008-11-17 15:27:03.000000000 +0100
+++ todo_plasmoid_KDE42/src/applet/todoapplet.h	2009-02-03 14:11:52.000000000 +0100
@@ -27,6 +27,7 @@
 // Plasma includes
 #include <Plasma/Applet>
 #include <Plasma/DataEngine>
+#include <plasma/widgets/iconwidget.h>
 
 namespace Plasma {
     class Dialog;
@@ -78,7 +79,7 @@
     Plasma::DataEngine    *m_engine;
 
     Plasma::Dialog        *m_dialog; // dialog in case ****
-    Plasma::Icon          *m_icon; // panel icon in case ****
+    Plasma::IconWidget    *m_icon; // panel icon in case ****
     QGraphicsLinearLayout *m_layout; // main applet layout
     QComboBox             *m_types; // todos categories combobox
     QGraphicsProxyWidget  *m_proxyWidget; // proxy widget
diff -U 3 -H -d -r -N -- todo_plasmoid/src/dataengine/CMakeLists.txt todo_plasmoid_KDE42/src/dataengine/CMakeLists.txt
--- todo_plasmoid/src/dataengine/CMakeLists.txt	2008-10-14 19:52:45.000000000 +0200
+++ todo_plasmoid_KDE42/src/dataengine/CMakeLists.txt	2009-02-03 14:10:44.000000000 +0100
@@ -8,7 +8,7 @@
 )
 
 kde4_add_plugin(plasma_engine_todo ${todo_engine_SRCS})
-target_link_libraries(plasma_engine_todo ${KDE4_KDECORE_LIBS} ${PLASMA_LIBS} 
+target_link_libraries(plasma_engine_todo ${KDE4_KDECORE_LIBS} ${KDE4_PLASMA_LIBS} 
                                          ${KDE4_KDEUI_LIBS} ${KDE4_KCAL_LIBS} ${KDE4_KPIM_LIBS})
 
 install(TARGETS plasma_engine_todo DESTINATION ${PLUGIN_INSTALL_DIR})
