--- src/xsd/CMakeLists.txt.orig	2009-02-28 23:23:33.000000000 +0300
+++ src/xsd/CMakeLists.txt	2009-03-01 00:08:08.000000000 +0300
@@ -22,6 +22,8 @@ endif (LIBXML2_FOUND)
 # javac >= 1.5 is required to compile the xsd parser
 OPTION(ENABLE_REGENERATEXSD
   "regenerate xsd parser from *.xsd files(not normally required)" ON)
+
+if(${ENABLE_REGENERATEXSD})
 find_package(Java)
 if (JAVA_COMPILE)
   execute_process(COMMAND ${JAVA_COMPILE} "-version" ERROR_VARIABLE
@@ -31,6 +33,7 @@ if (JAVA_COMPILE)
   endif (JAVAC_VERSION MATCHES " 1\\.[56]" AND NOT JAVAC_VERSION MATCHES "g[ci]j")
   MESSAGE("Java5 found ${JAVAC5_FOUND}")
 endif (JAVA_COMPILE)
+endif(${ENABLE_REGENERATEXSD})
 
 # loop over all xsd files
 set(XSDFILES "${CMAKE_CURRENT_SOURCE_DIR}/strigidaemonconfiguration.xsd" 
