diff -Nuar strigi-0.6.5.orig/src/estraierindexer/estraierindexmanager.cpp strigi-0.6.5/src/estraierindexer/estraierindexmanager.cpp
--- strigi-0.6.5.orig/src/estraierindexer/estraierindexmanager.cpp	2008-05-19 22:30:35.000000000 +0200
+++ strigi-0.6.5/src/estraierindexer/estraierindexmanager.cpp	2009-05-18 11:06:21.296581521 +0200
@@ -23,6 +23,7 @@
 #include "estraierindexwriter.h"
 #include "strigi_thread.h"
 #include "indexplugin.h"
+#include <cstdio>
 #include <cstring>
 #include <iostream>
 #include <assert.h>
diff -Nuar strigi-0.6.5.orig/src/estraierindexer/estraierindexreader.cpp strigi-0.6.5/src/estraierindexer/estraierindexreader.cpp
--- strigi-0.6.5.orig/src/estraierindexer/estraierindexreader.cpp	2008-05-19 22:30:35.000000000 +0200
+++ strigi-0.6.5/src/estraierindexer/estraierindexreader.cpp	2009-05-18 11:17:30.476583471 +0200
@@ -21,6 +21,7 @@
 #include "estraierindexmanager.h"
 #include "query.h"
 #include <estraier.h>
+#include <cstdio>
 #include <cstring>
 #include <set>
 #include <sstream>
