--- strigi-0.6.4.orig/src/daemon/dbus/dbuscpp/dbusmessagereader.h
+++ strigi-0.6.4/src/daemon/dbus/dbuscpp/dbusmessagereader.h
@@ -23,6 +23,7 @@
 #define DBUS_API_SUBJECT_TO_CHANGE 1
 #include <dbus/dbus.h>
 
+#include <stdint.h>
 #include <set>
 #include <map>
 #include <string>
--- strigi-0.6.4.orig/src/daemon/eventlistener/inotifylistener.h
+++ strigi-0.6.4/src/daemon/eventlistener/inotifylistener.h
@@ -24,6 +24,7 @@
 #include "strigi_thread.h"
 #include <map>
 #include <vector>
+#include <stdio.h>
 
 class Event;
 class PollingListener;
--- strigi-0.6.4.orig/src/daemon/strigithread.h
+++ strigi-0.6.4/src/daemon/strigithread.h
@@ -22,6 +22,7 @@
 
 #include <strigi_thread.h>
 #include <string>
+#include <stdio.h>
 
 extern "C" void *threadstarter(void *);
 
--- strigi-0.6.4.orig/src/streamanalyzer/indexpluginloader.h
+++ strigi-0.6.4/src/streamanalyzer/indexpluginloader.h
@@ -25,6 +25,7 @@
 #include <vector>
 #include <map>
 #include <string>
+#include <stdio.h>
 
 namespace Strigi {
   class IndexManager;
