--- Panelspacer_from_svn_pour_kdelook-tmp.original/CMakeLists.txt
+++ Panelspacer_from_svn_pour_kdelook-tmp/CMakeLists.txt
@@ -3,7 +3,12 @@
 # Find the required Libaries
 find_package(KDE4 REQUIRED)
 include(KDE4Defaults)
+
+if ( ${KDE_VERSION} VERSION_LESS "4.1.90" )
 find_package(Plasma REQUIRED)
+else ( ${KDE_VERSION} VERSION_LESS "4.1.90" )
+set( PLASMA_LIBS ${KDE4_PLASMA_LIBS} )
+endif ( ${KDE_VERSION} VERSION_LESS "4.1.90" )
 
 add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 include_directories(
