From d413b91839ca0e66beb56a71625e3194df1a6599 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Wed, 12 Jun 2019 19:44:12 +0200
Subject: Use non-deprecated KDE_INSTALL_TARGETS_DEFAULT_ARGS for alkimia target

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -147,7 +147,7 @@ set(INCLUDE_INSTALL_DIR include/alkimia/${ALKIMIA_INCLUDE_SUFFIX})
 
 install(TARGETS alkimia
     EXPORT  alkimiaTargets
-    ${INSTALL_TARGETS_DEFAULT_ARGS}
+    ${KDE_INSTALL_TARGETS_DEFAULT_ARGS}
 )
 
 if(BUILD_QT4)
-- 
2.23.0
