--- kmail/CMakeLists.txt.orig	2010-04-01 00:45:22.000000000 +0200
+++ kmail/CMakeLists.txt	2010-04-01 00:45:41.000000000 +0200
@@ -223,6 +223,13 @@
   customtemplates_kfg.kcfgc
 )
 
+kde4_add_kcfg_files(kmailprivate_LIB_SRCS
+  ../messagecomposer/messagecomposersettings_base.kcfgc USE_RELATIVE_PATH
+  ../messageviewer/globalsettings_base.kcfgc USE_RELATIVE_PATH
+  ../templateparser/customtemplates_kfg.kcfgc USE_RELATIVE_PATH
+  ../templateparser/globalsettings_base.kcfgc USE_RELATIVE_PATH
+  ../templateparser/templatesconfiguration_kfg.kcfgc USE_RELATIVE_PATH
+)
 
 kde4_add_library(kmailprivate SHARED ${kmailprivate_LIB_SRCS})
 target_link_libraries(kmailprivate
