--- a/app/CMakeLists.txt	2015-09-18 23:08:00.449716542 +0200
+++ b/app/CMakeLists.txt	2015-09-18 23:11:31.939072561 +0200
@@ -88,7 +88,8 @@
     ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
 install(FILES gwenviewui.rc
-    DESTINATION ${KDE_INSTALL_KXMLGUI5DIR}/gwenview)
+    DESTINATION ${KDE_INSTALL_KXMLGUI5DIR}/org.kde.gwenview
+    RENAME org.kde.gwenviewui.rc)
 
 install(PROGRAMS org.kde.gwenview.desktop
     DESTINATION ${KDE_INSTALL_APPDIR})
