diff -ruN perlqt-4.7.2.orig/CMakeLists.txt perlqt-4.7.2/CMakeLists.txt
--- perlqt-4.7.2.orig/CMakeLists.txt	2011-05-20 22:33:46.000000000 +0200
+++ perlqt-4.7.2/CMakeLists.txt	2011-10-19 01:03:04.040639531 +0200
@@ -38,7 +38,7 @@
 add_definitions(-DDEBUG)
 
 include (FindPerlMore)
-set(CUSTOM_PERL_SITE_ARCH_DIR ${PERL_SITE_ARCH_DIR} CACHE DIR "Custom installation directory for perl binary extension")
+set(CUSTOM_PERL_SITE_ARCH_DIR ${PERL_VENDOR_ARCH_DIR} CACHE DIR "Custom installation directory for perl binary extension")
 
 # the RPATH to be used when installing, but only if it's not a system directory
 GET_FILENAME_COMPONENT(SMOKE_LIB_DIR ${SMOKE_BASE_LIBRARY} PATH)
