diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0efd25..efcdbe0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,6 +21,7 @@ option(AKONADI_BUILD_QSQLITE "Build the Sqlite backend." TRUE)
 option(AKONADI_USE_STRIGI_SEARCH "Build Akonadi with strigi as search engine." FALSE)
 option(INSTALL_QSQLITE_IN_QT_PREFIX "Install the QSQLite plugin in QT_PLUGIN_DIR" FALSE)
 option(STATIC_LIBRARY "Build Akonadi as a static library." FALSE)
+OPTION(WITH_QT5 "Build against Qt5." TRUE)
 
 if(NOT DEFINED DATABASE_BACKEND)
   set(DATABASE_BACKEND "MYSQL" CACHE STRING "The default database backend to use for Akonadi. Can be either MYSQL, POSTGRES or SQLITE")
@@ -128,7 +129,10 @@ set_default_db_backend(${DATABASE_BACKEND})
 set(QT_USE_IMPORTED_TARGETS TRUE) # Qt 4 only
 set(QT_MIN_VERSION 4.6.0)         # Qt 4 only
 
-find_package(Qt5Core QUIET)
+if (WITH_QT5)
+  find_package(Qt5Core QUIET)
+endif()
+
 if (Qt5Core_FOUND)
   find_package(Qt5Gui REQUIRED)
   find_package(Qt5Widgets REQUIRED)
