diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3f76283..cc05451 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,8 +50,13 @@ if (BUILD_EXAMPLES)
     add_subdirectory(examples)
 endif(BUILD_EXAMPLES)
 
-enable_testing(true)
-add_subdirectory(tests)
+option(DISABLE_TESTING "Disable unit tests" FALSE)
+if (DISABLE_TESTING)
+    message(STATUS "Disabling unit tests")
+else()
+    enable_testing(true)
+    add_subdirectory(tests)
+endif()
 
 set(NetworkManagerQt_HEADERS
     accesspoint.h
