diff --git a/CMakeLists.txt b/CMakeLists.txt
index c892cbd..5ff3d05 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,7 +36,14 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${SIP_INCLUDE_DIR} ${PYTHON_INCL
 set(SIP_INCLUDES ${PYQT4_SIP_DIR} ${PYKDE4_SIP_DIR} sip/kate)
 set(SIP_CONCAT_PARTS 1)
 set(SIP_TAGS ALL WS_X11 ${PYQT4_VERSION_TAG} ${PYKDE4_VERSION_TAG})
-set(SIP_EXTRA_OPTIONS -g)
+
+if(PYQT4_VERSION LESS "040905")
+    set(SIP_EXTRA_OPTIONS -g)
+else (PYQT4_VERSION LESS "040905")
+    # Disable QVector<int> for newer PyQt
+    set(SIP_EXTRA_OPTIONS -g -x PyKDE_QVector)
+endif(PYQT4_VERSION LESS "040905")
+
 add_sip_python_module(PyKDE4.kate sip/kate/katemod.sip ${KDE4_KTEXTEDITOR_LIBS} kateinterfaces)
 
 
