diff -ur kdevplatform-1.2.0/language/duchain/repositories/itemrepository.h kdevplatform-1.2.0-gcc46/language/duchain/repositories/itemrepository.h
--- kdevplatform-1.2.0/language/duchain/repositories/itemrepository.h	2011-01-23 20:29:25.000000000 +0100
+++ kdevplatform-1.2.0-gcc46/language/duchain/repositories/itemrepository.h	2011-01-29 02:54:27.000000000 +0100
@@ -236,7 +236,7 @@
 
     QString m_name;
     int m_version;
-    mutable ItemRepositoryRegistry& m_registry;
+    ItemRepositoryRegistry& m_registry;
     AbstractRepositoryManager* (*m_shareMutex)();
 };
 
diff -ur kdevplatform-1.2.0/language/util/basicsetrepository.h kdevplatform-1.2.0-gcc46/language/util/basicsetrepository.h
--- kdevplatform-1.2.0/language/util/basicsetrepository.h	2011-01-23 20:29:25.000000000 +0100
+++ kdevplatform-1.2.0-gcc46/language/util/basicsetrepository.h	2011-01-29 02:53:21.000000000 +0100
@@ -129,7 +129,7 @@
   SetNodeData data;
   
   uint m_hash;
-  mutable SetDataRepository& repository;
+  SetDataRepository& repository;
   mutable BasicSetRepository* setRepository; //May be zero when no notifications are wanted
   mutable bool m_created;
 };
