diff -r -u kdebindings-ruby-4.4.90/ruby/nepomuk/CMakeLists.txt /home/huettel/tmp/kdebindings-ruby-4.4.90-modified/ruby/nepomuk/CMakeLists.txt
--- kdebindings-ruby-4.4.90/ruby/nepomuk/CMakeLists.txt	2010-05-05 11:39:10.000000000 +0200
+++ /home/huettel/tmp/kdebindings-ruby-4.4.90-modified/ruby/nepomuk/CMakeLists.txt	2010-07-03 19:53:03.000000000 +0200
@@ -7,6 +7,8 @@
 target_link_libraries(rubynepomuk 
     ${RUBY_LIBRARY} 
     ${NEPOMUK_LIBRARIES}
+    nepomuk
+    soprano
     smokeqtcore 
     smokesoprano
     smokekdecore 
diff -r -u kdebindings-ruby-4.4.90/ruby/qtscript/CMakeLists.txt /home/huettel/tmp/kdebindings-ruby-4.4.90-modified/ruby/qtscript/CMakeLists.txt
--- kdebindings-ruby-4.4.90/ruby/qtscript/CMakeLists.txt	2010-05-20 13:43:58.000000000 +0200
+++ /home/huettel/tmp/kdebindings-ruby-4.4.90-modified/ruby/qtscript/CMakeLists.txt	2010-07-03 19:43:39.000000000 +0200
@@ -4,7 +4,7 @@
 set(qtscript_LIBRARY_SRC qtscript.cpp qtscripthandlers.cpp)
 
 add_library(qtscript MODULE ${qtscript_LIBRARY_SRC})
-target_link_libraries(qtscript ${QT_QTCORE_LIBRARY} ${RUBY_LIBRARY} smokebase smokeqtcore smokeqtscript qtruby4shared)
+target_link_libraries(qtscript ${QT_QTCORE_LIBRARY} ${RUBY_LIBRARY} QtScript smokebase smokeqtcore smokeqtscript qtruby4shared)
 set_target_properties(qtscript PROPERTIES PREFIX "")
 install(TARGETS qtscript DESTINATION ${CUSTOM_RUBY_SITE_ARCH_DIR})
 install(FILES qtscript.rb DESTINATION ${CUSTOM_RUBY_SITE_LIB_DIR}/qtscript)
diff -r -u kdebindings-ruby-4.4.90/ruby/soprano/CMakeLists.txt /home/huettel/tmp/kdebindings-ruby-4.4.90-modified/ruby/soprano/CMakeLists.txt
--- kdebindings-ruby-4.4.90/ruby/soprano/CMakeLists.txt	2010-05-20 13:43:57.000000000 +0200
+++ /home/huettel/tmp/kdebindings-ruby-4.4.90-modified/ruby/soprano/CMakeLists.txt	2010-07-03 19:44:51.000000000 +0200
@@ -3,7 +3,7 @@
 
 set(rubysoprano_LIBRARY_SRC soprano.cpp sopranohandlers.cpp)
 add_library(rubysoprano MODULE ${rubysoprano_LIBRARY_SRC})
-target_link_libraries(rubysoprano ${QT_QTCORE_LIBRARY} ${RUBY_LIBRARY} smokebase smokeqtcore smokesoprano qtruby4shared)
+target_link_libraries(rubysoprano ${QT_QTCORE_LIBRARY} ${RUBY_LIBRARY} soprano smokebase smokeqtcore smokesoprano qtruby4shared)
 set_target_properties(rubysoprano PROPERTIES PREFIX "" OUTPUT_NAME soprano)
 
 install(TARGETS rubysoprano DESTINATION ${CUSTOM_RUBY_SITE_ARCH_DIR})
