diff -urN plasma-playground-20080907.old/applets/CMakeLists.txt plasma-playground-20080907/applets/CMakeLists.txt
--- plasma-playground-20080907.old/applets/CMakeLists.txt	2008-09-07 11:41:36.000000000 +0200
+++ plasma-playground-20080907/applets/CMakeLists.txt	2008-09-07 12:00:32.000000000 +0200
@@ -4,12 +4,12 @@
 FIND_PACKAGE(RUBY)
 
 #add_subdirectory(network)
-add_subdirectory(notify)
+#add_subdirectory(notify)
 #add_subdirectory(java)
 add_subdirectory(groupphoto)
 add_subdirectory(slideInSlideOut)
-add_subdirectory(extenderapplet)
-add_subdirectory(kuiserver)
+#add_subdirectory(extenderapplet)
+#add_subdirectory(kuiserver)
 add_subdirectory(unitconverter)
 
 #add_subdirectory(fortune-teller)
@@ -40,8 +40,8 @@
 if(QT_QTWEBKIT_FOUND)
    MESSAGE(STATUS "WebKit found, building those widgets as well")
    add_subdirectory(cia.vc)
-   add_subdirectory(weather)
-   add_subdirectory(scriptedhtml)
+   #add_subdirectory(weather)
+   #add_subdirectory(scriptedhtml)
 endif(QT_QTWEBKIT_FOUND)
 
 if(BLITZ_FOUND)
@@ -57,22 +57,22 @@
 add_subdirectory(bball)
 add_subdirectory(spacer)
 add_subdirectory(grid)
-add_subdirectory(mid_control)
-add_subdirectory(news)
+#add_subdirectory(mid_control)
+#add_subdirectory(news)
 add_subdirectory(rssnow)
 add_subdirectory(system-monitor)
 add_subdirectory(systembar)
-add_subdirectory(svgpaneltest)
+#add_subdirectory(svgpaneltest)
 add_subdirectory(serverhotlink)
 if ( Q_WS_X11 )
    add_subdirectory(menubar)
    add_subdirectory(keyboardLeds)
    add_subdirectory(paste)
-   add_subdirectory(quicklaunch)
-   add_subdirectory(peachydock)
-   add_subdirectory(systemcommand)
+   #add_subdirectory(quicklaunch)
+   #add_subdirectory(peachydock)
+   #add_subdirectory(systemcommand)
    add_subdirectory(embed-win)
-   add_subdirectory(systray-refactor)
+   #add_subdirectory(systray-refactor)
 endif ( Q_WS_X11 )
 ###uncomment this section when FindDecibel.cmake exists
 #macro_optional_find_package(Decibel)
@@ -82,16 +82,16 @@
 #endif (DECIBEL_FOUND)
 add_subdirectory(commandwatch)
 add_subdirectory(networkmanager)
-add_subdirectory(previewer)
+#add_subdirectory(previewer)
 add_subdirectory(toggle-compositing)
-add_subdirectory(webbrowser)
+#add_subdirectory(webbrowser)
 add_subdirectory(scriptedimage)
-add_subdirectory(weather_station)
+#add_subdirectory(weather_station)
 add_subdirectory(darkstat)
 add_subdirectory(leavenote)
 add_subdirectory(incomingmsg)
-add_subdirectory(wifi-signal-strength)
-add_subdirectory(activitybar)
+#add_subdirectory(wifi-signal-strength)
+#add_subdirectory(activitybar)
 
 include(MacroOptionalDependPackage)
 
@@ -103,7 +103,7 @@
 endif(DEPEND_PACKAGE_MarbleWidget)
 
 if(MARBLEWIDGET_FOUND)
-	 add_subdirectory(contacts)
+	#add_subdirectory(contacts)
 endif(MARBLEWIDGET_FOUND)
 
 if(WIN32)
@@ -112,5 +112,5 @@
 
 # add_subdirectory(embedded-ivan)
 
-add_subdirectory(eyes)
+#add_subdirectory(eyes)
 
diff -urN plasma-playground-20080907.old/engines/CMakeLists.txt plasma-playground-20080907/engines/CMakeLists.txt
--- plasma-playground-20080907.old/engines/CMakeLists.txt	2008-09-07 11:41:36.000000000 +0200
+++ plasma-playground-20080907/engines/CMakeLists.txt	2008-09-07 12:01:54.000000000 +0200
@@ -1,4 +1,4 @@
-add_subdirectory(kuiserver)
+#add_subdirectory(kuiserver)
 add_subdirectory(networkmanager)
 add_subdirectory(notifications)
 add_subdirectory(timetracker)
diff -urN plasma-playground-20080907.old/screenmanagement/CMakeLists.txt plasma-playground-20080907/screenmanagement/CMakeLists.txt
--- plasma-playground-20080907.old/screenmanagement/CMakeLists.txt	2008-09-07 11:41:36.000000000 +0200
+++ plasma-playground-20080907/screenmanagement/CMakeLists.txt	2008-09-07 12:05:50.000000000 +0200
@@ -2,5 +2,5 @@
 SET (CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/kephal/libkephal)
 
 ADD_SUBDIRECTORY (kephal)
-ADD_SUBDIRECTORY (screensengine)
-ADD_SUBDIRECTORY (screensapplet)
+#ADD_SUBDIRECTORY (screensengine)
+#ADD_SUBDIRECTORY (screensapplet)
