diff --git a/CMakeLists.txt b/CMakeLists.txt
index 97d5e82..f59e361 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,8 +10,7 @@ set (CMAKE_MODULE_PATH
 cmake_policy(SET CMP0002 OLD)
 
 find_package (Qt5 REQUIRED CONFIG COMPONENTS
-              Qml
-              Test)
+              Qml)
 
 find_package (KF5 REQUIRED COMPONENTS
               CoreAddons
@@ -101,7 +100,11 @@ ecm_setup_version(${KTP_VERSION} VARIABLE_PREFIX KTP
 add_subdirectory(KTp)
 add_subdirectory(tools)
 add_subdirectory(data)
-add_subdirectory(tests)
+
+if(BUILD_TESTING)
+    find_package(Qt5Test REQUIRED CONFIG)
+    add_subdirectory(tests)
+endif()
 
 if (OTR_LIBS_FOUND)
     include_directories (${LIBOTR_INCLUDE_DIR}
