--- configure.ac.orig	2010-09-20 21:51:50.289828895 +0200
+++ configure.ac	2010-09-20 21:58:03.536293089 +0200
@@ -49,10 +49,19 @@
 
 GTK_REQUIRED_VERSION=2.16
 
+AC_ARG_ENABLE([gtk],
+	AC_HELP_STRING([--disable-gtk], [Disable libdbusmenu-gtk library]),
+	[enable_gtk=$enableval], [enable_gtk=auto])
+AM_CONDITIONAL([WANT_LIBDBUSMENUGTK], [test "x$enable_gtk" != "xno"])
+
+AS_IF([test "x$enable_gtk" != "xno"],[
 PKG_CHECK_MODULES(DBUSMENUGTK,  gtk+-2.0 >= $GTK_REQUIRED_VERSION
                                 glib-2.0 >= $GLIB_REQUIRED_VERSION
                                 dbus-glib-1 >= $DBUS_REQUIRED_VERSION
-                                libxml-2.0 >= $XML_REQUIRED_VERSION)
+                                libxml-2.0 >= $XML_REQUIRED_VERSION,
+                                [have_gtk=yes]
+)
+])
 
 AC_SUBST(DBUSMENUGTK_CFLAGS)
 AC_SUBST(DBUSMENUGTK_LIBS)
@@ -193,1 +192,6 @@
 
+AS_IF([test "x$have_gtk" = "xyes"],
+	AC_MSG_NOTICE([	Gtk:                    yes]),
+	AC_MSG_NOTICE([	Gtk:                    no])
+)
+
--- Makefile.am.orig	2010-09-20 21:58:41.616751481 +0200
+++ Makefile.am	2010-09-20 21:59:23.492035983 +0200
@@ -5,5 +5,9 @@
 
+if WANT_LIBDBUSMENUGTK
+LIBDBUSMENUGTK_SUBDIR = libdbusmenu-gtk
+endif
+
 SUBDIRS = \
 	libdbusmenu-glib \
-	libdbusmenu-gtk \
+	$(LIBDBUSMENUGTK_SUBDIR) \
 	tools \
--- tests/Makefile.am.orig	2011-01-29 14:37:11.863805482 +0100
+++ tests/Makefile.am	2011-01-29 14:38:48.910883303 +0100
@@ -11,12 +11,16 @@
 	test-glib-proxy \
 	test-glib-simple-items \
 	test-glib-submenu \
-	test-json \
+	test-json
+
+if WANT_LIBDBUSMENUGTK
+TESTS += \
 	test-gtk-objects-test \
 	test-gtk-label \
 	test-gtk-shortcut \
 	test-gtk-reorder \
 	test-gtk-submenu
+endif
 
 check_PROGRAMS = \
 	glib-server-nomenu \
@@ -30,19 +34,23 @@
 	test-glib-proxy-client \
 	test-glib-proxy-server \
 	test-glib-proxy-proxy \
-	test-gtk-objects \
 	test-glib-submenu-client \
 	test-glib-submenu-server \
+	test-glib-simple-items \
+	test-json-client \
+	test-json-server
+
+if WANT_LIBDBUSMENUGTK
+check_PROGRAMS += \
+	test-gtk-objects \
 	test-gtk-label-client \
 	test-gtk-label-server \
 	test-gtk-shortcut-client \
 	test-gtk-shortcut-server \
-	test-glib-simple-items \
 	test-gtk-reorder-server \
-	test-json-client \
-	test-json-server \
 	test-gtk-submenu-server \
 	test-gtk-submenu-client
+endif
 
 XVFB_RUN=". $(srcdir)/run-xvfb.sh"
 

