--- configure.ac.orig	2010-09-20 21:51:50.289828895 +0200
+++ configure.ac	2010-09-20 22:05:38.292979567 +0200
@@ -63,8 +63,17 @@
 
 X11_REQUIRED_VERSION=1.3
 
+AC_ARG_ENABLE([dumper],
+	AC_HELP_STRING([--disable-dumper], [Disable dbusmenudumper tool]),
+	[enable_dumper=$enableval], [enable_dumper=auto])
+AM_CONDITIONAL([WANT_DBUSMENUDUMPER], [test "x$enable_dumper" != "xno"])
+
+AS_IF([test "x$enable_dumper" != "xno"],[
 PKG_CHECK_MODULES(DBUSMENUDUMPER, gtk+-2.0 >= $GTK_REQUIRED_VERSION
-                                  x11 >= $X11_REQUIRED_VERSION)
+                                  x11 >= $X11_REQUIRED_VERSION,
+                                  [have_dumper=yes]
+)
+])
 
 AC_SUBST(DBUSMENUDUMPER_CFLAGS)
 AC_SUBST(DBUSMENUDUMPER_LIBS)
@@ -193,1 +193,6 @@
 
+AS_IF([test "x$have_dumper" = "xyes"],
+   AC_MSG_NOTICE([	Dumper:                 yes]),
+   AC_MSG_NOTICE([	Dumper:                 no])
+)
+
--- tools/Makefile.am.orig	2010-09-20 22:02:07.280175897 +0200
+++ tools/Makefile.am	2010-09-20 22:05:15.621664358 +0200
@@ -3,5 +3,9 @@
 
-libexec_PROGRAMS = dbusmenu-dumper
+libexec_PROGRAMS =
+
+if WANT_DBUSMENUDUMPER
+libexec_PROGRAMS += dbusmenu-dumper
+endif
 
 libexec_SCRIPTS = dbusmenu-bench
 
--- tests/Makefile.am.orig	2011-02-07 00:01:11.066461559 +0100
+++ tests/Makefile.am	2011-02-07 00:02:33.487751202 +0100
@@ -10,6 +10,10 @@
 	test-glib-properties \
 	test-glib-proxy \
 	test-glib-simple-items \
-	test-glib-submenu \
+	test-glib-submenu
+
+if WANT_DBUSMENUDUMPER
+TESTS += \
 	test-json
+endif
 
@@ -36,7 +40,11 @@
 	test-glib-proxy-proxy \
 	test-glib-submenu-client \
 	test-glib-submenu-server \
-	test-glib-simple-items \
+	test-glib-simple-items
+
+if WANT_DBUSMENUDUMPER
+check_PROGRAMS += \
 	test-json-client \
 	test-json-server
+endif
 
