--- ./libdbusmenu-glib/Makefile.am.orig	2011-02-05 16:16:34.481082458 +0100
+++ ./libdbusmenu-glib/Makefile.am	2011-02-05 16:20:31.703396315 +0100
@@ -138,7 +138,7 @@
 
 introspection_sources = $(libdbusmenu_glibinclude_HEADERS)
 
-Dbusmenu_Glib-0.4.gir: libdbusmenu-glib.la
+Dbusmenu-Glib-0.4.gir: libdbusmenu-glib.la
 Dbusmenu_Glib_0_4_gir_INCLUDES = \
 	GObject-2.0
 Dbusmenu_Glib_0_4_gir_CFLAGS = $(DBUSMENUGLIB_CFLAGS) -I$(top_srcdir)
