--- tools/testapp/Makefile.am.orig	2011-01-29 20:50:54.266939219 +0100
+++ tools/testapp/Makefile.am	2011-01-29 20:51:16.185268917 +0100
@@ -1,10 +1,4 @@
 
-if USE_GTK3
-VER=3
-else
-VER=
-endif
-
 libexec_PROGRAMS =
 
 if WANT_TESTS
@@ -22,6 +16,4 @@
 
 dbusmenu_testapp_LDADD = \
 	$(builddir)/../../libdbusmenu-glib/libdbusmenu-glib.la \
-	$(builddir)/../../libdbusmenu-gtk/libdbusmenu-gtk$(VER).la \
-	$(DBUSMENUGTK_LIBS) \
 	$(DBUSMENUTESTS_LIBS)
