--- kcontrol/kxkb/xklavier_adaptor.cpp.old	2009-07-20 01:25:48.780492221 +0300
+++ kcontrol/kxkb/xklavier_adaptor.cpp	2009-07-20 01:26:02.047358319 +0300
@@ -222,7 +222,7 @@
     kDebug() << "Xklavier initialized";
     priv->config = xkl_config_registry_get_instance(priv->engine);
 
-    xkl_config_registry_load(priv->config);
+    xkl_config_registry_load(priv->config,false);
 	
     void *userData = priv;
 
