
--- a/kcontrol/fonts/fonts.cpp
+++ b/kcontrol/fonts/fonts.cpp
@@ -513,12 +513,12 @@
 
   // NOTE: keep in sync with kdelibs/kdeui/kernel/kglobalsettings.cpp
 
-  QFont f0("Sans Serif", 10);
-  QFont f1("Monospace", 10);
-  QFont f2("Sans Serif", 8);
-  QFont f3("Sans Serif", 9, QFont::Bold);
-  QFont f4("Sans Serif", 10);
-  QFont f5("Sans Serif", 8); // smallestReadableFont
+  QFont f0("DejaVu Sans", 9);
+  QFont f1("DejaVu Sans Mono", 9);
+  QFont f2("DejaVu Sans", 8);
+  QFont f3("DejaVu Sans", 9, QFont::Bold);
+  QFont f4("DejaVu Sans", 9);
+  QFont f5("DejaVu Sans", 8); // smallestReadableFont
 
   defaultFontList << f0 << f1 << f5 << f2 << f0 << f3 << f4 << f0;
 

