diff --git a/kcontrol/screensaver/screensaver.ui b/kcontrol/screensaver/screensaver.ui
index 6524e27..abf294a 100644
--- a/kcontrol/screensaver/screensaver.ui
+++ b/kcontrol/screensaver/screensaver.ui
@@ -108,6 +108,9 @@
           <property name="text">
            <string>&amp;Desktop Widgets</string>
           </property>
+          <property name="enabled">
+           <bool>false</bool>
+          </property>
          </widget>
         </item>
         <item>
