--- kdebindings.orig/csharp/qtscript/CMakeLists.txt
+++ kdebindings/csharp/qtscript/CMakeLists.txt
@@ -15,7 +15,7 @@ INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/
 	${CMAKE_CURRENT_SOURCE_DIR}/../qyoto/src ${QT_INCLUDES})
 LINK_DIRECTORIES (${LIBRARY_OUTPUT_PATH})
 ADD_LIBRARY(qtscript-sharp MODULE ${SRC_CPP})
-TARGET_LINK_LIBRARIES(qtscript-sharp smokeqt smokeqtscript qyotoshared ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY})
+TARGET_LINK_LIBRARIES(qtscript-sharp smokeqt smokeqtscript qyotoshared ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTSCRIPT_LIBRARY})
 
 SET(CS_FLAGS -r:${LIBRARY_OUTPUT_PATH}/qt-dotnet.dll -warn:0 -keyfile:${KEYFILE})
 ADD_CS_LIBRARY(qtscript "${SRC_CS}" ALL)
--- kdebindings.orig/csharp/qtwebkit/CMakeLists.txt
+++ kdebindings/csharp/qtwebkit/CMakeLists.txt
@@ -14,7 +14,7 @@ INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/
 	${CMAKE_CURRENT_SOURCE_DIR}/../qyoto/src ${QT_INCLUDES})
 LINK_DIRECTORIES (${LIBRARY_OUTPUT_PATH})
 ADD_LIBRARY(qtwebkit-sharp MODULE ${SRC_CPP})
-TARGET_LINK_LIBRARIES(qtwebkit-sharp smokeqt smokeqtwebkit qyotoshared ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY})
+TARGET_LINK_LIBRARIES(qtwebkit-sharp smokeqt smokeqtwebkit qyotoshared ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTWEBKIT_LIBRARY})
 
 SET(CS_FLAGS -r:${LIBRARY_OUTPUT_PATH}/qt-dotnet.dll -warn:0 -keyfile:${KEYFILE})
 ADD_CS_LIBRARY(qtwebkit "${SRC_CS}" ALL)
--- kdebindings.orig/csharp/qyoto/CMakeLists.txt
+++ kdebindings/csharp/qyoto/CMakeLists.txt
@@ -39,11 +39,11 @@ INCLUDE_DIRECTORIES (${QT_INCLUDES} ${CM
 LINK_DIRECTORIES (${LIBRARY_OUTPUT_PATH})
 
 ADD_LIBRARY (qyotoshared SHARED ${SRC_QYOTO_SHARED})
-TARGET_LINK_LIBRARIES (qyotoshared smokeqt ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY})
+TARGET_LINK_LIBRARIES (qyotoshared smokeqt ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTNETWORK_LIBRARY})
 set_target_properties(qyotoshared PROPERTIES VERSION 1.0.0 SOVERSION 1)
 
 ADD_LIBRARY (qyoto MODULE ${SRC_QYOTO})
-TARGET_LINK_LIBRARIES (qyoto qyotoshared smokeqt ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY})
+TARGET_LINK_LIBRARIES (qyoto qyotoshared smokeqt ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTDBUS_LIBRARY})
 
 SET(CS_FLAGS -debug -define:DEBUG -unsafe -warn:0 -keyfile:${KEYFILE})
 ADD_CS_LIBRARY(qt-dotnet "${SRC_CS_QYOTO}" ALL)
--- kdebindings.orig/csharp/soprano/CMakeLists.txt
+++ kdebindings/csharp/soprano/CMakeLists.txt
@@ -21,7 +21,7 @@ INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/
 LINK_DIRECTORIES (${LIBRARY_OUTPUT_PATH})
 ADD_LIBRARY(soprano-sharp MODULE ${SRC_CPP})
 ADD_DEPENDENCIES(soprano-sharp smokesoprano)
-TARGET_LINK_LIBRARIES(soprano-sharp smokeqt smokesoprano qyotoshared ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY})
+TARGET_LINK_LIBRARIES(soprano-sharp smokeqt smokesoprano qyotoshared ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${SOPRANO_LIBRARIES})
 
 SET(CS_FLAGS -r:${LIBRARY_OUTPUT_PATH}/qt-dotnet.dll -warn:0 -keyfile:${KEYFILE})
 ADD_CS_LIBRARY(soprano "${SRC_CS}" ALL)
