diff --git a/src/actionplugins/audio/helper/bluedevil-audio.desktop b/src/actionplugins/audio/helper/bluedevil-audio.desktop
index 79efae3..acdfa58 100644
--- a/src/actionplugins/audio/helper/bluedevil-audio.desktop
+++ b/src/actionplugins/audio/helper/bluedevil-audio.desktop
@@ -124,5 +124,5 @@ Comment[zh_CN]=连接
 Comment[zh_TW]=BlueDevil 音效連線
 Icon=preferences-system-bluetooth
 Terminal=false
-Categories=Qt;KDE;X-Bluetooth;Network;Audio;
+Categories=Qt;KDE;X-Bluetooth;Network;AudioVideo;Audio;
 NoDisplay=true
