--- a/kdebase/workspace/plasma/shells/desktop/panelview.cpp	2009/02/24 04:51:03	930724
+++ b/kdebase/workspace/plasma/shells/desktop/panelview.cpp	2009/03/03 23:39:55	934863
@@ -203,6 +203,11 @@
       m_triggerEntered(false)
 {
     Q_ASSERT(qobject_cast<Plasma::Corona*>(panel->scene()));
+
+    m_strutsTimer = new QTimer(this);
+    m_strutsTimer->setSingleShot(true);
+    connect(m_strutsTimer, SIGNAL(timeout()), this, SLOT(updateStruts()));
+
     if (panel) {
         connect(panel, SIGNAL(destroyed(QObject*)), this, SLOT(panelDeleted()));
         connect(panel, SIGNAL(toolBoxToggled()), this, SLOT(togglePanelController()));
@@ -248,10 +253,6 @@
     KWindowSystem::setOnAllDesktops(winId(), true);
 
     QTimer::singleShot(0, this, SLOT(init()));
-
-    m_strutsTimer = new QTimer(this);
-    m_strutsTimer->setSingleShot(true);
-    connect(m_strutsTimer,SIGNAL(timeout()),this,SLOT(updateStruts()));
 }
 
 PanelView::~PanelView()
