diff --git a/startkde/startkde.cmake b/startkde/startkde.cmake
index 14ce47a..58c3a7a 100644
--- a/startkde/startkde.cmake
+++ b/startkde/startkde.cmake
@@ -1,6 +1,6 @@
 #!/bin/sh
 #
-#  DEFAULT KDE STARTUP SCRIPT ( @KDE4WORKSPACE_VERSION@ )
+#  DEFAULT KDE STARTUP SCRIPT ( @PLASMAWORKSPACE_VERSION@ )
 #
 
 if test "x$1" = x--failsafe; then
@@ -17,17 +17,17 @@ trap 'echo GOT SIGHUP' HUP
 unset DYLD_FORCE_FLAT_NAMESPACE
 
 # in case we have been started with full pathname spec without being in PATH
-bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
-if [ -n "$bindir" ]; then
-  qbindir=`qtpaths --binaries-dir`
-  qdbus=$qbindir/qdbus
-  case $PATH in
-    $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
-    *) PATH=$bindir:$PATH; export PATH;;
-  esac
-else
+#bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
+#if [ -n "$bindir" ]; then
+#  qbindir=`qtpaths --binaries-dir`
+#  qdbus=$qbindir/qdbus
+#  case $PATH in
+#    $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
+#    *) PATH=$bindir:$PATH; export PATH;;
+#  esac
+#else
   qdbus=qdbus
-fi
+#fi
 
 # Check if a KDE session already is running and whether it's possible to connect to X
 kcheckrunning
@@ -162,7 +162,7 @@ unset DESKTOP_LOCKED # Don't want it in the environment
 ksplash_pid=
 if test -z "$dl"; then
   # languages as resolved by KLocale, for the splash screens use
-  # klocale_languages is assembled by kdostartupconfig4 calling KLocale
+  # klocale_languages is assembled by kdostartupconfig5 calling KLocale
   KLOCALE_LANGUAGES="$klocale_languages"
   export KLOCALE_LANGUAGES
   # the splashscreen and progress indicator
@@ -200,7 +200,7 @@ for prefix in `echo "$libpath" | sed -n -e 's,/lib[^/]*/,/env/,p'`; do
 done
 
 # Set the path for Qt plugins provided by KDE
-QT_PLUGIN_PATH=${QT_PLUGIN_PATH+$QT_PLUGIN_PATH:}`qtpaths --plugin-dir`
+QT_PLUGIN_PATH=${QT_PLUGIN_PATH+$QT_PLUGIN_PATH:}`kf5-config --qt-plugins`
 # TODO: Do we really need this?
 QT_PLUGIN_PATH=$QT_PLUGIN_PATH:$kdehome/lib/kde5/plugins/
 export QT_PLUGIN_PATH
