From e6e1db527e636649634db97846abd09862861134 Mon Sep 17 00:00:00 2001
From: Johannes Huber <johu@gentoo.org>
Date: Sun, 25 May 2014 17:29:16 +0200
Subject: [PATCH] [klipper] Re-enable prison

---
 klipper/CMakeLists.txt | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/klipper/CMakeLists.txt b/klipper/CMakeLists.txt
index c700f9e..7c66022 100644
--- a/klipper/CMakeLists.txt
+++ b/klipper/CMakeLists.txt
@@ -16,15 +16,15 @@ set(libklipper_common_SRCS
     clipcommandprocess.cpp
 )
 
-# find_package(Prison QUIET CONFIG)
-# set_package_properties(Prison PROPERTIES DESCRIPTION "Prison library"
-#                        URL "http://projects.kde.org/prison"
-#                        TYPE OPTIONAL
-#                        PURPOSE "Needed to create mobile barcodes from clipboard data"
-#                       )
-# if (PRISON_FOUND)
-#     include_directories(${PRISON_INCLUDE_DIR})
-# endif ()
+find_package(Prison QUIET CONFIG)
+set_package_properties(Prison PROPERTIES DESCRIPTION "Prison library"
+                       URL "http://projects.kde.org/prison"
+                       TYPE OPTIONAL
+                       PURPOSE "Needed to create mobile barcodes from clipboard data"
+                      )
+if (PRISON_FOUND)
+     include_directories(${PRISON_INCLUDE_DIR})
+endif ()
 
 set(HAVE_PRISON ${PRISON_FOUND})
 configure_file(config-klipper.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-klipper.h )
@@ -43,9 +43,9 @@ target_link_libraries(kdeinit_klipper Qt5::X11Extras KF5::GlobalAccel KF5::Windo
 if (X11_Xfixes_FOUND)
   target_link_libraries(kdeinit_klipper ${X11_Xfixes_LIB})
 endif ()
-# if (PRISON_FOUND)
-#   target_link_libraries(kdeinit_klipper ${PRISON_LIBRARIES})
-# endif ()
+if (PRISON_FOUND)
+  target_link_libraries(kdeinit_klipper ${PRISON_LIBRARIES})
+endif ()
 
 install(TARGETS kdeinit_klipper ${INSTALL_TARGETS_DEFAULT_ARGS})
 install(TARGETS klipper         ${INSTALL_TARGETS_DEFAULT_ARGS})
-- 
1.9.3

