--- solid-4.3.1/libs/solid/control/powermanager.cpp.orig	2009-09-24 19:11:31 +0200
+++ solid-4.3.1/libs/solid/control/powermanager.cpp	2009-09-24 19:11:46 +0200
@@ -25,7 +25,9 @@
 #include "ifaces/powermanager.h"
 #include <kdebug.h>
 #include <kglobal.h>
+#if !defined(Q_WS_MAC)
 #include <QX11Info>
+#endif
 
 K_GLOBAL_STATIC(Solid::Control::PowerManagerPrivate, globalPowerManager)
 
--- solid-4.3.1/libs/solid/control/powermanager.cpp.orig	2009-09-24 19:04:00 +0200
+++ solid-4.3.1/libs/solid/control/powermanager.cpp	2009-09-24 19:04:54 +0200
@@ -148,7 +148,9 @@ Solid::Control::PowerManager::Brightness
                       Solid::Control::PowerManager::BrightnessControlsList(), brightnessControlsAvailable());
 }
 
+#if !defined(Q_WS_MAC)
 extern float xrandr_brightlight(Display *dpy, long new_value = -1);
+#endif
 
 bool Solid::Control::PowerManager::setBrightness(float brightness, const QString &device)
 {
--- solid-4.3.1/libs/solid/control/powermanager.cpp.orig	2009-09-24 18:54:40 +0200
+++ solid-4.3.1/libs/solid/control/powermanager.cpp	2009-09-24 18:56:02 +0200
@@ -157,7 +157,7 @@ bool Solid::Control::PowerManager::setBr
         Solid::Control::PowerManager::BrightnessControlsList controls = brightnessControlsAvailable();
         if(controls.size() == 0)
         {
-#ifdef Q_WS_WIN
+#if defined(Q_WS_WIN) || defined(Q_WS_MAC)
             return false;
 #else
             return ( xrandr_brightlight( QX11Info::display(), brightness ) >= 0 );
@@ -187,7 +187,7 @@ float Solid::Control::PowerManager::brig
         Solid::Control::PowerManager::BrightnessControlsList controls = brightnessControlsAvailable();
         if(controls.size() == 0)
         {
-#ifdef Q_WS_WIN
+#if defined(Q_WS_WIN) || defined(Q_WS_MAC)
             return false;
 #else
             return xrandr_brightlight( QX11Info::display() );
--- solid-4.3.1/libs/solid/control/CMakeLists.txt.orig	2009-09-24 18:54:49 +0200
+++ solid-4.3.1/libs/solid/control/CMakeLists.txt	2009-09-24 18:55:10 +0200
@@ -28,9 +28,9 @@ set(solidcontrol_LIB_SRCS
 )
 
 # FIXME: this should be XRANDR_FOUND instead
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
 set(solidcontrol_LIB_SRCS ${solidcontrol_LIB_SRCS} powermanager_randr.cpp)
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
 
 qt4_add_dbus_interface(solidcontrol_LIB_SRCS org.kde.Solid.Networking.xml org_kde_solid_networking)
 
