--- solid-4.4.69.orig/libs/solid/control/powermanager.cpp
+++ solid-4.4.69/libs/solid/control/powermanager.cpp
@@ -25,7 +25,9 @@
 #include "ifaces/powermanager.h"
 #include <kdebug.h>
 #include <kglobal.h>
+#if !defined(Q_WS_MAC)
 #include <QX11Info>
+#endif
 
 K_GLOBAL_STATIC(Solid::Control::PowerManagerPrivate, globalPowerManager)
 
@@ -129,7 +131,9 @@ Solid::Control::PowerManager::Brightness
                       Solid::Control::PowerManager::BrightnessControlsList(), brightnessControlsAvailable());
 }
 
+#if !defined(Q_WS_MAC)
 extern float xrandr_brightlight(Display *dpy, long new_value = -1);
+#endif
 
 bool Solid::Control::PowerManager::setBrightness(float brightness, const QString &device)
 {
@@ -138,7 +142,7 @@ bool Solid::Control::PowerManager::setBr
         Solid::Control::PowerManager::BrightnessControlsList controls = brightnessControlsAvailable();
         if(controls.keys(Solid::Control::PowerManager::Screen).isEmpty())
         {
-#ifdef Q_WS_WIN
+#if defined(Q_WS_WIN) || defined(Q_WS_MAC)
             return false;
 #else
             return ( xrandr_brightlight( QX11Info::display(), brightness ) >= 0 );
@@ -168,7 +172,7 @@ float Solid::Control::PowerManager::brig
         Solid::Control::PowerManager::BrightnessControlsList controls = brightnessControlsAvailable();
         if(controls.keys(Solid::Control::PowerManager::Screen).isEmpty())
         {
-#ifdef Q_WS_WIN
+#if defined(Q_WS_WIN) || defined(Q_WS_MAC)
             return false;
 #else
             return xrandr_brightlight( QX11Info::display() );
--- solid-4.4.69.orig/libs/solid/control/CMakeLists.txt
+++ solid-4.4.69/libs/solid/control/CMakeLists.txt
@@ -28,9 +28,9 @@ set(solidcontrol_LIB_SRCS
 )
 
 # FIXME: this should be XRANDR_FOUND instead
-if(NOT WIN32)
+if(NOT WIN32 AND NOT APPLE)
 set(solidcontrol_LIB_SRCS ${solidcontrol_LIB_SRCS} powermanager_randr.cpp)
-endif(NOT WIN32)
+endif(NOT WIN32 AND NOT APPLE)
 
 qt4_add_dbus_interface(solidcontrol_LIB_SRCS org.kde.Solid.Networking.xml org_kde_solid_networking)
 
