diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5020a29..bf2c360 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@ ecm_setup_version(${KF5_VERSION} VARIABLE_PREFIX KWAYLAND
 set(REQUIRED_QT_VERSION 5.3.0)
 find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Gui)
 
-find_package(Wayland 1.2 COMPONENTS Client Egl Cursor Server)
+find_package(Wayland 1.2 COMPONENTS Client Cursor Server)
 set_package_properties(Wayland PROPERTIES
                        TYPE REQUIRED
                       )
