diff --git a/kpythonpluginfactory/CMakeLists.txt b/kpythonpluginfactory/CMakeLists.txt
index c24160e..9597cd1 100644
--- a/kpythonpluginfactory/CMakeLists.txt
+++ b/kpythonpluginfactory/CMakeLists.txt
@@ -4,7 +4,7 @@ set(kpythonpluginfactory_SRCS
     kpythonpluginfactory.cpp)
 
 GET_FILENAME_COMPONENT(LIB_PYTHON ${PYTHON_LIBRARIES} NAME)
-ADD_DEFINITIONS(-DLIB_PYTHON="${LIB_PYTHON}")
+ADD_DEFINITIONS(-DLIB_PYTHON="\\"${LIB_PYTHON}\\"")
 ADD_DEFINITIONS(-DKDE_DEFAULT_DEBUG_AREA=15000)
 
 kde4_add_plugin(kpythonpluginfactory ${kpythonpluginfactory_SRCS})
diff --git a/tools/pykdeuic4/CMakeLists.txt b/tools/pykdeuic4/CMakeLists.txt
index 092985d..2209fba 100644
--- a/tools/pykdeuic4/CMakeLists.txt
+++ b/tools/pykdeuic4/CMakeLists.txt
@@ -10,7 +10,7 @@ INSTALL(CODE "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -DTARGET=${PYTHON_SITE_PA
 
 if (NOT PYKDEUIC4_ALTINSTALL)
     # Step 2: Make a the generated symlink a link to "pykdeuic4"
-    INSTALL(CODE "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -DTARGET=${BIN_INSTALL_DIR}/${_uic_name} -DLINK_NAME=${BIN_INSTALL_DIR}/pykdeuic4 -P ${cmake_module_dir}/create_exe_symlink.cmake)" )
+    install(CODE "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E create_symlink ${BIN_INSTALL_DIR}/${_uic_name} ${BIN_INSTALL_DIR}/pykdeuic4)" )
 endif()
 
 unset(_uic_name)
