--- trunk/KDE/kdebindings/python/pykde4/sip/kdecore/ksycoca.sip	2008/07/04 19:03:58	828140
+++ trunk/KDE/kdebindings/python/pykde4/sip/kdecore/ksycoca.sip	2009/03/05 15:50:44	935553
@@ -58,7 +58,7 @@
     void                    disableAutoRebuild ();
     static bool             isChanged (const char* type);
     static void             flagError ();
-    static bool             readError ();
+//    static bool             readError ();
 
 signals:
     void                    databaseChanged ();
--- trunk/KDE/kdebindings/python/pykde4/sip/kdecore/kmimetype.sip	2008/07/04 19:03:58	828140
+++ trunk/KDE/kdebindings/python/pykde4/sip/kdecore/kmimetype.sip	2009/03/03 17:27:49	934730
@@ -69,6 +69,8 @@
                             KMimeType (const QString& fullpath, const QString& name, const QString& comment);
 //ig                        KMimeType (KMimeTypePrivate& dd);
 //ig                        KMimeType (KMimeTypePrivate& dd, const QString& name, const QString& comment);
+
+private:
     void                    setPatterns (const QStringList& patterns);
     void                    setParentMimeType (const QString& parent /Transfer/);
     void                    internalClearData ();
