diff -urN pykde4-4.2.71.old/python/pykde4/sip/kdecore/kdatetime.sip pykde4-4.2.71/python/pykde4/sip/kdecore/kdatetime.sip
--- pykde4-4.2.71.old/python/pykde4/sip/kdecore/kdatetime.sip	2009-04-30 22:02:22.000000000 +0400
+++ pykde4-4.2.71/python/pykde4/sip/kdecore/kdatetime.sip	2009-05-05 09:14:55.334924170 +0400
@@ -161,19 +161,3 @@
     static void             setSimulatedSystemTime (const KDateTime& newTime);
     static KDateTime        realCurrentLocalDateTime ();
 };   // KDateTime
-
-
-class KSystemTimeZones_Simulated : KSystemTimeZones /Abstract/
-{
-%TypeHeaderCode
-#include <kdatetime.h>
-%End
-
-public:
-    static KTimeZone        realLocalZone ();
-    static KTimeZone        local ();
-    static void             setLocalZone (const KTimeZone& tz);
-    static bool             isSimulated ();
-};   // KSystemTimeZones_Simulated
-
-
