diff -ruN pykde4-4.6.4.orig/python/pykde4/tools/pykdeuic4/pykdeuic4.py pykde4-4.6.4/python/pykde4/tools/pykdeuic4/pykdeuic4.py
--- pykde4-4.6.4.orig/python/pykde4/tools/pykdeuic4/pykdeuic4.py	2011-06-02 12:03:32.000000000 +0200
+++ pykde4-4.6.4/python/pykde4/tools/pykdeuic4/pykdeuic4.py	2011-06-21 21:59:16.310764723 +0200
@@ -78,7 +78,7 @@
     indenter.indentwidth = indent
     comp = compiler.UICompiler()
     comp.factory._cwFilters.append(kdeFilter())
-    winfo = comp.compileUi(uifile, output)
+    winfo = comp.compileUi(uifile, output, None)
 
     if exe:
         output.write("""
@@ -111,7 +111,7 @@
     app = kdeui.KApplication()
     mainWindow = MainWin(None, "main window")
     mainWindow.show()
-    app.connect (app, QtCore.SIGNAL ("lastWindowClosed ()"), app.quit)
+    app.lastWindowClosed.connect(app.quit)
     app.exec_ ()
 """)
 
