diff -urN pykde4-4.2.85.old/python/pykde4/sip/akonadi/agentfilterproxymodel.sip pykde4-4.2.85/python/pykde4/sip/akonadi/agentfilterproxymodel.sip
--- pykde4-4.2.85.old/python/pykde4/sip/akonadi/agentfilterproxymodel.sip	2009-05-08 19:31:58.000000000 +0400
+++ pykde4-4.2.85/python/pykde4/sip/akonadi/agentfilterproxymodel.sip	2009-05-09 16:35:49.604263584 +0400
@@ -173,8 +173,6 @@
                 sipClass = sipClass_Akonadi_ItemSync;
             else if (dynamic_cast<Akonadi::LinkJob*>(sipCpp))
                 sipClass = sipClass_Akonadi_LinkJob;
-            else if (dynamic_cast<Akonadi::ResourceSelectJob*>(sipCpp))
-                sipClass = sipClass_Akonadi_ResourceSelectJob;
             else if (dynamic_cast<Akonadi::SearchCreateJob*>(sipCpp))
                 sipClass = sipClass_Akonadi_SearchCreateJob;
             else if (dynamic_cast<Akonadi::TransactionBeginJob*>(sipCpp))
