--- pykde4-4.3.72/python/pykde4/sip/kdeui/kcategorydrawer.sip.def	2008-07-08 13:27:45.000000000 +0400
+++ pykde4-4.3.72/python/pykde4/sip/kdeui/kcategorydrawer.sip	2009-10-25 00:39:24.573756918 +0400
@@ -29,6 +29,28 @@
                             KCategoryDrawer ();
     virtual void            drawCategory (const QModelIndex& index, int sortRole, const QStyleOption& option, QPainter* painter) const;
     virtual int             categoryHeight (const QModelIndex& index, const QStyleOption& option) const;
+    int                     leftMargin () const;
+    void                    setLeftMargin (int leftMargin);
+    int                     rightMargin () const;
+    void                    setRightMargin (int rightMargin);
+
+private:
+                            KCategoryDrawer (const KCategoryDrawer&);
 };   // KCategoryDrawer
 
+class KCategoryDrawerV2 : QObject, KCategoryDrawer
+{
+%TypeHeaderCode
+#include <kcategorydrawer.h>
+%End
+
+public:
+                            KCategoryDrawerV2 (QObject *parent = 0);
+    virtual void            mouseButtonPressed (const QModelIndex &index, QMouseEvent *event);
+    virtual void            mouseButtonReleased (const QModelIndex &index, QMouseEvent *event);
+    virtual void            mouseButtonMoved(const QModelIndex &index, QMouseEvent *event);
+    virtual void            mouseButtonDoubleClicked(const QModelIndex &index, QMouseEvent *event);
 
+signals:
+    void                    collapseOrExpandClicked(const QModelIndex &index);
+};   // KCategoryDrawerV2
