--- configure.ac
+++ configure.ac
@@ -577,7 +577,6 @@
 src/settings/plugins/ifupdown/Makefile
 src/settings/plugins/ifupdown/tests/Makefile
 src/settings/plugins/ifnet/Makefile
-src/settings/plugins/ifnet/tests/Makefile
 src/settings/plugins/ifcfg-rh/Makefile
 src/settings/plugins/ifcfg-rh/tests/Makefile
 src/settings/plugins/ifcfg-rh/tests/network-scripts/Makefile
--- src/settings/plugins/ifnet/Makefile.am
+++ src/settings/plugins/ifnet/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = . tests
+SUBDIRS = .
 INCLUDES = \
        -I$(top_srcdir)/src/settings \
        -I$(top_srcdir)/include \
