--- analiyzagui/graph2d.cpp
+++ analitzagui/graph2d.cpp
@@ -33,6 +33,18 @@
 
 #include <analitza/analyzer.h>
 #include "functionsmodel.h"
+
+// Stolen from kcalc.
+#if defined(Q_OS_SOLARIS)
+// Strictly by the standard, ininf() is a c99-ism which
+// is unavailable in C++. The IEEE FP headers provide
+// a function with similar functionality, so use that instead.
+// However, !finite(a) == isinf(a) || isnan(a), so it's
+// not 100% correct.
+#include <ieeefp.h>
+#define isinf(a) !finite(a)
+#endif
+
 #include "functionutils.h"
 #include "functionspainter.h"
 #include <cmath>
