From 683274cace247ba9ca5b972f37ffd4d40d8892fa Mon Sep 17 00:00:00 2001
From: Johannes Huber <johu@gmx.de>
Date: Thu, 24 Nov 2011 15:13:38 +0100
Subject: [PATCH 2/2] Remove internal quazip hack

---
 CMakeLists.txt |   12 ------------
 1 files changed, 0 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 070db41..0e15c61 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,18 +110,6 @@ macro_log_feature(LIBATTICA_FOUND "libattica" "Provides support for automatic fe
 macro_optional_find_package(QuaZip)
 macro_log_feature(QuaZip_FOUND "QuaZip" "Provides support for extracting downloaded resolvers automatically." "http://quazip.sourceforge.net/" FALSE "" "")
 
-IF( NOT QuaZip_FOUND )
-    add_subdirectory( ${CMAKE_SOURCE_DIR}/src/libtomahawk/thirdparty/quazip )
-    SET( QuaZip_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/libtomahawk/thirdparty/quazip )
-    SET( QuaZip_LIBRARY quazip )
-    SET( QuaZip_LIBRARIES ${QuaZip_LIBRARY} )
-    SET( QuaZip_FOUND true )
-    macro_log_feature(QuaZip_FOUND "QuaZip" "Provides support for extracting downloaded resolvers automatically. Building internal copy" "http://quazip.sourceforge.net/" FALSE "" "")
-
-    # copy headers to build/quazip so we can use proper includes inside the code
-    FILE( COPY ${CMAKE_SOURCE_DIR}/src/libtomahawk/thirdparty/quazip/quazip/ DESTINATION ${CMAKE_BINARY_DIR}/libtomahawk/thirdparty/quazip )
-ENDIF()
-
 # required
 #While we distribute our own liblastfm2, don't need to look for it
 #macro_optional_find_package(LibLastFm 0.3.3)
-- 
1.7.8.rc3

