diff -upr tomahawk.orig//CMakeLists.txt tomahawk//CMakeLists.txt
--- tomahawk.orig//CMakeLists.txt	2011-11-10 14:30:31.000000000 +0100
+++ tomahawk//CMakeLists.txt	2011-11-10 14:32:49.000000000 +0100
@@ -110,17 +110,6 @@ macro_log_feature(LIBATTICA_FOUND "libat
 macro_optional_find_package(QuaZip)
 macro_log_feature(QuaZip_FOUND "QuaZip" "Provides support for extracting downloaded resolvers autmatically. Will build internal copy instead." "http://quazip.sourceforge.net/" FALSE "" "")
 
-IF( NOT QuaZip_FOUND )
-    add_subdirectory( ${CMAKE_SOURCE_DIR}/src/libtomahawk/thirdparty/quazip )
-    SET( QuaZip_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/libtomahawk/thirdparty/quazip )
-    SET( QuaZip_LIBRARY quazip )
-    SET( QuaZip_LIBRARIES ${QuaZip_LIBRARY} )
-    SET( QuaZip_FOUND true )
-
-    # copy headers to build/quazip so we can use proper includes inside the code
-    FILE( COPY ${CMAKE_SOURCE_DIR}/src/libtomahawk/thirdparty/quazip/quazip/ DESTINATION ${CMAKE_BINARY_DIR}/libtomahawk/thirdparty/quazip )
-ENDIF()
-
 # required
 #While we distribute our own liblastfm2, don't need to look for it
 #macro_optional_find_package(LibLastFm 0.3.3)
