diff --git a/cppchecks/cplusplus/cppmodel/CMakeLists.txt b/cppchecks/cplusplus/cppmodel/CMakeLists.txt
index ecdcdd8..a075415 100644
--- a/cppchecks/cplusplus/cppmodel/CMakeLists.txt
+++ b/cppchecks/cplusplus/cppmodel/CMakeLists.txt
@@ -20,7 +20,5 @@ add_library(cppmodel SHARED ${cppmodel_SRCS})
 target_link_libraries(cppmodel preprocessor cpp_parser ${QT_QTCORE_LIBRARY})
 set_target_properties(cppmodel PROPERTIES DEFINE_SYMBOL CPLUSPLUSMODEL_BUILD_LIB)
 
-if(WIN32)
 install(TARGETS cppmodel ${INSTALL_TARGETS_DEFAULT_ARGS})
-endif(WIN32)
 
diff --git a/cppchecks/cplusplus/parser/CMakeLists.txt b/cppchecks/cplusplus/parser/CMakeLists.txt
index 08382f4..a57e037 100644
--- a/cppchecks/cplusplus/parser/CMakeLists.txt
+++ b/cppchecks/cplusplus/parser/CMakeLists.txt
@@ -45,6 +45,4 @@ include_directories(${QT_INCLUDES})
 add_library(cpp_parser SHARED ${cplusplus_SRCS})
 set_target_properties(cpp_parser PROPERTIES DEFINE_SYMBOL CPLUSPLUS_BUILD_LIB)
 
-if(WIN32)
 install(TARGETS cpp_parser ${INSTALL_TARGETS_DEFAULT_ARGS})
-endif(WIN32)
diff --git a/cppchecks/cplusplus/preprocessor/CMakeLists.txt b/cppchecks/cplusplus/preprocessor/CMakeLists.txt
index 6251d44..44ed8df 100644
--- a/cppchecks/cplusplus/preprocessor/CMakeLists.txt
+++ b/cppchecks/cplusplus/preprocessor/CMakeLists.txt
@@ -17,7 +17,5 @@ add_library(preprocessor SHARED ${preproc_SRCS})
 target_link_libraries(preprocessor cpp_parser ${QT_QTCORE_LIBRARY})
 set_target_properties(preprocessor PROPERTIES DEFINE_SYMBOL CPLUSPLUSPREPROCESSOR_BUILD_LIB)
 
-if(WIN32)
 install(TARGETS preprocessor ${INSTALL_TARGETS_DEFAULT_ARGS})
-endif(WIN32)
 
