diff -Naur PyQt-x11-gpl-4.4.4~orig/configure.py PyQt-x11-gpl-4.4.4/configure.py
--- PyQt-x11-gpl-4.4.4~orig/configure.py	2009-03-26 01:01:43.000000000 +0100
+++ PyQt-x11-gpl-4.4.4/configure.py	2009-03-26 01:09:50.000000000 +0100
@@ -666,6 +666,7 @@
                 install_dir=pydbusmoddir,
                 qt=["QtCore"],
                 debug=opts.debug,
+                strip=0,
                 universal=sipcfg.universal
             )
 
@@ -1350,6 +1351,7 @@
         warnings=1,
         static=opts.static,
         debug=opts.debug,
+        strip=0,
         universal=sipcfg.universal
     )
 
@@ -1510,7 +1512,7 @@
         # The Qt binary installer has macx-xcode as the default.
         args = "-spec macx-g++ " + args
 
-    return args
+    return " QMAKE_STRIP= " + args
 
 
 def get_qt_configuration():
