--- PyQt-x11-gpl-4.7.2/configure.py.orig	2010-03-18 00:59:51.000000000 +0100
+++ PyQt-x11-gpl-4.7.2/configure.py	2010-03-18 01:08:46.000000000 +0100
@@ -478,7 +478,7 @@
             generate_code("QtXmlPatterns")
 
         if "phonon" in pyqt_modules:
-            generate_code("phonon")
+            generate_code("phonon", extra_include_dirs=["/usr/include/phonon"])
 
         if "QtAssistant" in pyqt_modules:
             generate_code("QtAssistant")
@@ -797,6 +797,7 @@
                 install_dir=pydbusmoddir,
                 qt=["QtCore"],
                 debug=opts.debug,
+                strip=0,
                 universal=sipcfg.universal,
                 arch=sipcfg.arch
             )
@@ -1611,6 +1612,7 @@
         warnings=1,
         static=opts.static,
         debug=opts.debug,
+        strip=0,
         universal=sipcfg.universal,
         arch=sipcfg.arch,
         prot_is_public=opts.prot_is_public
