diff --git a/src/global.h b/src/global.h
--- a/src/global.h
+++ b/src/global.h
@@ -22,6 +22,7 @@
 
 #include <QList>
 #include <QUrl>
+#include <QWidget>
 
 namespace Dolphin {
     QList<QUrl> validateUris(const QStringList& uriList);
diff --git a/src/kitemviews/kstandarditem.h b/src/kitemviews/kstandarditem.h
--- a/src/kitemviews/kstandarditem.h
+++ b/src/kitemviews/kstandarditem.h
@@ -25,6 +25,7 @@
 #include <QByteArray>
 #include <QHash>
 #include <QObject>
+#include <QSet>
 #include <QVariant>
 
 class KStandardItemModel;
diff --git a/src/search/dolphinsearchbox.cpp b/src/search/dolphinsearchbox.cpp
--- a/src/search/dolphinsearchbox.cpp
+++ b/src/search/dolphinsearchbox.cpp
@@ -39,9 +39,11 @@
 #include <QFontDatabase>
 #include <QHBoxLayout>
 #include <QIcon>
+#include <QKeyEvent>
 #include <QLabel>
 #include <QLineEdit>
 #include <QScrollArea>
+#include <QShowEvent>
 #include <QTimer>
 #include <QToolButton>
 #include <QUrlQuery>
diff --git a/src/settings/dolphinsettingsdialog.cpp b/src/settings/dolphinsettingsdialog.cpp
--- a/src/settings/dolphinsettingsdialog.cpp
+++ b/src/settings/dolphinsettingsdialog.cpp
@@ -34,6 +34,7 @@
 #include <KWindowConfig>
 #include <KMessageBox>
 
+#include <QCloseEvent>
 #include <QPushButton>
 
 DolphinSettingsDialog::DolphinSettingsDialog(const QUrl& url, QWidget* parent) :
diff --git a/src/settings/general/previewssettingspage.cpp b/src/settings/general/previewssettingspage.cpp
--- a/src/settings/general/previewssettingspage.cpp
+++ b/src/settings/general/previewssettingspage.cpp
@@ -32,6 +32,7 @@
 #include <QLabel>
 #include <QListView>
 #include <QPainter>
+#include <QShowEvent>
 #include <QSortFilterProxyModel>
 #include <QSpinBox>
 
diff --git a/src/settings/services/servicessettingspage.cpp b/src/settings/services/servicessettingspage.cpp
--- a/src/settings/services/servicessettingspage.cpp
+++ b/src/settings/services/servicessettingspage.cpp
@@ -36,6 +36,7 @@
 #include <QGridLayout>
 #include <QLabel>
 #include <QListWidget>
+#include <QShowEvent>
 #include <QSortFilterProxyModel>
 
 namespace
diff --git a/src/views/draganddrophelper.cpp b/src/views/draganddrophelper.cpp
--- a/src/views/draganddrophelper.cpp
+++ b/src/views/draganddrophelper.cpp
@@ -25,6 +25,7 @@
 
 #include <QDBusConnection>
 #include <QDBusMessage>
+#include <QDropEvent>
 #include <QMimeData>
 
 QHash<QUrl, bool> DragAndDropHelper::m_urlListMatchesUrlCache;
