Index: trunk/KDE/kdebase/workspace/libs/kworkspace/kdisplaymanager.cpp
===================================================================
--- trunk/KDE/kdebase/workspace/libs/kworkspace/kdisplaymanager.cpp	(revision 911248)
+++ trunk/KDE/kdebase/workspace/libs/kworkspace/kdisplaymanager.cpp	(revision 911249)
@@ -201,7 +201,7 @@
               KWorkSpace::ShutdownMode shutdownMode, /* NOT Default */
               const QString &bootOption )
 {
-	if (shutdownType == KWorkSpace::ShutdownTypeNone)
+	if (shutdownType == KWorkSpace::ShutdownTypeNone || shutdownType == KWorkSpace::ShutdownTypeLogout)
 		return;
 
 	bool cap_ask;
