diff --git a/cmake/modules/FindPAM.cmake b/cmake/modules/FindPAM.cmake
index 3499836..92527b5 100644
--- a/cmake/modules/FindPAM.cmake
+++ b/cmake/modules/FindPAM.cmake
@@ -5,6 +5,8 @@
 #  PAM_INCLUDE_DIR - the pam include directory
 #  PAM_LIBRARIES - libpam library
 
+if (WITH_PAM)
+
 if (PAM_INCLUDE_DIR AND PAM_LIBRARY)
 	# Already in cache, be silent
 	set(PAM_FIND_QUIETLY TRUE)
@@ -72,3 +74,5 @@ else (PAM_FOUND)
 endif (PAM_FOUND)
 
 mark_as_advanced(PAM_INCLUDE_DIR PAM_LIBRARY DL_LIBRARY PAM_MESSAGE_CONST)
+
+endif (WITH_PAM)
