Gentoo-bug: 530012

diff --git a/src/server/storage/dbconfigmysql.cpp b/src/server/storage/dbconfigmysql.cpp
index a18144d..68b0ecd 100644
--- a/src/server/storage/dbconfigmysql.cpp
+++ b/src/server/storage/dbconfigmysql.cpp
@@ -77,6 +77,7 @@ bool DbConfigMysql::init(QSettings &settings)
                                          << QLatin1String("/usr/sbin")
                                          << QLatin1String("/usr/local/sbin")
                                          << QLatin1String("/usr/local/libexec")
+                                         << QLatin1String("/usr/share/mysql/scripts")
                                          << QLatin1String("/usr/libexec")
                                          << QLatin1String("/opt/mysql/libexec")
                                          << QLatin1String("/opt/local/lib/mysql5/bin")
@@ -280,7 +281,7 @@ void DbConfigMysql::startInternalServer()
     // first run, some MySQL versions need a mysql_install_db run for that
     const QString confFile = XdgBaseDirs::findResourceFile("config", QLatin1String("akonadi/mysql-global.conf"));
     if (QDir(dataDir).entryList(QDir::NoDotAndDotDot | QDir::AllEntries).isEmpty() && !mMysqlInstallDbPath.isEmpty()) {
-        const QStringList arguments = QStringList() << QString::fromLatin1("--force") << QString::fromLatin1("--defaults-file=%1").arg(confFile) << QString::fromLatin1("--datadir=%1/").arg(dataDir);
+        const QStringList arguments = QStringList() << QString::fromLatin1("--force") << QString::fromLatin1("--defaults-file=%1").arg(confFile) << QString::fromLatin1("--datadir=%1/").arg(dataDir) << QString::fromLatin1( "--basedir=/usr" ) ;
         QProcess::execute(mMysqlInstallDbPath, arguments);
     }
 
