commit 75f87727dbcad2e6b6f1e97c8a6784fcf8616da9
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Thu Jan 21 00:10:16 2016 +0100

    Move LibXML2 detection into xml subdir, make optional

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 503ba48..b05eb4d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,13 +66,6 @@ set_package_properties(Boost PROPERTIES
     TYPE REQUIRED
 )
 
-find_package(LibXml2)
-set_package_properties(LibXml2 PROPERTIES
-    DESCRIPTION "The LibXML2 library"
-    URL "http://xmlsoft.org"
-    TYPE REQUIRED
-)
-
 set(AKONADI_TESTS_EXPORT AKONADICORE_EXPORT)
 configure_file(akonaditests_export.h.in "${CMAKE_CURRENT_BINARY_DIR}/akonaditests_export.h")
 
diff --git a/src/xml/CMakeLists.txt b/src/xml/CMakeLists.txt
index ad116d8..71e6ef4 100644
--- a/src/xml/CMakeLists.txt
+++ b/src/xml/CMakeLists.txt
@@ -1,4 +1,11 @@
 
+find_package(LibXml2)
+set_package_properties(LibXml2 PROPERTIES
+    DESCRIPTION "Required for XML schema validation in akonadixml"
+    URL "http://xmlsoft.org"
+    TYPE OPTIONAL
+)
+
 # Turn exceptions on
 if (LIBXML2_FOUND)
     add_definitions(-DHAVE_LIBXML2)
