Desc:	Fix missing isinf() for Solaris.
Author:	Heiko Przybyl <zuxez@cs.tu-berlin.de>
--- kcalc-4.3.2/kcalc/knumber/knumber_priv.cpp.orig	2009-10-16 13:14:46.107208004 +0200
+++ kcalc-4.3.2/kcalc/knumber/knumber_priv.cpp	2009-10-16 13:14:56.639773377 +0200
@@ -25,7 +25,7 @@
 
 #include <QRegExp>
 
-#if defined(Q_OS_SOLARIS) && defined(__SUNPRO_CC)
+#if defined(Q_OS_SOLARIS)
 // Strictly by the standard, ininf() is a c99-ism which
 // is unavailable in C++. The IEEE FP headers provide
 // a function with similar functionality, so use that instead.
