--- a/client/client.pro	2011-07-28 21:18:24.330394388 +0200
+++ b/client/client.pro	2011-07-28 21:06:19.223851824 +0200
@@ -91,7 +91,7 @@
 LIBS += -L../bin/plugins_client -lmapnikrendererclient -lcontractionhierarchiesclient -lgpsgridclient -losmrendererclient -lunicodetournamenttrieclient -lqtilerendererclient
 
 #required by osmrendererclient
-QT += network
+QT += network mobility
 CONFIG += mobility
 MOBILITY += location
 
