diff --git a/ksmserver/screenlocker/ksldapp.cpp b/ksmserver/screenlocker/ksldapp.cpp
index 22755ea..9e43c36 100644
--- a/ksmserver/screenlocker/ksldapp.cpp
+++ b/ksmserver/screenlocker/ksldapp.cpp
@@ -271,12 +271,12 @@ void KSldApp::lockProcessFinished(int exitCode, QProcess::ExitStatus exitStatus)
 
 bool KSldApp::startLockProcess()
 {
-    if (m_plasmaEnabled) {
-        m_lockProcess->setProgram(KStandardDirs::findExe(QLatin1String("plasma-overlay")));
-        *m_lockProcess << QLatin1String("--nofork");
-    } else {
+//    if (m_plasmaEnabled) {
+//        m_lockProcess->setProgram(KStandardDirs::findExe(QLatin1String("plasma-overlay")));
+//        *m_lockProcess << QLatin1String("--nofork");
+//    } else {
         m_lockProcess->setProgram(KStandardDirs::findExe(QLatin1String("kscreenlocker_greet")));
-    }
+//    }
     m_lockProcess->start();
     // we wait one minute
     if (!m_lockProcess->waitForStarted()) {
