diff --git a/ksmserver/screenlocker/ksldapp.cpp b/ksmserver/screenlocker/ksldapp.cpp
index 6490145..2403b45 100644
--- a/ksmserver/screenlocker/ksldapp.cpp
+++ b/ksmserver/screenlocker/ksldapp.cpp
@@ -286,16 +286,11 @@ void KSldApp::lockProcessReady()
 
 bool KSldApp::startLockProcess(bool immediateLock)
 {
-    if (m_plasmaEnabled) {
-        m_lockProcess->start(KStandardDirs::findExe(QLatin1String("plasma-overlay")),
-                             QStringList() << QLatin1String("--nofork"));
-    } else {
         QStringList args;
         if (immediateLock) {
             args << "--immediateLock";
         }
         m_lockProcess->start(KStandardDirs::findExe(QLatin1String("kscreenlocker_greet")), args);
-    }
     // we wait one minute
     if (!m_lockProcess->waitForStarted(60000)) {
         m_lockProcess->kill();
