diff --git a/ksmserver/screenlocker/ksldapp.cpp b/ksmserver/screenlocker/ksldapp.cpp
index f0526cf..feb3c8e 100644
--- a/ksmserver/screenlocker/ksldapp.cpp
+++ b/ksmserver/screenlocker/ksldapp.cpp
@@ -278,12 +278,12 @@ void KSldApp::lockProcessReady()
 
 bool KSldApp::startLockProcess()
 {
-    if (m_plasmaEnabled) {
-        m_lockProcess->start(KStandardDirs::findExe(QLatin1String("plasma-overlay")),
-                             QStringList() << QLatin1String("--nofork"));
-    } else {
+//    if (m_plasmaEnabled) {
+//        m_lockProcess->start(KStandardDirs::findExe(QLatin1String("plasma-overlay")),
+//                             QStringList() << QLatin1String("--nofork"));
+//    } else {
         m_lockProcess->start(KStandardDirs::findExe(QLatin1String("kscreenlocker_greet")));
-    }
+//    }
     // we wait one minute
     if (!m_lockProcess->waitForStarted(60000)) {
         m_lockProcess->kill();
