diff -ruN kdeedu-4.6.4.orig/libkdeedu/CMakeLists.txt kdeedu-4.6.4/libkdeedu/CMakeLists.txt
--- kdeedu-4.6.4.orig/libkdeedu/CMakeLists.txt	2011-06-03 23:41:38.000000000 +0200
+++ kdeedu-4.6.4/libkdeedu/CMakeLists.txt	2011-06-08 23:55:50.000000000 +0200
@@ -23,10 +23,10 @@
     set(_LibKdeEdu_CONFIG_DEST "${LIB_INSTALL_DIR}/libkdeedu")
 endif(KDE4_USE_COMMON_CMAKE_PACKAGE_CONFIG_DIR)
 
-install( FILES
-        "${LibKdeEdu_BINARY_DIR}/LibKdeEduConfig.cmake"
-        DESTINATION "${_LibKdeEdu_CONFIG_DEST}" )
-install( EXPORT LibKdeEduTargets
-         DESTINATION "${_LibKdeEdu_CONFIG_DEST}"
-         NAMESPACE LibKdeEduImport__
-         FILE LibKdeEduTargets.cmake )
+#install( FILES
+#        "${LibKdeEdu_BINARY_DIR}/LibKdeEduConfig.cmake"
+#        DESTINATION "${_LibKdeEdu_CONFIG_DEST}" )
+#install( EXPORT LibKdeEduTargets
+#         DESTINATION "${_LibKdeEdu_CONFIG_DEST}"
+#         NAMESPACE LibKdeEduImport__
+#         FILE LibKdeEduTargets.cmake )
