Index: kalgebra/src/CMakeLists.txt
===================================================================
--- kalgebra/src/CMakeLists.txt	(revision 767350)
+++ kalgebra/src/CMakeLists.txt	(working copy)
@@ -24,7 +24,6 @@
 	functionedit.cpp
 	varedit.cpp
 	main.cpp
-	graph3d.cpp
 	expressionedit.cpp
 	graph2d.cpp
 	operatorsmodel.cpp
@@ -33,6 +32,10 @@
 	functionimpl.cpp
 )
 
+if(OPENGL_FOUND)
+   set(kalgebra_SRCS ${kalgebra_SRCS} graph3d.cpp)
+endif(OPENGL_FOUND)
+
 #uncomment this if oxygen icons for kalgebra are available
 #kde4_add_app_icon(kalgebra_SRCS "${CMAKE_INSTALL_PREFIX}/share/icons/oxygen/*/apps/kalgebra.png")
 
