commit 7161f629b39ca84972510eb8333a78c5a5efcec9
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Sun Jan 17 13:34:17 2016 +0100

    [libakonadi] Make tools optional

diff --git a/akonadi/CMakeLists.txt b/akonadi/CMakeLists.txt
index c80db92..e36c0ab 100644
--- a/akonadi/CMakeLists.txt
+++ b/akonadi/CMakeLists.txt
@@ -43,6 +43,8 @@ else()
     set(KF5Akonadi_DATA_DIR "${CMAKE_INSTALL_PREFIX}/${KDE_INSTALL_DATADIR_KF5}/akonadi")
 endif()
 
+option(BUILD_TOOLS "Build and install tools for testing purposes." TRUE)
+
 ########### Find packages ###########
 find_package(KF5ItemViews ${KF5_VERSION} CONFIG REQUIRED)
 find_package(KF5KIO ${KF5_VERSION} CONFIG REQUIRED)
@@ -93,12 +95,16 @@ add_definitions("-DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII")
 configure_file(akonadiprivate_export.h.in "${CMAKE_CURRENT_BINARY_DIR}/akonadiprivate_export.h")
 
 add_subdirectory(src)
-add_subdirectory(autotests/testrunner)
-add_subdirectory(autotests/testresource)
+# add testrunner (application for managing a self-contained test environment)
+if(BUILD_TOOLS)
+    add_subdirectory(autotests/testrunner)
+    add_subdirectory(autotests/testresource)
+    add_subdirectory(autotests/testsearchplugin)
+endif()
 
-if(BUILD_TESTING)
-   add_subdirectory(autotests)
-   add_subdirectory(tests)
+if(BUILD_TOOLS AND BUILD_TESTING)
+    add_subdirectory(autotests)
+    add_subdirectory(tests)
 endif()
 ########### CMake Config Files ###########
 
diff --git a/akonadi/autotests/CMakeLists.txt b/akonadi/autotests/CMakeLists.txt
index 308cf22..dc69a26 100644
--- a/akonadi/autotests/CMakeLists.txt
+++ b/akonadi/autotests/CMakeLists.txt
@@ -25,11 +25,6 @@ include_directories(
   ${Boost_INCLUDE_DIR}
 )
 
-# add testrunner (application for managing a self-contained test
-# environment)
-
-add_subdirectory(testsearchplugin)
-
 # convenience macro to add akonadi qtestlib unit-tests
 macro(add_akonadi_test _source)
   set(_test ${_source})
