http://bugs.gentoo.org/show_bug.cgi?id=465898

--- Modules/FindImageMagick.cmake
+++ Modules/FindImageMagick.cmake
@@ -81,7 +81,7 @@
       ${ImageMagick_INCLUDE_DIRS}
       "[HKEY_LOCAL_MACHINE\\SOFTWARE\\ImageMagick\\Current;BinPath]/include"
     PATH_SUFFIXES
-      ImageMagick
+      ImageMagick ImageMagick-6
     DOC "Path to the ImageMagick include dir."
     )
   find_library(ImageMagick_${component}_LIBRARY
@@ -147,17 +147,17 @@
     )
   if(component STREQUAL "Magick++")
     FIND_IMAGEMAGICK_API(Magick++ Magick++.h
-      Magick++ CORE_RL_Magick++_ Magick++-6.Q16 Magick++-Q16 Magick++-6.Q8 Magick++-Q8
+      Magick++ CORE_RL_Magick++_ Magick++.Q8 Magick++-Q8 Magick++.Q8HDRI Magick++-Q8HDRI Magick++.Q16 Magick++-Q16 Magick++.Q16HDRI Magick++-Q16HDRI Magick++.Q32 Magick++-Q32 Magick++.Q32HDRI Magick++-Q32HDRI Magick++.Q64 Magick++-Q64 Magick++.Q64HDRI Magick++-Q64HDRI Magick++-6.Q8 Magick++-6-Q8 Magick++-6.Q8HDRI Magick++-6-Q8HDRI Magick++-6.Q16 Magick++-6-Q16 Magick++-6.Q16HDRI Magick++-6-Q16HDRI Magick++-6.Q32 Magick++-6-Q32 Magick++-6.Q32HDRI Magick++-6-Q32HDRI Magick++-6.Q64 Magick++-6-Q64 Magick++-6.Q64HDRI Magick++-6-Q64HDRI
       )
     list(APPEND ImageMagick_REQUIRED_VARS ImageMagick_Magick++_LIBRARY)
   elseif(component STREQUAL "MagickWand")
     FIND_IMAGEMAGICK_API(MagickWand wand/MagickWand.h
-      Wand MagickWand CORE_RL_wand_ MagickWand-6.Q16 MagickWand-Q16 MagickWand-6.Q8 MagickWand-Q8
+      Wand MagickWand CORE_RL_wand_ MagickWand.Q8 MagickWand-Q8 MagickWand.Q8HDRI MagickWand-Q8HDRI MagickWand.Q16 MagickWand-Q16 MagickWand.Q16HDRI MagickWand-Q16HDRI MagickWand.Q32 MagickWand-Q32 MagickWand.Q32HDRI MagickWand-Q32HDRI MagickWand.Q64 MagickWand-Q64 MagickWand.Q64HDRI MagickWand-Q64HDRI MagickWand-6.Q8 MagickWand-6-Q8 MagickWand-6.Q8HDRI MagickWand-6-Q8HDRI MagickWand-6.Q16 MagickWand-6-Q16 MagickWand-6.Q16HDRI MagickWand-6-Q16HDRI MagickWand-6.Q32 MagickWand-6-Q32 MagickWand-6.Q32HDRI MagickWand-6-Q32HDRI MagickWand-6.Q64 MagickWand-6-Q64 MagickWand-6.Q64HDRI MagickWand-6-Q64HDRI
       )
     list(APPEND ImageMagick_REQUIRED_VARS ImageMagick_MagickWand_LIBRARY)
   elseif(component STREQUAL "MagickCore")
     FIND_IMAGEMAGICK_API(MagickCore magick/MagickCore.h
-      Magick MagickCore CORE_RL_magick_ MagickCore-6.Q16 MagickCore-Q16 MagickCore-6.Q8 MagickCore-Q8
+      Magick MagickCore CORE_RL_magick_ MagickCore.Q8 MagickCore-Q8 MagickCore.Q8HDRI MagickCore-Q8HDRI MagickCore.Q16 MagickCore-Q16 MagickCore.Q16HDRI MagickCore-Q16HDRI MagickCore.Q32 MagickCore-Q32 MagickCore.Q32HDRI MagickCore-Q32HDRI MagickCore.Q64 MagickCore-Q64 MagickCore.Q64HDRI MagickCore-Q64HDRI MagickCore-6.Q8 MagickCore-6-Q8 MagickCore-6.Q8HDRI MagickCore-6-Q8HDRI MagickCore-6.Q16 MagickCore-6-Q16 MagickCore-6.Q16HDRI MagickCore-6-Q16HDRI MagickCore-6.Q32 MagickCore-6-Q32 MagickCore-6.Q32HDRI MagickCore-6-Q32HDRI MagickCore-6.Q64 MagickCore-6-Q64 MagickCore-6.Q64HDRI MagickCore-6-Q64HDRI
       )
     list(APPEND ImageMagick_REQUIRED_VARS ImageMagick_MagickCore_LIBRARY)
   else()
