--- Source/QtDoalog/CMakeLists.txt
+++ Source/QtDialog/CMakeLists.txt
@@ -11,7 +11,12 @@
 #=============================================================================
 
 project(QtDialog)
-find_package(Qt5Widgets QUIET)
+option(WITH_Qt5 ON)
+
+if(WITH_Qt5)
+  find_package(Qt5Widgets QUIET)
+endif()
+
 if (Qt5Widgets_FOUND)
   include_directories(${Qt5Widgets_INCLUDE_DIRS})
   add_definitions(${Qt5Widgets_DEFINITONS})
