diff -ruN cmake-2.6.0.orig/Modules/Platform/Interix.cmake cmake-2.6.0/Modules/Platform/Interix.cmake
--- cmake-2.6.0.orig/Modules/Platform/Interix.cmake	1970-01-01 01:00:00 +0100
+++ cmake-2.6.0/Modules/Platform/Interix.cmake	2008-06-02 14:22:45 +0200
@@ -0,0 +1,31 @@
+# This is a proposed Platform definition for Interix GCC/G++
+
+SET(CMAKE_DL_LIBS "dl")
+SET(CMAKE_SHARED_LIBRARY_C_FLAGS "")
+SET(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "-shared")
+SET(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "")
+SET(CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG "-Wl,-rpath,")
+SET(CMAKE_SHARED_LIBRARY_RUNTIME_C_FLAG_SEP ":")
+SET(CMAKE_SHARED_LIBRARY_SONAME_C_FLAG "-Wl,-h,")
+SET(CMAKE_SHARED_LIBRARY_SONAME_CXX_FLAG "-Wl,-h,")
+
+SET(UNIX 1)
+
+# also add the install directory of the running cmake to the search directories
+# CMAKE_ROOT is CMAKE_INSTALL_PREFIX/share/cmake, so we need to go two levels up
+GET_FILENAME_COMPONENT(_CMAKE_INSTALL_DIR "${CMAKE_ROOT}" PATH)
+GET_FILENAME_COMPONENT(_CMAKE_INSTALL_DIR "${_CMAKE_INSTALL_DIR}" PATH)
+
+LIST(APPEND CMAKE_SYSTEM_PREFIX_PATH
+	"${_CMAKE_INSTALL_DIR}"
+	"${CMAKE_INSTALL_PREFIX}"
+	)
+
+LIST(APPEND CMAKE_SYSTEM_INCLUDE_PATH
+	"${CMAKE_INSTALL_PREFIX}"/include
+	)
+
+LIST(APPEND CMAKE_SYSTEM_LIBRARY_PATH
+	"${CMAKE_INSTALL_PREFIX}"/lib
+	)
+
diff -ruN cmake-2.6.0.orig/Utilities/cmtar/encode.c cmake-2.6.0/Utilities/cmtar/encode.c
--- cmake-2.6.0.orig/Utilities/cmtar/encode.c	2008-06-02 13:00:44 +0200
+++ cmake-2.6.0/Utilities/cmtar/encode.c	2008-06-02 14:02:39 +0200
@@ -24,6 +24,9 @@
 # include <stdlib.h>
 #endif
 
+#ifdef HAVE_SYS_MKDEV_H
+# include <sys/mkdev.h>
+#endif
 
 /* magic, version, and checksum */
 void
