--- onelinecalc-0.2.original/CMakeLists.txt
+++ onelinecalc-0.2/CMakeLists.txt
@@ -4,8 +4,12 @@
 # Find the required Libaries
 find_package(KDE4 REQUIRED)
 include(KDE4Defaults)
-find_package(Plasma REQUIRED)
  
+if ( ${KDE_VERSION} VERSION_LESS "4.1.90" )
+else ( ${KDE_VERSION} VERSION_LESS "4.1.90" )
+set( PLASMA_LIBS ${KDE4_PLASMA_LIBS} )
+endif ( ${KDE_VERSION} VERSION_LESS "4.1.90" )
+
 add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 include_directories( ${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES} )
  
