--- nss-mdns-0.10/configure.ac.orig	2011-02-10 09:15:35.577997072 +0100
+++ nss-mdns-0.10/configure.ac	2011-02-10 09:17:06.846285985 +0100
@@ -26,6 +26,8 @@
 AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([foreign 1.9 -Wall])
 
+AH_BOTTOM([#include "config_extra.h"])
+
 AC_SUBST(PACKAGE_URL, [http://0pointer.de/lennart/projects/nss-mdns/])
 
 ac_default_prefix="/"
--- /dev/null	2011-02-10 08:15:37.505999851 +0100
+++ nss-mdns-0.10/config_extra.h	2011-02-10 09:16:34.830938060 +0100
@@ -0,0 +1,3 @@
+#ifdef MDNS_MINIMAL
+#  undef HONOUR_SEARCH_DOMAINS
+#endif
