--- nss-mdns-0.10/src/nss.c.orig	2011-02-09 20:48:14.183106464 +0100
+++ nss-mdns-0.10/src/nss.c	2011-02-09 20:49:41.816945892 +0100
@@ -177,7 +177,7 @@
     return ends_with(name, ".local") || ends_with(name, ".local."); 
 }
 
-#ifdef HONOUR_SEARCH_DOMAINS
+#if defined(HONOUR_SEARCH_DOMAINS) && !defined(MDNS_MINIMAL)
 
 static char **alloc_domains(unsigned ndomains) {
     char **domains;
@@ -379,7 +379,7 @@
             status = NSS_STATUS_NOTFOUND;
     }
 
-#ifdef HONOUR_SEARCH_DOMAINS
+#if defined(HONOUR_SEARCH_DOMAINS) && !defined(MDNS_MINIMAL)
     if (u.count == 0 && avahi_works && !ends_with(name, ".")) {
         char **domains;
 
@@ -449,7 +449,7 @@
                 status = NSS_STATUS_NOTFOUND;
         }
 
-#ifdef HONOUR_SEARCH_DOMAINS
+#if defined(HONOUR_SEARCH_DOMAINS) && !defined(HONOUR_SEARCH_DOMAINS)
         if (u.count == 0 && !ends_with(name, ".")) {
             char **domains;
             
