ndex: extra/libkmap/libkmap/CMakeLists.txt
===================================================================
--- extra/libkmap/libkmap/CMakeLists.txt        (revision 1217491)
+++ extra/libkmap/libkmap/CMakeLists.txt        (working copy)
@@ -93,6 +93,7 @@
               kmap_primitives.h
               kmap_modelhelper.h
               kmap_geocoordinates.h
+              kmap_tileindex.h
               altitude-backend.h
               abstractmarkertiler.h
               itemmarkertiler.h
