diff -ur amarok-2.2.1-old/src/collection/mysqlecollection/CMakeLists.txt amarok-2.2.1/src/collection/mysqlecollection/CMakeLists.txt
--- amarok-2.2.1-old/src/collection/mysqlecollection/CMakeLists.txt	2009-11-30 23:31:05.000000000 -0100
+++ amarok-2.2.1/src/collection/mysqlecollection/CMakeLists.txt	2009-11-30 23:31:32.000000000 -0100
@@ -64,7 +64,6 @@
     ${KDE4_KIO_LIBS}
     ${KDE4_THREADWEAVER_LIBRARIES}
     ${MYSQL_EMBEDDED_LIBRARIES}
-    ${MYSQL_LIBRARIES}
     ${CMAKE_DL_LIBS}
     ${ZLIB_LIBRARIES}
     ${TAGLIB_LIBRARIES}
