--- a/cmake/modules/bindingstest/QtScriptBindingsTest.cpp	2009-11-14 15:07:20.400971341 +0200
+++ b/cmake/modules/bindingstest/QtScriptBindingsTest.cpp	2009-11-14 16:13:48.156222032 +0200
@@ -16,9 +16,7 @@
  * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-#include <KDE/KApplication>
-#include <KDE/KAboutData>
-#include <KDE/KCmdLineArgs>
+#include <qt4/QtCore/QCoreApplication>
 #include <QtScript/QScriptEngine>
 #include <QtCore/QStringList>
 #include <QtCore/QDebug>
@@ -27,9 +25,7 @@
 
 int main(int argc, char **argv)
 {
-    KAboutData about( "findgeneratorplugins", "", ki18n("Generator Exists?"), "1.0", ki18n("Find if the QtScript Plugins Are Installed"), KAboutData::License_LGPL_V2 );
-    KCmdLineArgs::init( argc, argv, &about );
-    KApplication app( false ); //no gui pls
+    QCoreApplication app( argc, argv ); //no gui pls
     
     QStringList allowedBindings;
     allowedBindings << "qt.core" << "qt.gui" << "qt.sql" << "qt.xml" << "qt.uitools" << "qt.network";
