--- CMakeLists.txt	2009-03-24 16:51:54.000000000 +0200
+++ CMakeLists.txt2	2009-03-24 17:06:57.000000000 +0200
@@ -38,9 +38,6 @@
 
     macro_log_feature( KDE4_FOUND "kdelibs" "The toolkit Amarok uses to build" "http://www.kde.org" TRUE "4.2.0" "" )
 
-    find_package( QtScriptQtBindings REQUIRED )
-    macro_log_feature( QTSCRIPTQTBINDINGS_FOUND "qtscript-qt" "QtScript Qt Bindings" "http://code.google.com/p/qtscriptgenerator/" TRUE "" "" )
-
     add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 
     find_package(MySQLAmarok REQUIRED)
