diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0dd2c31..a6d8467 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,8 @@ project(FactExtractor)
 
 cmake_minimum_required(VERSION 2.6)
 
+set(LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)")
+
 if (CMAKE_COMPILER_IS_GNUCXX)
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor -Wno-long-long -ansi -Wundef -Wcast-align -Wchar-subscripts -Wall -W -Wpointer-arith -Wformat-security -fno-check-new -fno-common -pedantic-errors")
   if (NOT MINGW)
diff --git a/libs/compilerprofile/CMakeLists.txt b/libs/compilerprofile/CMakeLists.txt
index 97a46cb..b081cbb 100644
--- a/libs/compilerprofile/CMakeLists.txt
+++ b/libs/compilerprofile/CMakeLists.txt
@@ -21,4 +21,4 @@ set_target_properties(compilerprofile PROPERTIES
                       DEFINE_SYMBOL MAKE_CPP_COMPILERPROFILE_LIB
 )
 
-install(TARGETS compilerprofile DESTINATION lib)
+install(TARGETS compilerprofile DESTINATION lib${LIB_SUFFIX})
diff --git a/libs/project/CMakeLists.txt b/libs/project/CMakeLists.txt
index 1299860..3c04bf7 100644
--- a/libs/project/CMakeLists.txt
+++ b/libs/project/CMakeLists.txt
@@ -22,5 +22,5 @@ set_target_properties(project PROPERTIES
                       DEFINE_SYMBOL MAKE_CPP_PROJECT_LIB
 )
 
-install(TARGETS project DESTINATION lib)
+install(TARGETS project DESTINATION lib${LIB_SUFFIX})
 
