--- trunk/KDE/kdebase/workspace/krunner/lock/lockdlg.cc	2010/02/12 17:50:14	1089212
+++ trunk/KDE/kdebase/workspace/krunner/lock/lockdlg.cc	2010/02/12 17:51:21	1089213
@@ -85,7 +85,8 @@
     : KDialog(parent, Qt::X11BypassWindowManagerHint),
       mPlugin( plugin ),
       mCapsLocked(-1),
-      mUnlockingFailed(false)
+      mUnlockingFailed(false),
+      sNot(0)
 {
     QWidget* w = mainWidget();
 
@@ -435,6 +436,8 @@
 //
 void PasswordDlg::gplugStart()
 {
+    if (sNot)
+        return;
     int sfd[2];
     char fdbuf[16];
 
@@ -465,6 +468,7 @@
     sFd = sfd[0];
     sNot = new QSocketNotifier(sFd, QSocketNotifier::Read, this);
     connect(sNot, SIGNAL(activated(int)), SLOT(handleVerify()));
+    connect(sNot, SIGNAL(destroyed()), SLOT(slotNotifierDestroyed()));
 }
 
 void PasswordDlg::gplugChanged()
@@ -651,6 +655,11 @@
         KDisplayManager().switchVT( itm->vt );
 }
 
+void PasswordDlg::slotNotifierDestroyed()
+{
+    sNot = 0;
+}
+
 void PasswordDlg::capsLocked()
 {
     unsigned int lmask;
--- trunk/KDE/kdebase/workspace/krunner/lock/lockdlg.h	2010/02/12 17:50:14	1089212
+++ trunk/KDE/kdebase/workspace/krunner/lock/lockdlg.h	2010/02/12 17:51:21	1089213
@@ -63,6 +63,7 @@
     void slotOK();
     void slotActivity();
     void handleVerify();
+    void slotNotifierDestroyed();
 
 private:
     void capsLocked();
