diff -ruN krunner-4.6.2.orig/krunner/lock/CMakeLists.txt krunner-4.6.2/krunner/lock/CMakeLists.txt
--- krunner-4.6.2.orig/krunner/lock/CMakeLists.txt	2011-02-25 22:54:48.000000000 +0100
+++ krunner-4.6.2/krunner/lock/CMakeLists.txt	2011-04-08 00:09:58.749240497 +0200
@@ -2,7 +2,7 @@
 
 ########### next target ###############
 
-check_library_exists(Xxf86misc XF86MiscSetGrabKeysState "" HAVE_XF86MISCSETGRABKEYSSTATE)
+set(HAVE_XF86MISC 0)
 if(WITH_OpenGL)
 check_library_exists(GL glXChooseVisual "" HAVE_GLXCHOOSEVISUAL)
 endif(WITH_OpenGL)
