Index: kontact/plugins/CMakeLists.txt
===================================================================
--- kontact/plugins/CMakeLists.txt	(revision 1073383)
+++ kontact/plugins/CMakeLists.txt	(working copy)
@@ -1,18 +1,16 @@
 remove_definitions(-DKDE_DEFAULT_DEBUG_AREA=5600) #remove kontact debug area
 add_definitions(-DKDE_DEFAULT_DEBUG_AREA=5602)
 
-if(QGPGME_FOUND)
-  if(BUILD_kmail)
-    add_subdirectory( kmail )
-  endif(BUILD_kmail)
-  if (BUILD_kmail AND BUILD_korganizer)
-    add_subdirectory( korganizer )
-  endif (BUILD_kmail AND BUILD_korganizer)
-  add_subdirectory( summary )
-  add_subdirectory( specialdates )
+if(BUILD_kmail)
+  add_subdirectory( kmail )
+endif(BUILD_kmail)
+if (BUILD_kmail AND BUILD_korganizer)
+  add_subdirectory( korganizer )
+endif (BUILD_kmail AND BUILD_korganizer)
+add_subdirectory( summary )
+add_subdirectory( specialdates )
 #re-enable planner for 4.2 if we have people who want to develop it
-  add_subdirectory( planner )
-endif(QGPGME_FOUND)
+add_subdirectory( planner )
 if (BUILD_knotes)
   add_subdirectory( knotes )
 endif (BUILD_knotes)
