diff -ruN ksnapshot-4.6.2.orig/ksnapshot/CMakeLists.txt ksnapshot-4.6.2/ksnapshot/CMakeLists.txt
--- ksnapshot-4.6.2.orig/ksnapshot/CMakeLists.txt	2011-04-01 21:38:15.000000000 +0200
+++ ksnapshot-4.6.2/ksnapshot/CMakeLists.txt	2011-04-10 22:02:39.035066966 +0200
@@ -1,7 +1,7 @@
 project(ksnapshot)
 
 find_package(KDE4 REQUIRED)
-find_package(Kipi)
+
 include(KDE4Defaults)
 include(MacroLibrary)
 include(MacroOptionalAddSubdirectory)
@@ -17,7 +17,6 @@
 if (KIPI_FOUND)
         include_directories(${KIPI_INCLUDE_DIR})
 endif (KIPI_FOUND)
-macro_log_feature(KIPI_FOUND "KIPI plugins" "KIPI plugins to export, print, etc" "http://www.kipi-plugins.org/" FALSE "" "Neeed to print, send by e-mail, export to Facebook, Picasa, etc")
 
 configure_file(config-ksnapshot.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-ksnapshot.h)
 
