--- a/CMakeLists.txt	2015-02-23 23:12:24.420881583 +0100
+++ b/CMakeLists.txt	2015-02-23 23:17:07.173858317 +0100
@@ -18,7 +18,17 @@
 include(ECMAddTests)
 
 
-find_package(Qt5 CONFIG REQUIRED COMPONENTS Core DBus Test Widgets PrintSupport Svg Qml Quick)
+find_package(Qt5 CONFIG REQUIRED COMPONENTS Core DBus Widgets PrintSupport Svg Qml Quick)
+
+find_package(Qt5Test CONFIG QUIET)
+set_package_properties(Qt5Test PROPERTIES
+    PURPOSE "Required for tests"
+    TYPE OPTIONAL)
+add_feature_info("Qt5Test" Qt5Test_FOUND "Required for building tests")
+if (NOT Qt5Test_FOUND)
+    set(BUILD_TESTING OFF CACHE BOOL "Build the testing tree.")
+endif()
+
 find_package(Qt5 OPTIONAL_COMPONENTS TextToSpeech)
 if (NOT Qt5TextToSpeech_FOUND)
     message(STATUS "Qt5TextToSpeech not found, speech features will be disabled")
