commit de71c97def75f047ec0dbaabffb8bc698938e84f
Author: David Edmundson <kde@davidedmundson.co.uk>
Date:   Wed Oct 30 19:19:23 2013 +0100

    Make each plugin optional

diff --git a/plugins/CMakeLists.txt b/plugins/CMakeLists.txt
index e270078..e752ef1 100644
--- a/plugins/CMakeLists.txt
+++ b/plugins/CMakeLists.txt
@@ -1,8 +1,8 @@
-add_subdirectory (gabble)
-add_subdirectory (butterfly)
-add_subdirectory (idle)
-add_subdirectory (haze)
-add_subdirectory (rakia)
-add_subdirectory (salut)
-add_subdirectory (sunshine)
-add_subdirectory (pintxo)
+macro_optional_add_subdirectory (gabble)
+macro_optional_add_subdirectory (butterfly)
+macro_optional_add_subdirectory (idle)
+macro_optional_add_subdirectory (haze)
+macro_optional_add_subdirectory (rakia)
+macro_optional_add_subdirectory (salut)
+macro_optional_add_subdirectory (sunshine)
+macro_optional_add_subdirectory (pintxo)
