--- virtuoso-opensource-6.1.4/libsrc/Wi/http_client.c
+++ virtuoso-opensource-6.1.4/libsrc/Wi/http_client.c
@@ -423,7 +423,7 @@
   socksreq[8] = 0; /* no name */
   if (name)
     {
-      strncat ((char*)socksreq + 8, name, sizeof(socksreq) - 8);
+      strncat ((char*)socksreq + 8, name, sizeof(socksreq) - 9);
       socksreq[sizeof (socksreq) - 1] = 0;
       packetsize = 9 + strlen ((char *) socksreq + 8);
     }
