diff -Naur krossjava-4.2.3.old/java/krossjava/krossjava/jvmvariant.h krossjava-4.2.3/java/krossjava/krossjava/jvmvariant.h
--- krossjava-4.2.3.old/java/krossjava/krossjava/jvmvariant.h	2009-05-23 15:50:12.000000000 +1200
+++ krossjava-4.2.3/java/krossjava/krossjava/jvmvariant.h	2009-05-23 15:53:20.000000000 +1200
@@ -30,9 +30,9 @@
 #include <QStringList>
 #include <QVariant>
 #include <QMetaType>
-//#include <QSize>
-//#include <QPoint>
-//#include <QRect>
+#include <QSize>
+#include <QPoint>
+#include <QRect>
 #include <QUrl>
 //#include <QDate>
 //#include <QTime>
