diff -r -u koffice-libs-2.2.0.orig/libs/koreport/CMakeLists.txt koffice-libs-2.2.0/libs/koreport/CMakeLists.txt
--- koffice-libs-2.2.0.orig/libs/koreport/CMakeLists.txt	2010-05-23 08:21:05.000000000 +0200
+++ koffice-libs-2.2.0/libs/koreport/CMakeLists.txt	2010-06-02 22:56:18.104790383 +0200
@@ -100,7 +100,7 @@
 target_link_libraries(koreport ${KDE4_KDEUI_LIBS} ${QT_QTGUI_LIBRARY}
   ${KOPROPERTY_LIBS} kdchart ${KDE4_KROSSCORE_LIBS} komain ${QT_QT3SUPPORT_LIBRARY})
 if (SHOULD_BUILD_KSPREAD)
-    target_link_libraries(koreport kspreadcommon)
+    target_link_libraries(koreport)
 endif (SHOULD_BUILD_KSPREAD)
 
 set_target_properties(koreport PROPERTIES VERSION ${GENERIC_KOFFICE_LIB_VERSION} SOVERSION ${GENERIC_KOFFICE_LIB_SOVERSION} )
