--- okteta/core/piecetable/tests/CMakeLists.txt
+++ okteta/core/piecetable/tests/CMakeLists.txt
@@ -134,26 +134,6 @@
   oktetacoretestutil ${KDE4_KDECORE_LIBS} ${QT_QTTEST_LIBRARY} )
 
 
-########### grouppiecetablechangeabstractpiecetablechangeiftest ###############
-
-set( grouppiecetablechangeabstractpiecetablechangeiftest_SRCS
-  grouppiecetablechangeabstractpiecetablechangeiftest.cpp
-  ${abstractpiecetablechangeiftest_SRCS}
-  ${SUTDIR}/addressrangelist.cpp
-  ${SUTDIR}/piecetable/replacepiecetablechange.cpp
-  ${SUTDIR}/piecetable/insertpiecetablechange.cpp
-  ${SUTDIR}/piecetable/removepiecetablechange.cpp
-  ${SUTDIR}/piecetable/grouppiecetablechange.cpp
-)
-
-kde4_add_unit_test( grouppiecetablechangeabstractpiecetablechangeiftest
-  TESTNAME libpiecetable-grouppiecetablechangeabstractpiecetablechangeiftest
-  ${grouppiecetablechangeabstractpiecetablechangeiftest_SRCS}
-)
-target_link_libraries( grouppiecetablechangeabstractpiecetablechangeiftest
-  oktetacoretestutil ${KDE4_KDECORE_LIBS} ${QT_QTTEST_LIBRARY} )
-
-
 ########### grouppiecetablechangetest ###############
 
 set( grouppiecetablechangetest_SRCS
