diff --git a/src/plugins/runner/CMakeLists.txt b/src/plugins/runner/CMakeLists.txt
index 4ab295a..0b82fa7 100644
--- a/src/plugins/runner/CMakeLists.txt
+++ b/src/plugins/runner/CMakeLists.txt
@@ -26,7 +26,7 @@ add_subdirectory( osm )
 add_subdirectory( pnt )
 add_subdirectory( log )
 
-find_package( libshp )
+macro_optional_find_package( libshp )
 if( LIBSHP_FOUND )
   add_subdirectory( shp )
 endif( LIBSHP_FOUND )
