diff -ruN marble-4.6.2.orig/marble/src/plugins/positionprovider/CMakeLists.txt marble-4.6.2/marble/src/plugins/positionprovider/CMakeLists.txt
--- marble-4.6.2.orig/marble/src/plugins/positionprovider/CMakeLists.txt	2011-01-19 23:06:11.000000000 +0100
+++ marble-4.6.2/marble/src/plugins/positionprovider/CMakeLists.txt	2011-04-11 03:56:00.449066967 +0200
@@ -1,4 +1,4 @@
-find_package(libgps)
+macro_optional_find_package(libgps)
 if(LIBGPS_FOUND)
      message( STATUS "Building with gpsd position provider")
      add_subdirectory( gpsd )
@@ -6,7 +6,7 @@
      message( STATUS "Not building with gpsd position provider")
 endif(LIBGPS_FOUND)
 
-find_package(liblocation)
+macro_optional_find_package(liblocation)
 if(liblocation_FOUND)
      message( STATUS "Building with Maemo GPS support")
      add_subdirectory( maemo )
