diff -urN marble-4.7.49.9999.orig/src/plugins/positionprovider/CMakeLists.txt marble-4.7.49.9999/src/plugins/positionprovider/CMakeLists.txt
--- marble-4.7.49.9999.orig/src/plugins/positionprovider/CMakeLists.txt	2011-07-04 12:42:13.869879320 +0200
+++ marble-4.7.49.9999/src/plugins/positionprovider/CMakeLists.txt	2011-07-04 12:43:28.567021582 +0200
@@ -1,4 +1,4 @@
-FIND_PACKAGE(libgps)
+macro_optional_find_package(libgps)
 IF(LIBGPS_FOUND)
      MESSAGE( STATUS "Building with gpsd position provider")
      ADD_SUBDIRECTORY( gpsd )
@@ -6,7 +6,7 @@
      MESSAGE( STATUS "Not building with gpsd position provider")
 ENDIF(LIBGPS_FOUND)
 
-FIND_PACKAGE(liblocation)
+macro_optional_find_package(liblocation)
 IF(liblocation_FOUND)
      MESSAGE( STATUS "Building with Maemo GPS support")
      ADD_SUBDIRECTORY( maemo )
@@ -19,4 +19,4 @@
 ENDIF(QTLOCATION_FOUND)
 
 # experimental implementation
-# ADD_SUBDIRECTORY( geoclue )
\ Kein Zeilenumbruch am Dateiende.
+# ADD_SUBDIRECTORY( geoclue )
