diff --git a/src/bindings/python/sip/MarbleModel.sip b/src/bindings/python/sip/MarbleModel.sip
index b27c855..818e896 100644
--- a/src/bindings/python/sip/MarbleModel.sip
+++ b/src/bindings/python/sip/MarbleModel.sip
@@ -31,7 +31,7 @@ public:
     void                    removePlacemarkKey (const QString& key);
     qreal                   planetRadius () const;
     QString                 planetName () const;
-    Marble::SunLocator*     sunLocator () const;
+    const Marble::SunLocator*     sunLocator () const;
 //ig    quint64                 volatileTileCacheLimit () const;
     const Marble::Planet*   planet () const;
     void                    clearPersistentTileCache ();
