diff -ruN digikam-2.6.0-beta3.orig/extra/kipi-plugins/CMakeLists.txt digikam-2.6.0-beta3/extra/kipi-plugins/CMakeLists.txt
--- digikam-2.6.0-beta3.orig/extra/kipi-plugins/CMakeLists.txt	2012-04-01 12:54:26.000000000 +0200
+++ digikam-2.6.0-beta3/extra/kipi-plugins/CMakeLists.txt	2012-04-08 18:50:53.918419219 +0200
@@ -391,7 +391,11 @@
     ADD_SUBDIRECTORY(picasawebexport)
     ADD_SUBDIRECTORY(smug)
     ADD_SUBDIRECTORY(printimages)
-    ADD_SUBDIRECTORY(expoblending)
+
+    IF(ENABLE_expoblending)
+        ADD_SUBDIRECTORY(expoblending)
+    ENDIF(ENABLE_expoblending)
+
     ADD_SUBDIRECTORY(kopete)
     ADD_SUBDIRECTORY(batchprocessimages)
     ADD_SUBDIRECTORY(rajceexport)
@@ -457,9 +461,11 @@
         ADD_SUBDIRECTORY(mediawiki)
     ENDIF(MEDIAWIKI_FOUND)
 
-    IF(Boost_FOUND)
-        ADD_SUBDIRECTORY(panorama)
-    ENDIF(Boost_FOUND)
+    IF(ENABLE_panorama)
+        IF(Boost_FOUND)
+            ADD_SUBDIRECTORY(panorama)
+        ENDIF(Boost_FOUND)
+    ENDIF(ENABLE_panorama)
 
 #   Laurent Montel : wallpaper is currently disabled. Waiting new api with KDE 4.2
 #
