diff -ruN kipi-plugins-1.7.0.orig/CMakeLists.txt kipi-plugins-1.7.0/CMakeLists.txt
--- kipi-plugins-1.7.0.orig/CMakeLists.txt	2010-12-19 10:44:04.000000000 +0100
+++ kipi-plugins-1.7.0/CMakeLists.txt	2010-12-28 18:11:40.000000000 +0100
@@ -358,7 +358,11 @@
     ADD_SUBDIRECTORY(smug)
     ADD_SUBDIRECTORY(facebook)
     ADD_SUBDIRECTORY(printimages)
-    ADD_SUBDIRECTORY(expoblending)
+
+    IF(ENABLE_expoblending)
+        ADD_SUBDIRECTORY(expoblending)
+    ENDIF(ENABLE_expoblending)
+
     ADD_SUBDIRECTORY(kopete)
 
     # Plugins which are optional depencies.
