diff -r -u kipi-plugins-1.2.0.orig/CMakeLists.txt kipi-plugins-1.2.0/CMakeLists.txt
--- kipi-plugins-1.2.0.orig/CMakeLists.txt	2010-03-28 23:57:54.000000000 +0200
+++ kipi-plugins-1.2.0/CMakeLists.txt	2010-07-08 22:56:13.518262763 +0200
@@ -368,9 +368,9 @@
         ENDIF(CMAKE_USE_PTHREADS_INIT OR CMAKE_USE_WIN32_THREADS_INIT)
     ENDIF(EXPAT_FOUND AND KDCRAW_FOR_DNGCONVERTER)
 
-    IF(KDCRAW_FOR_EXPOBLENDING)
+    IF(KDCRAW_FOR_EXPOBLENDING AND ENABLE_expoblending)
         ADD_SUBDIRECTORY(expoblending)
-    ENDIF(KDCRAW_FOR_EXPOBLENDING)
+    ENDIF(KDCRAW_FOR_EXPOBLENDING AND ENABLE_expoblending)
 
 #   Laurent Montel : wallpaper is currently disabled. Waiting new api with KDE 4.2
 #
