--- a/CMakeLists.txt	2015-08-02 14:04:46.650113184 +0200
+++ b/CMakeLists.txt	2015-09-15 19:07:39.000000000 +0200
@@ -310,7 +310,10 @@
         add_subdirectory(kmlexport)
         add_subdirectory(sendimages)
         add_subdirectory(flashexport)
-        add_subdirectory(expoblending)
+
+        if(ENABLE_expoblending)
+            add_subdirectory(expoblending)
+        endif()
 
         if(BUILD_VIEWERS)
             add_subdirectory(advancedslideshow)
