--- kstars/skycomponents/supernovaecomponent.cpp	2015-04-23 15:10:15.927651208 +0200
+++ kstars/skycomponents/supernovaecomponent.cpp.new	2015-04-23 15:11:03.107647326 +0200
@@ -233,7 +233,7 @@
     //qDebug()<<filename;
     m_Parser = new QProcess;
     connect( m_Parser, SIGNAL( finished( int, QProcess::ExitStatus ) ), this, SLOT( slotDataFileUpdateFinished( int, QProcess::ExitStatus ) ) );
-    m_Parser->start("python", args);
+    m_Parser->start("python2", args);
 }
 
 void SupernovaeComponent::slotDataFileUpdateFinished( int exitCode, QProcess::ExitStatus exitStatus )
