--- a/CMakeLists.txt	2015-01-21 22:20:04.283180998 +0100
+++ b/CMakeLists.txt	2015-01-21 22:20:28.328951019 +0100
@@ -30,7 +30,7 @@
 include(KDECMakeSettings)
 include(KDECompilerSettings)
 
-find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core DBus Widgets Script Sql Test)
+find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core DBus Widgets Script Sql)
 
 # Load the frameworks we need
 find_package(KF5 REQUIRED COMPONENTS
