Index: kdenlive-0.7/cmake/modules/FindLIBFFMPEG.cmake
===================================================================
--- kdenlive-0.7.orig/cmake/modules/FindLIBFFMPEG.cmake
+++ kdenlive-0.7/cmake/modules/FindLIBFFMPEG.cmake
@@ -33,6 +33,13 @@ FIND_LIBRARY(LIBFFMPEG_LIBRARY
   NO_DEFAULT_PATH
 )
 
+FIND_LIBRARY(LIBAVCODEC_LIBRARY
+  NAMES avcodec
+  PATHS ${FFMPEG_LIBS_PATH}
+  /usr/lib /usr/local/lib
+  NO_DEFAULT_PATH
+)
+
 IF (LIBFFMPEG_LIBRARY)
   MESSAGE(STATUS "Found FFmpeg Libavformat library: ${LIBFFMPEG_LIBRARY}")
 ELSE (LIBFFMPEG_LIBRARY)
Index: kdenlive-0.7/src/CMakeLists.txt
===================================================================
--- kdenlive-0.7.orig/src/CMakeLists.txt
+++ kdenlive-0.7/src/CMakeLists.txt
@@ -17,6 +17,7 @@ LINK_LIBRARIES(
 ${LIBMLT_LIBRARY}
 ${LIBMLTPLUS_LIBRARY}
 ${LIBFFMPEG_LIBRARY}
+${LIBAVCODEC_LIBRARY}
 ${NEPOMUK_LIBRARIES}
 )
 
