commit 1d0d86f017ab4b53442c1c6eb04d3d50b483a741
Author: Montel Laurent <montel@kde.org>
Date:   Tue Jul 28 07:01:16 2015 +0200

    Remove KF5KDELibs4Support

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2655c6a..bb55630 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,8 +34,7 @@ find_package(KF5TextWidgets ${KF5_VERSION} CONFIG REQUIRED)
 find_package(KF5WidgetsAddons ${KF5_VERSION} CONFIG REQUIRED)
 find_package(KF5KIO ${KF5_VERSION} CONFIG REQUIRED)
 find_package(KF5Codecs ${KF5_VERSION} CONFIG REQUIRED)
-#REMOVE IT WHEN all kdelib4support removed
-find_package(KF5KDELibs4Support ${KF5_VERSION} CONFIG REQUIRED)
+find_package(KF5IconThemes ${KF5_VERSION} CONFIG REQUIRED)
 
 add_definitions(-DTRANSLATION_DOMAIN=\"libkpimtextedit\")
 add_definitions("-DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_TO_ASCII")
