--- apps/konsole/src/ViewContainer.cpp	2009/12/05 21:05:51	1059123
+++ apps/konsole/src/ViewContainer.cpp	2009/12/05 21:11:32	1059124
@@ -429,7 +429,7 @@
     _tabBar = new ViewContainerTabBar(_containerWidget,this);
     _tabBar->setDrawBase(true);
     _tabBar->setDocumentMode(true);
-    _tabBar->setFocusPolicy(Qt::ClickFocus);
+    _tabBar->setFocusPolicy(Qt::NoFocus);
 
     _newTabButton = new QToolButton(_containerWidget);
     _newTabButton->setIcon(KIcon("tab-new"));
