Index: kdebase/apps/konsole/src/ViewManager.cpp
===================================================================
--- kdebase/apps/konsole/src/ViewManager.cpp	(Revision 1133013)
+++ kdebase/apps/konsole/src/ViewManager.cpp	(Revision 1133014)
@@ -646,6 +646,7 @@
         return;
 
     createView(controller->session(),container,index);
+    controller->session()->refresh();
     moved = true;
 }
 void ViewManager::setNavigationMethod(NavigationMethod method)
