diff -u -r konsole-4.4.3.orig/konsole/data/keyboard-layouts/default.keytab konsole-4.4.3/konsole/data/keyboard-layouts/default.keytab
--- konsole-4.4.3.orig/konsole/data/keyboard-layouts/default.keytab	2009-08-12 11:59:24.000000000 +0200
+++ konsole-4.4.3/konsole/data/keyboard-layouts/default.keytab	2010-05-22 22:53:47.000000000 +0200
@@ -32,9 +32,7 @@
 
 key Return+Shift         : "\EOM"
 
-# Backspace and Delete codes are preserving CTRL-H.
-
-key Backspace      : "\x7f"
+key Backspace      : "\b"
 
 # Arrow keys in VT52 mode
 # shift up/down are reserved for scrolling.
