diff -ruN konsole-4.7.0.orig/src/tests/CMakeLists.txt konsole-4.7.0/src/tests/CMakeLists.txt
--- konsole-4.7.0.orig/src/tests/CMakeLists.txt	2011-05-21 01:08:13.000000000 +0200
+++ konsole-4.7.0/src/tests/CMakeLists.txt	2011-08-03 23:05:10.864318665 +0200
@@ -12,6 +12,6 @@
 kde4_add_unit_test(SessionManagerTest SessionManagerTest.cpp)
 target_link_libraries(SessionManagerTest ${KONSOLE_TEST_LIBS})
 
-kde4_add_unit_test(DBusTest DBusTest.cpp)
-target_link_libraries(DBusTest ${KONSOLE_TEST_LIBS})
+#kde4_add_unit_test(DBusTest DBusTest.cpp)
+#target_link_libraries(DBusTest ${KONSOLE_TEST_LIBS})
 
