diff -ruN rocs-4.5.80.orig/doc/rocs/index.docbook rocs-4.5.80/doc/rocs/index.docbook
--- rocs-4.5.80.orig/doc/rocs/index.docbook	2010-10-27 21:55:08.000000000 +0200
+++ rocs-4.5.80/doc/rocs/index.docbook	2010-11-28 22:35:11.590646446 +0100
@@ -177,9 +177,9 @@
 </programlisting> 
 <para>Node Variables:</para>
 <programlisting>
-    double x
-    double y
-    double width
+    qreal x
+    qreal y
+    qreal width
     string value
     string name
     string color  // in HEXA
@@ -194,7 +194,7 @@
     string color ( em hexa )
     string value
     string name
-    double width
+    qreal width
     string style ( dot, dash, dash dot, solid )
 
     node start();
