diff -ru rocs-4.5.3.orig/rocs/src/Core/edge.h rocs-4.5.3/rocs/src/Core/edge.h
--- rocs-4.5.3.orig/rocs/src/Core/edge.h	2010-06-08 12:10:59.000000000 +0200
+++ rocs-4.5.3/rocs/src/Core/edge.h	2010-11-10 01:23:28.283410355 +0100
@@ -62,7 +62,7 @@
     Q_PROPERTY(QString name READ name WRITE setName)
 
     /*! this property holds the width of the edge */
-    Q_PROPERTY(double width READ width WRITE setWidth)
+    Q_PROPERTY(qreal width READ width WRITE setWidth)
 
     /*! this property holds the style of the edge */
     Q_PROPERTY(QString style READ style WRITE setStyle)
@@ -169,7 +169,7 @@
     qreal width () const {
         return _width;
     }
-    void setWidth(double w) {
+    void setWidth(qreal w) {
         _width = w;
         emit changed();
     }
@@ -239,7 +239,7 @@
     bool _showValue;
 
     QString _style;
-    double _width;
+    qreal _width;
 
      Graph *_graph;
 #ifdef USING_QTSCRIPT
diff -ru rocs-4.5.3.orig/rocs/src/Core/node.h rocs-4.5.3/rocs/src/Core/node.h
--- rocs-4.5.3.orig/rocs/src/Core/node.h	2010-06-08 12:10:59.000000000 +0200
+++ rocs-4.5.3/rocs/src/Core/node.h	2010-11-10 01:23:32.241410345 +0100
@@ -85,7 +85,7 @@
 
     void setX(int x);
     void setY(int y);
-    void setWidth(double w);
+    void setWidth(qreal w);
     void setPos(qreal x, qreal y);
     qreal x() const;
     qreal y() const;
diff -ru rocs-4.5.3.orig/rocs/src/GraphicsItem/math_constants.h rocs-4.5.3/rocs/src/GraphicsItem/math_constants.h
--- rocs-4.5.3.orig/rocs/src/GraphicsItem/math_constants.h	2010-06-08 12:10:59.000000000 +0200
+++ rocs-4.5.3/rocs/src/GraphicsItem/math_constants.h	2010-11-10 01:23:04.291410343 +0100
@@ -23,10 +23,10 @@
 #define MATH_CONSTANTS_H
 
 
-static const double Pi = 3.14159265358979323846264338327950288419717;
+static const qreal Pi = 3.14159265358979323846264338327950288419717;
 
 /// The value of Pi * 2
-static const double TwoPi = 2.0 * Pi;
+static const qreal TwoPi = 2.0 * Pi;
 
 /// The Value of PI / 3
 static const qreal PI_3 = Pi / 3.0;
diff -ru rocs-4.5.3.orig/rocs/src/Interface/NodePropertiesWidget.cpp rocs-4.5.3/rocs/src/Interface/NodePropertiesWidget.cpp
--- rocs-4.5.3.orig/rocs/src/Interface/NodePropertiesWidget.cpp	2010-06-16 14:03:03.000000000 +0200
+++ rocs-4.5.3/rocs/src/Interface/NodePropertiesWidget.cpp	2010-11-10 01:22:25.998410281 +0100
@@ -43,7 +43,7 @@
     connect( _value,        SIGNAL(textChanged(QString)),   _node, SLOT(setValue(QString)));
     connect( _x,            SIGNAL(valueChanged(int)),      _node, SLOT(setX(int)));
     connect( _y,            SIGNAL(valueChanged(int)),      _node, SLOT(setY(int)));
-    connect( _width,        SIGNAL(valueChanged(double)),   _node, SLOT(setWidth(double)));
+    connect( _width,        SIGNAL(valueChanged(qreal)),   _node, SLOT(setWidth(qreal)));
 
     GraphPropertiesModel *model = new GraphPropertiesModel();
     model->setDataSource(_node);
@@ -136,6 +136,6 @@
     disconnect( _value,        SIGNAL(textChanged(QString)),   n, SLOT(setValue(QString)));
     disconnect( _x,            SIGNAL(valueChanged(int)),      n, SLOT(setX(int)));
     disconnect( _y,            SIGNAL(valueChanged(int)),      n, SLOT(setY(int)));
-    disconnect( _width,        SIGNAL(valueChanged(double)),   n, SLOT(setWidth(double)));
+    disconnect( _width,        SIGNAL(valueChanged(qreal)),   n, SLOT(setWidth(qreal)));
 
 }
\ Kein Zeilenumbruch am Dateiende.
diff -ru rocs-4.5.3.orig/rocs/src/Interface/edgepropertieswidget.cpp rocs-4.5.3/rocs/src/Interface/edgepropertieswidget.cpp
--- rocs-4.5.3.orig/rocs/src/Interface/edgepropertieswidget.cpp	2010-06-08 12:10:59.000000000 +0200
+++ rocs-4.5.3/rocs/src/Interface/edgepropertieswidget.cpp	2010-11-10 01:22:33.872410348 +0100
@@ -52,7 +52,7 @@
 
     connect(_value,     SIGNAL(textChanged(QString)),   _edge, SLOT(setValue(QString)));
     connect(_name,      SIGNAL(textChanged(QString)),   _edge, SLOT(setName(QString)));
-    connect(_width,     SIGNAL(valueChanged(double)),    _edge, SLOT(setWidth(double)));
+    connect(_width,     SIGNAL(valueChanged(qreal)),    _edge, SLOT(setWidth(qreal)));
     connect(_showName,  SIGNAL(toggled(bool)),          _edge, SLOT(hideName(bool)));
     connect(_showValue, SIGNAL(toggled(bool)),          _edge, SLOT(hideValue(bool)));
 
@@ -101,7 +101,7 @@
 
     disconnect(_value,     SIGNAL(textChanged(QString)),   _edge, SLOT(setValue(QString)));
     disconnect(_name,      SIGNAL(textChanged(QString)),   _edge, SLOT(setName(QString)));
-    disconnect(_width,     SIGNAL(valueChanged(double)),    _edge, SLOT(setWidth(double)));
+    disconnect(_width,     SIGNAL(valueChanged(qreal)),    _edge, SLOT(setWidth(qreal)));
     disconnect(_showName,  SIGNAL(toggled(bool)),          _edge, SLOT(hideName(bool)));
     disconnect(_showValue, SIGNAL(toggled(bool)),          _edge, SLOT(hideValue(bool)));
 }
