From 53bea19f57029c2953d6397f20620afcb9a07626 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Thu, 24 Sep 2020 22:34:28 +0200
Subject: [PATCH] xembed-sni-proxy: Allow standalone build

---
 xembed-sni-proxy/CMakeLists.txt | 36 +++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/xembed-sni-proxy/CMakeLists.txt b/xembed-sni-proxy/CMakeLists.txt
index f1677e754c..6e6f3b0590 100644
--- a/xembed-sni-proxy/CMakeLists.txt
+++ b/xembed-sni-proxy/CMakeLists.txt
@@ -1,3 +1,35 @@
+project(xembed-sni-proxy)
+
+if(${CMAKE_SOURCE_DIR} STREQUAL ${xembed-sni-proxy_SOURCE_DIR})
+    set(PROJECT_VERSION "5.90.0")
+    set(PROJECT_VERSION_MAJOR 6)
+
+    cmake_minimum_required(VERSION 3.16)
+
+    set(QT_MIN_VERSION "6.5.0")
+    set(KF6_MIN_VERSION "5.240")
+    find_package(Qt6 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS Core DBus Gui)
+    find_package(ECM ${KF6_MIN_VERSION} REQUIRED NO_MODULE)
+    set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
+
+    include(KDEInstallDirs)
+    include(KDECMakeSettings)
+    include(KDECompilerSettings NO_POLICY_SCOPE)
+    include(FeatureSummary)
+    include(ECMQtDeclareLoggingCategory)
+    include(ECMConfiguredInstall)
+
+    find_package(KF6WindowSystem ${KF6_MIN_VERSION} REQUIRED)
+    find_package(X11 REQUIRED)
+    find_package(XCB MODULE REQUIRED COMPONENTS XCB RANDR)
+    set_package_properties(XCB PROPERTIES TYPE REQUIRED)
+
+    configure_file(../config-X11.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-X11.h)
+else()
+    set(PROJECT_VERSION "5.90.0")
+    set(PROJECT_VERSION_MAJOR 6)
+endif()
+
 add_definitions(-DQT_NO_CAST_TO_ASCII
 -DQT_NO_CAST_FROM_ASCII
 -DQT_NO_CAST_FROM_BYTEARRAY)
@@ -67,3 +99,7 @@ install(TARGETS xembedsniproxy ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 install(FILES xembedsniproxy.desktop DESTINATION ${KDE_INSTALL_AUTOSTARTDIR})
 
 ecm_install_configured_files(INPUT plasma-xembedsniproxy.service.in @ONLY DESTINATION  ${KDE_INSTALL_SYSTEMDUSERUNITDIR})
+
+if ("${CMAKE_BINARY_DIR}" STREQUAL "${CMAKE_CURRENT_BINARY_DIR}")
+    feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
+endif()
-- 
2.43.0

