diff -ruN baloo-4.13.0.orig/src/file/CMakeLists.txt baloo-4.13.0/src/file/CMakeLists.txt
--- baloo-4.13.0.orig/src/file/CMakeLists.txt	2014-04-15 18:30:06.000000000 +0200
+++ baloo-4.13.0/src/file/CMakeLists.txt	2014-05-04 02:58:00.312192920 +0200
@@ -66,6 +66,6 @@
 add_subdirectory(autotest)
 add_subdirectory(tests)
 add_subdirectory(extractor)
-add_subdirectory(kcm)
+#add_subdirectory(kcm)
 add_subdirectory(inotify)
 add_subdirectory(cleaner)
