commit 4dad707126176aadf7904660e7d11e0e9fc3d99f
Author: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date:   Mon Aug 15 00:46:02 2016 +0200

    Fix build with disabled TelepathyQt5

diff --git a/plugins/imstatus/CMakeLists.txt b/plugins/imstatus/CMakeLists.txt
index 7756eb8..55154eb 100644
--- a/plugins/imstatus/CMakeLists.txt
+++ b/plugins/imstatus/CMakeLists.txt
@@ -1,6 +1,8 @@
 find_package(TelepathyQt5)
 
-configure_file(config-imstatus.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-imstatus.h)
+if(TelepathyQt5_FOUND)
+    add_definitions( -DTELEPATHY_FOUND )
+endif()
 
 include_directories(
     ${CHOQOK_INCLUDES}
diff --git a/plugins/imstatus/config-imstatus.h.cmake b/plugins/imstatus/config-imstatus.h.cmake
deleted file mode 100644
index 12a498a..0000000
--- a/plugins/imstatus/config-imstatus.h.cmake
+++ /dev/null
@@ -1 +0,0 @@
-#define TELEPATHY_FOUND ${TelepathyQt5_FOUND}
diff --git a/plugins/imstatus/imqdbus.cpp b/plugins/imstatus/imqdbus.cpp
index 5840035..ce11790 100644
--- a/plugins/imstatus/imqdbus.cpp
+++ b/plugins/imstatus/imqdbus.cpp
@@ -28,7 +28,7 @@
 #include <QDBusReply>
 #include <QDebug>
 
-#if TELEPATHY_FOUND
+#ifdef TELEPATHY_FOUND
 #include <TelepathyQt/Account>
 #include <TelepathyQt/AccountManager>
 #include <TelepathyQt/AccountSet>
@@ -49,7 +49,7 @@ IMQDBus::IMQDBus(QObject *parent) : QObject(parent)
     - qutIM (>0.3)
     - gajim ( doesn't want work :( )
     */
-#if TELEPATHY_FOUND
+#ifdef TELEPATHY_FOUND
     m_accountManager =  Tp::AccountManager::create(Tp::AccountFactory::create(QDBusConnection::sessionBus(), Tp::Account::FeatureCore));
     connect(m_accountManager->becomeReady(), SIGNAL(finished(Tp::PendingOperation*)), SLOT(slotFinished(Tp::PendingOperation*)));
 
@@ -71,7 +71,7 @@ void IMQDBus::updateStatusMessage(const QString &im, const QString &statusMessag
     if (im == IM_PIDGIN) {
         usePidgin(statusMessage);
     }
-#if TELEPATHY_FOUND
+#ifdef TELEPATHY_FOUND
     if (im == IM_TELEPATHY) {
         useTelepathy(statusMessage);
     }
@@ -190,7 +190,7 @@ void IMQDBus::usePidgin(const QString &statusMessage)
     }
 }
 
-#if TELEPATHY_FOUND
+#ifdef TELEPATHY_FOUND
 void IMQDBus::useTelepathy(const QString &statusMessage)
 {
     if (m_accountManager->isReady()) {
@@ -235,7 +235,7 @@ QStringList IMQDBus::scanForIMs()
     if (QDBusConnection::sessionBus().interface()->isServiceRegistered(QLatin1String("im.pidgin.purple.PurpleService")).value()) {
         ims << IM_PIDGIN;
     }
-#if TELEPATHY_FOUND
+#ifdef TELEPATHY_FOUND
     if (QDBusConnection::sessionBus().interface()->isServiceRegistered(QLatin1String("org.freedesktop.Telepathy.AccountManager")).value()) {
         ims << IM_TELEPATHY;
     }
diff --git a/plugins/imstatus/imqdbus.h b/plugins/imstatus/imqdbus.h
index 6546e20..c9564df 100644
--- a/plugins/imstatus/imqdbus.h
+++ b/plugins/imstatus/imqdbus.h
@@ -27,9 +27,7 @@
 #include <QObject>
 #include <QString>
 
-#include "config-imstatus.h"
-
-#if TELEPATHY_FOUND
+#ifdef TELEPATHY_FOUND
 #include <TelepathyQt/Types>
 
 namespace Tp
@@ -58,7 +56,7 @@ private:
     void useSkype(const QString &statusMessage);
     void usePidgin(const QString &statusMessage);
 
-#if TELEPATHY_FOUND
+#ifdef TELEPATHY_FOUND
 private Q_SLOTS:
     void slotFinished(Tp::PendingOperation *po);
 
