diff -ruN ../akonadi-googledata-9999.orig/calendar/gcalresource.cpp ./calendar/gcalresource.cpp
--- ../akonadi-googledata-9999.orig/calendar/gcalresource.cpp	2010-08-10 23:28:59.968928637 +0200
+++ ./calendar/gcalresource.cpp	2010-08-10 23:30:55.608085396 +0200
@@ -89,6 +89,16 @@
 	Settings::self()->writeConfig();
 }
 
+void GCalResource::fixOffset(KDateTime &date, bool toGoogle)
+{
+	// fix for issue #57: GCal events end on 00:00:00 on the next day
+	if(date.isDateOnly()) {
+		if(toGoogle)
+			date = date.addDays(1);
+		else
+			date = date.addDays(-1);
+	}
+}
 
 void GCalResource::retrieveCollections()
 {
@@ -191,6 +201,7 @@
 		start = start.fromString(temp, KDateTime::ISODate);
 		temp = gcal_event_get_end(event);
 		end = end.fromString(temp, KDateTime::ISODate);
+		fixOffset(end);
 		kevent->setDtStart(start);
 		kevent->setDtEnd(end);
 
@@ -361,6 +372,7 @@
 			start = start.fromString(temp, KDateTime::ISODate);
 			temp = gcal_event_get_end(event);
 			end = end.fromString(temp, KDateTime::ISODate);
+			fixOffset(end);
 			kevent->setDtStart(start);
 			kevent->setDtEnd(end);
 
@@ -507,6 +519,7 @@
 	gcal_event_set_start(event, t_byte.data());
 
 	time = kevent->dtEnd();
+	fixOffset(time, true);
 	temp = time.toString(KDateTime::ISODate);
 	t_byte = temp.toUtf8();
 	gcal_event_set_end(event, t_byte.data());
@@ -607,6 +620,7 @@
 	gcal_event_set_start(event, t_byte.data());
 
 	time = kevent->dtEnd();
+	fixOffset(time, true);
 	temp = time.toString(KDateTime::ISODate);
 	t_byte = temp.toUtf8();
 	gcal_event_set_end(event, t_byte.data());
diff -ruN ../akonadi-googledata-9999.orig/calendar/gcalresource.h ./calendar/gcalresource.h
--- ../akonadi-googledata-9999.orig/calendar/gcalresource.h	2010-08-10 23:28:59.968928637 +0200
+++ ./calendar/gcalresource.h	2010-08-10 23:30:55.611418168 +0200
@@ -57,6 +57,9 @@
 	/* Event itens update lists */
 	Akonadi::Item::List pending;
 	Akonadi::Item::List deleted;
+   
+private:  
+	void fixOffset(KDateTime &date, bool toGoogle=false);
 
 };
 
