--- filters/kword/msword-odf/graphicshandler.cpp	2009-02-25 23:43:39.000000000 +0200
+++ filters/kword/msword-odf/graphicshandler.cpp	2009-03-07 13:23:26.881172525 +0200
@@ -66,7 +66,7 @@
         wvWare::U8* buf = new wvWare::U8[IMG_BUF_SIZE];
         size_t n = reader.read( buf, qMin( len, IMG_BUF_SIZE ) );
         Q_LONG n1 = m_store->write( (const char*)buf, n );
-        kDebug(30513) << "n=" << n << ", n1=" << n1 << "; buf contains " << (int) buf;
+        kDebug(30513) << "n=" << n << ", n1=" << n1 << "; buf contains " << (void*) buf;
         len -= n;
         delete [] buf;
         //error checking
@@ -109,7 +109,7 @@
         }
         //size_t n = reader.read( buf, qMin( len, IMG_BUF_SIZE ) );
         Q_LONG n1 = m_store->write( (const char*)buf, n );
-        kDebug(30513) << "n=" << n << ", n1=" << n1 << "; buf contains " << (int) buf;
+        kDebug(30513) << "n=" << n << ", n1=" << n1 << "; buf contains " << (void*) buf;
         len -= n;
         delete [] buf;
         //error checking
