--- kalzium-9999/KalziumConfigureChecks.cmake  2012-10-10 10:58:33.140864269 +0800
+++ kalzium-9999/KalziumConfigureChecks.cmake.new      2012-10-10 11:07:46.266956978 +0800
@@ -6,6 +6,8 @@
 macro_optional_find_package(OpenBabel2)
 macro_optional_find_package(Eigen2)
 macro_optional_find_package(Avogadro 1.0.0 NO_MODULE)
+
+find_package(PkgConfig)
 pkg_check_modules(CHEMICAL_MIME_DATA chemical-mime-data)
 
 check_include_files(ieeefp.h HAVE_IEEEFP_H)

