--- kalzium-4.2.1/work/kalzium-4.2.1/kalzium/libavogadro-kalzium/src/CMakeLists.txt	2009-02-26 09:17:16.000000000 -0500
+++ /root/CMakeLists.txt	2009-03-11 18:19:48.000000000 -0400
@@ -15,10 +15,12 @@
   ${CMAKE_CURRENT_BINARY_DIR}/config.h
 )
 
-include(${QT_USE_FILE})
+# Ensure the Avogadro include directory is always first
+get_directory_property(tmp_include_dirs INCLUDE_DIRECTORIES)
+set_directory_properties(PROPERTIES INCLUDE_DIRECTORIES
+	"${libavogadro-kalzium_SOURCE_DIR}/include;${tmp_include_dirs}")
 
 include_directories(
-  ${libavogadro-kalzium_SOURCE_DIR}/include
   ${CMAKE_SOURCE_DIR}
   ${CMAKE_CURRENT_BINARY_DIR}
   ${CMAKE_CURRENT_BINARY_DIR}/..
@@ -26,6 +28,8 @@
   ${EIGEN_INCLUDE_DIR}
 )
 
+include(${QT_USE_FILE})
+
 # Need to build with visibility enabled if KDE is
 if(__KDE_HAVE_GCC_VISIBILITY)
   add_definitions(-DHAVE_GCC_VISIBILITY)
