--- kstars/kstars/indi/CMakeLists.txt	2008-08-11 01:01:07.000000000 -0400
+++ kstars/kstars/indi/CMakeLists.txt	2008-08-11 01:04:35.000000000 -0400
@@ -1,4 +1,4 @@
-KDE4_NO_ENABLE_FINAL(indi)
+KDE4_NO_ENABLE_FINAL(indi)
 
 include_directories( ${CMAKE_CURRENT_BINARY_DIR}/../.. )
 include_directories( ${CMAKE_SOURCE_DIR}/kstars/kstars/indi/libs)
@@ -209,10 +209,10 @@
 #TODO fix for windows.
 if(UNIX)
 file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/make_lx200generic_symlink.cmake 
-"exec_program(${CMAKE_COMMAND} ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic ${BIN_INSTALL_DIR}/indi_lx200classic)\n
-exec_program(${CMAKE_COMMAND} ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic ${BIN_INSTALL_DIR}/indi_lx200autostar)\n 
-exec_program(${CMAKE_COMMAND} ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic ${BIN_INSTALL_DIR}/indi_lx200_16)\n
-exec_program(${CMAKE_COMMAND} ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic ${BIN_INSTALL_DIR}/indi_lx200gps)\n
+"exec_program(${CMAKE_COMMAND} ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic \"$ENV{D}/${BIN_INSTALL_DIR}/indi_lx200classic\")\n
+exec_program(${CMAKE_COMMAND} ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic \"$ENV{D}/${BIN_INSTALL_DIR}/indi_lx200autostar\")\n 
+exec_program(${CMAKE_COMMAND} ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic \"$ENV{D}/${BIN_INSTALL_DIR}/indi_lx200_16\")\n
+exec_program(${CMAKE_COMMAND} ARGS -E create_symlink ${BIN_INSTALL_DIR}/indi_lx200generic \"$ENV{D}/${BIN_INSTALL_DIR}/indi_lx200gps\")\n
 ")
 set_target_properties(indi_lx200generic PROPERTIES POST_INSTALL_SCRIPT ${CMAKE_CURRENT_BINARY_DIR}/make_lx200generic_symlink.cmake)
 endif(UNIX)
