Desc:	Fix compilation on Solaris with some defines.
Author:	Heiko Przybyl <zuxez@cs.tu-berlin.de>
--- kstars-4.3.2/kstars/kstars/htmesh/SpatialGeneral.h.orig	2009-10-16 17:32:00.241610236 +0200
+++ kstars-4.3.2/kstars/kstars/htmesh/SpatialGeneral.h	2009-10-16 17:32:36.949967839 +0200
@@ -80,7 +80,7 @@ typedef unsigned long long	uint64;
 
 // emulate the standard bool type where not supported by compiler
 
-#  if !defined(SXGENERAL_H) && !defined(__sgi) && !defined(__linux__) && !defined(_WIN32) && !defined(__FreeBSD__) && !defined(__NetBSD__) && !defined(__OpenBSD__)
+#  if !defined(SXGENERAL_H) && !defined(__sgi) && !defined(__linux__) && !defined(_WIN32) && !defined(__FreeBSD__) && !defined(__NetBSD__) && !defined(__OpenBSD__) && !defined(__sun)
 #    ifdef __unix__
 /*
  * The following ifndef must ALWAYS be present since C++ may use
--- kstars-4.3.2/kstars/kstars/skycomponents/starcomponent.cpp.orig	2009-10-16 17:43:41.053792704 +0200
+++ kstars-4.3.2/kstars/kstars/skycomponents/starcomponent.cpp	2009-10-16 17:44:40.365860457 +0200
@@ -44,6 +44,10 @@
 #include <sys/endian.h>
 #define bswap_16(x) bswap16(x)
 #define bswap_32(x) bswap32(x)
+#elif defined(__sun)
+#include <sys/byteorder.h>
+#define bswap_16(x) BSWAP_16(x)
+#define bswap_32(x) BSWAP_32(x)
 #else
 #include <byteswap.h>
 #endif
