Index: lib/konq/CMakeLists.txt
===================================================================
--- lib/konq/CMakeLists.txt	(revision 1172114)
+++ lib/konq/CMakeLists.txt	(working copy)
@@ -58,7 +58,9 @@
  konq_dndpopupmenuplugin.h
  knewmenu.h
  konq_menuactions.h
- # konq_copytomenu.h - anyone needs it?
+ konq_copytomenu.h 
+ konq_popupmenuinformation.h
+ konq_statusbarmessagelabel.h
  konq_operations.h
  libkonq_export.h
  konq_events.h
