diff -ru libkonq/apps/lib/konq/CMakeLists.txt libkonqfixed/apps/lib/konq/CMakeLists.txt
--- libkonq/apps/lib/konq/CMakeLists.txt	2008-04-27 19:08:19.000000000 +0000
+++ libkonqfixed/apps/lib/konq/CMakeLists.txt	2008-04-27 19:09:31.000000000 +0000
@@ -42,7 +42,8 @@
  konq_popupmenu.h
  knewmenu.h
  konq_menuactions.h
- # konq_copytomenu.h - anyone needs it?
+ konq_copytomenu.h 
+ konq_popupmenuinformation.h
  konq_operations.h
  libkonq_export.h
  konq_events.h
