These headers are required to compile dolphin.

diff --git a/lib/konq/CMakeLists.txt b/lib/konq/CMakeLists.txt
index 8ecbfa9..9fcbb3c 100644
--- a/lib/konq/CMakeLists.txt
+++ b/lib/konq/CMakeLists.txt
@@ -59,7 +59,7 @@ install( FILES
  konq_dndpopupmenuplugin.h
  knewmenu.h
  konq_menuactions.h
- # konq_copytomenu.h - anyone needs it?
+ konq_copytomenu.h
  konq_operations.h
  libkonq_export.h
  konq_events.h
@@ -69,6 +69,7 @@ install( FILES
  kversioncontrolplugin2.h
  konq_historyprovider.h
  konq_historyentry.h
+ konq_statusbarmessagelabel.h
 DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Devel
 )
 install( FILES konqpopupmenuplugin.desktop konqdndpopupmenuplugin.desktop  DESTINATION  ${SERVICETYPES_INSTALL_DIR} )
