diff -ru ../old/configure.in ./configure.in
--- ../old/configure.in	2009-04-21 18:28:32.000000000 +0200
+++ ./configure.in	2009-06-25 17:04:11.000000000 +0200
@@ -1593,7 +1594,7 @@
 if test "x$RUBY" != "x"  -a "x$with_ruby" != "xno" ; then
     RUBY_ARCHDIR=`$RUBY -e 'require "rbconfig.rb"; puts Config::expand("\$(archdir)")'`
     RUBY_CFLAGS="-I$RUBY_ARCHDIR"
-    RUBY_LDFLAGS="-lruby"
+    RUBY_LDFLAGS="-lruby18"
     save_CPPFLAGS="$CPPFLAGS"
     save_LIBS="$LIBS"
     CPPFLAGS="$CPPFLAGS $RUBY_CFLAGS"
