Index: trunk/KDE/kdeplasma-addons/libs/lancelot-datamodels/MessagesKmail_p.h
===================================================================
--- trunk/KDE/kdeplasma-addons/libs/lancelot-datamodels/MessagesKmail_p.h	(revision 1162199)
+++ trunk/KDE/kdeplasma-addons/libs/lancelot-datamodels/MessagesKmail_p.h	(revision 1162200)
@@ -21,11 +21,14 @@
 #include <QHash>
 #include <KIcon>
 #include <KJob>
+
+#include "config-lancelot-datamodels.h"
+
+#ifdef LANCELOT_DATAMODELS_HAS_PIMLIBS
 #include <Akonadi/Entity>
 #include <Akonadi/Collection>
+#endif
 
-#include "config-lancelot-datamodels.h"
-
 namespace Lancelot {
 namespace Models {
 
