diff --git a/libs/lancelot-datamodels/MessagesKmail.cpp b/libs/lancelot-datamodels/MessagesKmail.cpp
index 8fa89e0..f56203d 100644
--- a/libs/lancelot-datamodels/MessagesKmail.cpp
+++ b/libs/lancelot-datamodels/MessagesKmail.cpp
@@ -35,10 +35,10 @@
 #ifndef LANCELOT_DATAMODELS_HAS_PIMLIBS
 
 // just in case messages:
-I18N_NOOP("Unread messages");
+/*I18N_NOOP("Unread messages");
 I18N_NOOP("Unable to find Kontact");
 I18N_NOOP("Start Akonadi server");
-I18N_NOOP("Akonadi server is not running");
+I18N_NOOP("Akonadi server is not running");*/
 
 #warning "Pimlibs are not present"
 
diff --git a/libs/lancelot-datamodels/MessagesKmail.h b/libs/lancelot-datamodels/MessagesKmail.h
index 8db2a1c..3dfbd6f 100644
--- a/libs/lancelot-datamodels/MessagesKmail.h
+++ b/libs/lancelot-datamodels/MessagesKmail.h
@@ -32,6 +32,7 @@ namespace Models {
  */
 class LANCELOT_EXPORT MessagesKmail : public BaseModel {
     Q_OBJECT
+
 public:
     /**
      * Creates a new MessagesKmail instance
@@ -52,11 +53,13 @@ protected:
 
     void load();
 
+#ifdef LANCELOT_DATAMODELS_HAS_PIMLIBS
     L_Override QString selfShortTitle() const;
 
 protected Q_SLOTS:
     void updateLater();
     void update();
+#endif
 
 private:
     class Private;
diff --git a/runners/contacts/contactsrunner.cpp b/runners/contacts/contactsrunner.cpp
index 2b5f3ec..aa64761 100644
--- a/runners/contacts/contactsrunner.cpp
+++ b/runners/contacts/contactsrunner.cpp
@@ -23,7 +23,10 @@
 #include <KLocale>
 #include <KIcon>
 #include <KToolInvocation>
+
+#ifdef LANCELOT_DATAMODELS_HAS_PIMLIBS
 #include <akonadi/contact/contactdefaultactions.h>
+#endif
 /*#include <KMimeType>
 
 */
diff --git a/runners/events/events_config.h b/runners/events/events_config.h
index f73b22b..8bb0148 100644
--- a/runners/events/events_config.h
+++ b/runners/events/events_config.h
@@ -26,7 +26,9 @@
 
 //KDE-Includes
 #include <KCModule>
+#ifdef LANCELOT_DATAMODELS_HAS_PIMLIBS
 #include <Akonadi/Collection>
+#endif
 
 //Qt
 
