Index: applets/CMakeLists.txt
===================================================================
--- applets/CMakeLists.txt	(revision 1122904)
+++ applets/CMakeLists.txt	(working copy)
@@ -16,6 +16,7 @@
 add_subdirectory(calculator)
 add_subdirectory(charselect)
 add_subdirectory(comic)
+add_subdirectory(dict)
 add_subdirectory(fifteenPuzzle)
 add_subdirectory(fileWatcher)
 add_subdirectory(frame)
@@ -30,14 +31,21 @@
 add_subdirectory(microblog)
 add_subdirectory(notes)
 add_subdirectory(nowplaying)
+
 if(LIBATTICA_FOUND)
   add_subdirectory(community)
   add_subdirectory(social-news)
 endif(LIBATTICA_FOUND)
+
 add_subdirectory(pastebin)
 add_subdirectory(previewer)
 add_subdirectory(rememberthemilk)
-add_subdirectory(rssnow)
+
+if(KDEPIMLIBS_FOUND)
+  add_subdirectory(news)
+  add_subdirectory(rssnow)
+endif(KDEPIMLIBS_FOUND)
+
 add_subdirectory(spellcheck)
 add_subdirectory(showdashboard)
 add_subdirectory(timer)
@@ -48,11 +56,6 @@
 add_subdirectory(weatherstation)
 add_subdirectory(weather)
 
-if(QT_QTWEBKIT_FOUND)
-  add_subdirectory(dict)
-  add_subdirectory(news)
-endif(QT_QTWEBKIT_FOUND)
-
 if(PHONON_FOUND)
   add_subdirectory(mediaplayer)
 endif(PHONON_FOUND)
Index: libs/lancelot-datamodels/CMakeLists.txt
===================================================================
--- libs/lancelot-datamodels/CMakeLists.txt	(revision 1122904)
+++ libs/lancelot-datamodels/CMakeLists.txt	(working copy)
@@ -23,10 +23,8 @@
 
    set (ADDITIONAL_LINK_LIBRARIES
         ${ADDITIONAL_LINK_LIBRARIES}
-        ${KDE4_AKONADI_LIBS}
-        )
-
-
+        ${KDEPIMLIBS_AKONADI_LIBS}
+   )
 endif(KDEPIMLIBS_FOUND)
 
 macro_log_feature(LANCELOT_DATAMODELS_HAS_PIMLIBS
Index: runners/CMakeLists.txt
===================================================================
--- runners/CMakeLists.txt	(revision 1122904)
+++ runners/CMakeLists.txt	(working copy)
@@ -9,10 +9,9 @@
 add_subdirectory(characters)
 
 if(KDEPIMLIBS_FOUND)
-add_subdirectory(contacts)
+   add_subdirectory(contacts)
 endif(KDEPIMLIBS_FOUND)
 
 if(NOT WIN32)
- add_subdirectory(konsolesessions)
+   add_subdirectory(konsolesessions)
 endif(NOT WIN32)
-
Index: runners/contacts/CMakeLists.txt
===================================================================
--- runners/contacts/CMakeLists.txt	(revision 1122904)
+++ runners/contacts/CMakeLists.txt	(working copy)
@@ -5,7 +5,7 @@
 
 kde4_add_plugin(krunner_contacts ${krunner_contacts_SRCS})
 target_link_libraries(krunner_contacts ${KDE4_KIO_LIBS} ${KDE4_PLASMA_LIBS}
-${KDE4_KABC_LIBS})
+${KDEPIMLIBS_KABC_LIBS})
 
 install(TARGETS krunner_contacts DESTINATION ${PLUGIN_INSTALL_DIR} )
 
