diff -ruN kdeplasma-addons-4.7.3.orig/applets/CMakeLists.txt kdeplasma-addons-4.7.3/applets/CMakeLists.txt
--- kdeplasma-addons-4.7.3.orig/applets/CMakeLists.txt	2011-10-02 15:26:40.000000000 +0200
+++ kdeplasma-addons-4.7.3/applets/CMakeLists.txt	2011-11-20 18:05:27.833155727 +0100
@@ -26,7 +26,6 @@
 add_subdirectory(kolourpicker)
 add_subdirectory(konqprofiles)
 add_subdirectory(konsoleprofiles)
-add_subdirectory(knowledgebase)
 add_subdirectory(life)
 add_subdirectory(luna)
 add_subdirectory(magnifique)
@@ -34,6 +33,7 @@
 add_subdirectory(notes)
 add_subdirectory(nowplaying)
 if(LIBATTICA_FOUND)
+  add_subdirectory(knowledgebase)
   add_subdirectory(community)
   add_subdirectory(social-news)
 endif(LIBATTICA_FOUND)
