--- libs/lancelot-datamodels/BaseModel.cpp	2010/06/24 19:25:17	1142379
+++ libs/lancelot-datamodels/BaseModel.cpp	2010/07/29 09:07:25	1156567
@@ -165,7 +165,7 @@
 {
     const KService::Ptr service = KService::serviceByStorageId(serviceName);
 
-    qDebug() << "BaseModel::addService Applic:" << service->isValid() << " " << serviceName;
+    // qDebug() << "BaseModel::addService Applic:" << service->isValid() << " " << serviceName;
 
     return addService(service);
 }
