commit b0c70d7fd866efb628036f43978bc791e0485f90
Author: Aaron Seigo <aseigo@kde.org>
Date:   Tue Nov 22 14:07:59 2011 +0100

    remove outdated using statements

diff --git a/applets/icontasks/taskitemlayout.h b/applets/icontasks/taskitemlayout.h
index 9ebf82f..64142e1 100644
--- a/applets/icontasks/taskitemlayout.h
+++ b/applets/icontasks/taskitemlayout.h
@@ -32,9 +32,6 @@ class TaskGroupItem;
 class AbstractTaskItem;
 class LauncherSeparator;
 
-using TaskManager::StartupPtr;
-using TaskManager::TaskPtr;
-
 /**
  * A Layout for the expanded group
  */
diff --git a/applets/icontasks/tasks.h b/applets/icontasks/tasks.h
index e7bab6b..bf8a665 100644
--- a/applets/icontasks/tasks.h
+++ b/applets/icontasks/tasks.h
@@ -54,15 +54,6 @@ namespace TaskManager
 class GroupManager;
 } // namespace TaskManager
 
-using TaskManager::StartupPtr;
-using TaskManager::TaskPtr;
-using TaskManager::StartupPtr;
-using TaskManager::GroupPtr;
-using TaskManager::AbstractGroupableItem;
-using TaskManager::TaskItem;
-using TaskManager::TaskGroup;
-using TaskManager::LauncherItem;
-
 class TaskGroupItem;
 class GroupManager;
 
