commit e089568a73ec0fa9f8e9ec61c9853b1905d3a1d7
Author: Ben Cooksley <bcooksley@kde.org>
Date:   Mon Nov 28 08:14:54 2011 +1300

    Fix build of Marble wallpaper
    REVIEW: 103243

diff --git a/wallpapers/marble/marble.cpp b/wallpapers/marble/marble.cpp
index 3cecaae..988a8b6 100644
--- a/wallpapers/marble/marble.cpp
+++ b/wallpapers/marble/marble.cpp
@@ -137,7 +137,6 @@ QWidget *MarbleWallpaper::createConfigurationInterface(QWidget *parent)
     m_ui.showPlacemarks->setChecked(m_showPlacemarks);
 
     MapThemeManager themeManager;
-    themeManager.updateMapThemeModel();
     // FIXME: Going manually through the model is ugly as hell, but plugging the
     //        model into the view didn't work for me
     for (int i = 0; i < themeManager.mapThemeModel()->rowCount(); i++) {
