Index: applets/CMakeLists.txt
===================================================================
--- applets/CMakeLists.txt	(revision 1122757)
+++ applets/CMakeLists.txt	(working copy)
@@ -15,6 +15,7 @@
 add_subdirectory(calculator)
 add_subdirectory(charselect)
 add_subdirectory(comic)
+add_subdirectory(dict)
 add_subdirectory(fifteenPuzzle)
 add_subdirectory(fileWatcher)
 add_subdirectory(frame)
@@ -29,14 +30,21 @@
 add_subdirectory(microblog)
 add_subdirectory(notes)
 add_subdirectory(nowplaying)
+
 if(LIBATTICA_FOUND)
   add_subdirectory(opendesktop)
   add_subdirectory(opendesktop-activities)
 endif(LIBATTICA_FOUND)
+
 add_subdirectory(pastebin)
 add_subdirectory(previewer)
 add_subdirectory(rememberthemilk)
-add_subdirectory(rssnow)
+
+if(KDEPIMLIBS_FOUND)
+  add_subdirectory(news)
+  add_subdirectory(rssnow)
+endif(KDEPIMLIBS_FOUND)
+
 add_subdirectory(spellcheck)
 add_subdirectory(showdashboard)
 add_subdirectory(timer)
@@ -47,11 +55,6 @@
 add_subdirectory(weatherstation)
 add_subdirectory(weather)
 
-if(QT_QTWEBKIT_FOUND)
-  add_subdirectory(dict)
-  add_subdirectory(news)
-endif(QT_QTWEBKIT_FOUND)
-
 if(PHONON_FOUND)
   add_subdirectory(mediaplayer)
 endif(PHONON_FOUND)
Index: runners/CMakeLists.txt
===================================================================
--- runners/CMakeLists.txt	(revision 1122757)
+++ runners/CMakeLists.txt	(working copy)
@@ -8,10 +8,9 @@
 add_subdirectory(spellchecker)
 
 if(KDEPIMLIBS_FOUND)
-add_subdirectory(contacts)
+   add_subdirectory(contacts)
 endif(KDEPIMLIBS_FOUND)
 
 if(NOT WIN32)
- add_subdirectory(konsolesessions)
+   add_subdirectory(konsolesessions)
 endif(NOT WIN32)
-
Index: runners/contacts/CMakeLists.txt
===================================================================
--- runners/contacts/CMakeLists.txt	(revision 1122757)
+++ runners/contacts/CMakeLists.txt	(working copy)
@@ -5,7 +5,7 @@
 
 kde4_add_plugin(krunner_contacts ${krunner_contacts_SRCS})
 target_link_libraries(krunner_contacts ${KDE4_KIO_LIBS} ${KDE4_PLASMA_LIBS}
-${KDE4_KABC_LIBS})
+${KDEPIMLIBS_KABC_LIBS})
 
 install(TARGETS krunner_contacts DESTINATION ${PLUGIN_INSTALL_DIR} )
 
