Index: applets/CMakeLists.txt
===================================================================
--- applets/CMakeLists.txt	(revision 1172136)
+++ applets/CMakeLists.txt	(working copy)
@@ -16,6 +16,7 @@
 add_subdirectory(calculator)
 add_subdirectory(charselect)
 add_subdirectory(comic)
+add_subdirectory(dict)
 add_subdirectory(fifteenPuzzle)
 add_subdirectory(fileWatcher)
 add_subdirectory(frame)
@@ -30,14 +31,21 @@
 add_subdirectory(microblog)
 add_subdirectory(notes)
 add_subdirectory(nowplaying)
+
 if(LIBATTICA_FOUND)
   add_subdirectory(community)
   add_subdirectory(social-news)
 endif(LIBATTICA_FOUND)
+
 add_subdirectory(pastebin)
 add_subdirectory(previewer)
 add_subdirectory(rememberthemilk)
-add_subdirectory(rssnow)
+
+if(KDEPIMLIBS_FOUND)
+  add_subdirectory(news)
+  add_subdirectory(rssnow)
+endif(KDEPIMLIBS_FOUND)
+
 add_subdirectory(spellcheck)
 add_subdirectory(showdashboard)
 add_subdirectory(timer)
@@ -48,11 +56,6 @@
 add_subdirectory(weatherstation)
 add_subdirectory(weather)
 
-if(QT_QTWEBKIT_FOUND)
-  add_subdirectory(dict)
-  add_subdirectory(news)
-endif(QT_QTWEBKIT_FOUND)
-
 if(PHONON_FOUND)
   add_subdirectory(mediaplayer)
 endif(PHONON_FOUND)
