diff -r -u kdebase-workspace-4.4.2-orig/ConfigureChecks.cmake kdebase-workspace-4.4.2/ConfigureChecks.cmake
--- kdebase-workspace-4.4.2-orig/ConfigureChecks.cmake	2009-11-21 10:44:36.000000000 +0100
+++ kdebase-workspace-4.4.2/ConfigureChecks.cmake	2010-04-02 00:04:32.000000000 +0200
@@ -71,7 +71,14 @@
 macro_bool_to_01(X11_Xcursor_FOUND HAVE_XCURSOR) # many uses
 macro_bool_to_01(X11_Xdamage_FOUND HAVE_XDAMAGE) # kwin
 macro_bool_to_01(X11_Xfixes_FOUND HAVE_XFIXES) # klipper, kicker, kwin
+
+if(NOT WITH_X11_Xinerama)
+  set(X11_Xinerama_FOUND FALSE)
+  MESSAGE(STATUS "Disabling Xinerama as requested on commandline.")
+endif(NOT WITH_X11_Xinerama)
+
 macro_bool_to_01(X11_Xinerama_FOUND HAVE_XINERAMA)
+
 macro_bool_to_01(X11_Xrandr_FOUND HAVE_XRANDR) # kwin
 macro_bool_to_01(X11_Xrender_FOUND HAVE_XRENDER) # kcontrol/style, kicker
 macro_bool_to_01(X11_xf86misc_FOUND HAVE_XF86MISC) # kdesktop and kcontrol/lock
