diff -ruN ksplash-4.9.95.orig/ksplash/ksplashx/CMakeLists.txt ksplash-4.9.95/ksplash/ksplashx/CMakeLists.txt
--- ksplash-4.9.95.orig/ksplash/ksplashx/CMakeLists.txt	2012-12-24 11:33:22.326087112 +0100
+++ ksplash-4.9.95/ksplash/ksplashx/CMakeLists.txt	2012-12-24 11:34:53.649089265 +0100
@@ -2,7 +2,6 @@
 
 add_subdirectory( scale )
 add_subdirectory( utils )
-add_subdirectory( themes )
 
 include_directories(${PNG_INCLUDE_DIR}  # Deprecated in CMake 2.8.7
                     ${PNG_INCLUDE_DIRS} # Added in CMake 2.8.7
