diff -ru ../system-config-printer-1.1.16/configure.in ./configure.in
--- ../system-config-printer-1.1.16/configure.in	2009-12-22 15:44:19.000000000 +0100
+++ ./configure.in	2010-01-16 05:48:52.458480903 +0100
@@ -1,6 +1,6 @@
 AC_INIT(system-config-printer, 1.1.16)
 AC_CONFIG_SRCDIR(system-config-printer.py)
-AM_INIT_AUTOMAKE([dist-xz dist-bzip2 subdir-objects 1.6])
+AM_INIT_AUTOMAKE([dist-bzip2 subdir-objects 1.6])
 IT_PROG_INTLTOOL
 AM_GNU_GETTEXT([external])
 AM_PATH_PYTHON
diff -ru ../system-config-printer-1.1.16/Makefile.am ./Makefile.am
--- ../system-config-printer-1.1.16/Makefile.am	2009-12-22 15:44:19.000000000 +0100
+++ ./Makefile.am	2010-01-16 05:59:13.766945612 +0100
@@ -71,89 +71,20 @@
 	rm -f .stamp-distutils-in-builddir
 
 nobase_pkgdata_SCRIPTS=				\
-	my-default-printer.py			\
-	pysmb.py				\
-	system-config-printer.py		\
-	troubleshoot/__init__.py		\
-	applet.py
+	pysmb.py
 
 nobase_pkgdata_DATA=					\
-	AdvancedServerSettings.py			\
-	asyncconn.py					\
-	asyncipp.py					\
-	asyncpk0.py					\
-	asyncpk1.py					\
-	authconn.py					\
 	config.py					\
-	cupspk.py					\
 	debug.py					\
-	errordialogs.py					\
-	HIG.py						\
-	firewall.py					\
-	glade.py					\
-	GroupsPane.py					\
-	GroupsPaneModel.py				\
-	gtkinklevel.py					\
-	gtkspinner.py					\
 	installpackage.py				\
-	jobviewer.py					\
 	monitor.py					\
-	options.py					\
-	optionwidgets.py				\
 	PhysicalDevice.py				\
 	ppdippstr.py					\
 	probe_printer.py				\
-	gtk_label_autowrap.py				\
 	SearchCriterion.py				\
 	smburi.py					\
 	statereason.py					\
-	timedops.py					\
-	ToolbarSearchEntry.py				\
-	userdefault.py					\
-	XmlHelper.py					\
-	glade/AboutDialog.glade				\
-	glade/ConnectDialog.glade			\
-	glade/ConnectingDialog.glade			\
-	glade/InstallDialog.glade			\
-	glade/JobsWindow.glade				\
-	glade/NewPrinterName.glade			\
-	glade/NewPrinterWindow.glade			\
-	glade/PrinterPropertiesDialog.glade		\
-	glade/PrintersWindow.glade			\
-	glade/ServerSettingsDialog.glade		\
-	glade/SMBBrowseDialog.glade			\
-	glade/statusicon_popupmenu.glade		\
-	glade/WaitWindow.glade				\
-	icons/i-network-printer.png			\
-	troubleshoot/base.py				\
-	troubleshoot/CheckLocalServerPublishing.py	\
-	troubleshoot/CheckNetworkServerSanity.py	\
-	troubleshoot/CheckPPDSanity.py			\
-	troubleshoot/CheckPrinterSanity.py		\
-	troubleshoot/CheckUSBPermissions.py		\
-	troubleshoot/ChooseNetworkPrinter.py		\
-	troubleshoot/ChoosePrinter.py			\
-	troubleshoot/DeviceListed.py			\
-	troubleshoot/ErrorLogCheckpoint.py		\
-	troubleshoot/ErrorLogFetch.py			\
-	troubleshoot/ErrorLogParse.py			\
-	troubleshoot/Locale.py				\
-	troubleshoot/LocalOrRemote.py			\
-	troubleshoot/NetworkCUPSPrinterShared.py	\
-	troubleshoot/PrinterStateReasons.py		\
-	troubleshoot/PrintTestPage.py			\
-	troubleshoot/QueueNotEnabled.py			\
-	troubleshoot/QueueRejectingJobs.py		\
-	troubleshoot/RemoteAddress.py			\
-	troubleshoot/SchedulerNotRunning.py		\
-	troubleshoot/ServerFirewalled.py		\
-	troubleshoot/Shrug.py				\
-	troubleshoot/Welcome.py
-
-bin_SCRIPTS=\
-	system-config-printer			\
-	system-config-printer-applet		\
-	my-default-printer
+	XmlHelper.py
 
 if UDEV_RULES
 udevrulesdir=/lib/udev/rules.d
@@ -168,61 +99,16 @@
 	udev/udev-add-printer
 endif
 
-man_MANS=					\
-	man/system-config-printer.1		\
-	man/system-config-printer-applet.1
-
 dbus_DATA =\
 	newprinternotification.conf
 dbusdir = $(sysconfdir)/dbus-1/system.d/
 
-desktop_DATA =\
-	system-config-printer.desktop \
-	print-applet.desktop \
-	manage-print-jobs.desktop \
-	my-default-printer.desktop
 desktopdir = $(datadir)/applications/
 autostartdir = $(sysconfdir)/xdg/autostart/
 
 DESKTOP_VENDOR=@DESKTOPVENDOR@
 DESKTOP_PREFIX=@DESKTOPPREFIX@
 
-install-desktopDATA: $(desktop_DATA)
-	mkdir -p $(DESTDIR)$(desktopdir)
-	mkdir -p $(DESTDIR)$(desktopdir)
-	desktop-file-install $(DESKTOP_VENDOR)	\
-	  --dir $(DESTDIR)$(desktopdir)		\
-	  --add-category System			\
-	  --add-category Settings		\
-	  --add-category HardwareSettings	\
-	  --add-category Printing		\
-	  --add-category GTK			\
-	  system-config-printer.desktop
-	desktop-file-install $(DESKTOP_VENDOR)	\
-	  --dir $(DESTDIR)$(desktopdir)		\
-	  --add-category System			\
-	  --add-category Monitor		\
-	  --add-category GTK			\
-	  manage-print-jobs.desktop
-	desktop-file-install $(DESKTOP_VENDOR)	\
-	  --dir $(DESTDIR)$(desktopdir)		\
-	  --add-category Settings		\
-	  --add-category HardwareSettings	\
-	  --add-category GTK			\
-	  my-default-printer.desktop
-	desktop-file-install $(DESKTOP_VENDOR)	\
-	  --dir $(DESTDIR)$(autostartdir)	\
-	  --add-category System			\
-	  --add-category Monitor		\
-	  --add-category GTK			\
-	  print-applet.desktop
-
-uninstall-desktopDATA:
-	rm -f $(DESTDIR)$(desktopdir)/$(DESKTOP_PREFIX)system-config-printer.desktop
-	rm -f $(DESTDIR)$(desktopdir)/$(DESKTOP_PREFIX)manage-print-jobs.desktop
-	rm -f $(DESTDIR)$(desktopdir)/$(DESKTOP_PREFIX)my-default-printer.desktop
-	rm -f $(DESTDIR)$(autostartdir)/$(DESKTOP_PREFIX)print-applet.desktop
-
 EXTRA_DIST=\
 	$(nobase_pkgdata_SCRIPTS) \
 	$(nobase_pkgdata_DATA) \
