Index: trunk/KDE/kdenetwork/kopete/protocols/jabber/jabberbasecontact.cpp
===================================================================
--- trunk/KDE/kdenetwork/kopete/protocols/jabber/jabberbasecontact.cpp	(revision 1062362)
+++ trunk/KDE/kdenetwork/kopete/protocols/jabber/jabberbasecontact.cpp	(revision 1062363)
@@ -335,7 +335,7 @@
 
 	if ( jid.resource().isEmpty () )
 	{
-		return jid.withResource ( account()->resourcePool()->bestResource ( jid ).name () );
+		jid.withResource ( account()->resourcePool()->bestResource ( jid ).name () );
 	}
 
 	return jid.full ();
