diff --git a/CMakeLists.txt b/CMakeLists.txt
index e85ecc9..540c201 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -275,7 +275,6 @@ if (NOT KDEPIM_BUILD_WITH_INSTALLED_LIB)
 add_subdirectory(libfollowupreminder)
 add_subdirectory(libsendlater)
 add_subdirectory(libgravatar)
-add_subdirectory(noteshared)
 add_subdirectory(libkdepim)
 add_subdirectory(calendarsupport)
 add_subdirectory(eventviews)
@@ -324,6 +323,7 @@ add_subdirectory(akregator)
 add_subdirectory(importwizard)
 add_subdirectory(kaddressbook)
 add_subdirectory(mboximporter)
+add_subdirectory(noteshared)
 add_subdirectory(knotes)
 add_subdirectory(ktnef)
 add_subdirectory(pimsettingexporter)
