--- generator/generator.qrc.orig	2009-03-22 16:38:28.000000000 +0100
+++ generator/generator.qrc	2009-03-22 16:38:40.000000000 +0100
@@ -7,7 +7,6 @@
 <file alias="typesystem_svg.txt">typesystem_svg.xml</file>
 <file alias="typesystem_network.txt">typesystem_network.xml</file>
 <file alias="typesystem_xml.txt">typesystem_xml.xml</file>
-<file alias="typesystem_phonon.txt">typesystem_phonon.xml</file>
 <file alias="typesystem_webkit.txt">typesystem_webkit.xml</file>
 <file alias="typesystem_xmlpatterns.txt">typesystem_xmlpatterns.xml</file>
 <file>parser/rpp/pp-qt-configuration</file>
--- qtbindings/qtbindings.pro.orig	2009-03-22 17:03:05.000000000 +0100
+++ qtbindings/qtbindings.pro	2009-03-22 17:03:15.000000000 +0100
@@ -6,7 +6,6 @@
 	  qtscript_sql \
 	  qtscript_svg \
 	  qtscript_xml \
-	  qtscript_phonon \
 	  qtscript_webkit \
 	  qtscript_xmlpatterns \
 	  qtscript_uitools \