--- orig/generator/parser/rpp/pp-macro-expander.h    2009-02-20 12:42:24.000000000 +0100
+++ generator/parser/rpp/pp-macro-expander.h 2009-05-01 12:22:51.000000000 +0200
@@ -15,6 +15,8 @@
 #ifndef PP_MACRO_EXPANDER_H
 #define PP_MACRO_EXPANDER_H

+#include <stdio.h>
+
 namespace rpp {

 struct pp_frame
--- orig/generator/parser/rpp/pp-engine-bits.h       2009-02-20 12:42:24.000000000 +0100
+++ generator/parser/rpp/pp-engine-bits.h    2009-05-01 12:16:23.000000000 +0200
@@ -15,6 +15,8 @@
 #ifndef PP_ENGINE_BITS_H
 #define PP_ENGINE_BITS_H

+#include <stdio.h>
+
 namespace rpp {

 inline std::string pp::fix_file_path(std::string const &filename) const
--- orig/generator/parser/rpp/pp-internal.h	2009-02-20 12:42:24.000000000 +0100
+++ generator/parser/rpp/pp-internal.h	2009-05-01 12:24:47.000000000 +0200
@@ -16,6 +16,7 @@
 #define PP_INTERNAL_H
 
 #include <algorithm>
+#include <stdio.h>
 
 namespace rpp {
 
