diff -u -r smoke-4.4.4.orig/CMakeLists.txt smoke-4.4.4/CMakeLists.txt
--- smoke-4.4.4.orig/CMakeLists.txt	2009-12-31 14:35:32.000000000 +0100
+++ smoke-4.4.4/CMakeLists.txt	2010-06-21 23:39:15.751408526 +0200
@@ -42,6 +42,8 @@
 endif(DEPEND_PACKAGE_Okular)
 
 # Look for Akonadi
+set( AKONADI_FOUND "no" )
+option(WITH_Akonadi "Search for Akonadi" OFF)
 our_macro_optional_depend_package(Akonadi "Akonadi")
 if(DEPEND_PACKAGE_Akonadi)
   macro_optional_find_package(Akonadi)
