diff -r -u kdebindings-4.4.0.orig/smoke/plasma/CMakeLists.txt kdebindings-4.4.0/smoke/plasma/CMakeLists.txt
--- kdebindings-4.4.0.orig/smoke/plasma/CMakeLists.txt	2010-02-04 10:48:34.000000000 +0100
+++ kdebindings-4.4.0/smoke/plasma/CMakeLists.txt	2010-02-20 01:01:49.000000000 +0100
@@ -41,15 +41,27 @@
 
 kde4_add_library(smokeplasma SHARED ${smokeplasma_LIB_SRCS})
 
-target_link_libraries(smokeplasma 
-    ${KDE4_KDEUI_LIBS}
-    ${KDE4_PLASMA_LIBS}
-    smokeqtcore
-    smokeqtgui
-    smokeqtwebkit
-    smokekdecore
-    smokekdeui
-)
+
+if(ENABLE_QTWEBKIT_SMOKE)
+    target_link_libraries(smokeplasma 
+        ${KDE4_KDEUI_LIBS}
+        ${KDE4_PLASMA_LIBS}
+        smokeqtcore
+        smokeqtgui
+        smokeqtwebkit
+        smokekdecore
+        smokekdeui
+    )
+else(ENABLE_QTWEBKIT_SMOKE)
+    target_link_libraries(smokeplasma 
+        ${KDE4_KDEUI_LIBS}
+        ${KDE4_PLASMA_LIBS}
+        smokeqtcore
+        smokeqtgui
+        smokekdecore
+        smokekdeui
+    )
+endif(ENABLE_QTWEBKIT_SMOKE)
 
 set_target_properties(smokeplasma PROPERTIES VERSION 3.0.0 SOVERSION 3 )
 install(TARGETS smokeplasma DESTINATION ${LIB_INSTALL_DIR} )
