--- kdebindings.orig/smoke/phonon/CMakeLists.txt
+++ kdebindings/smoke/phononCMakeLists.txt
@@ -12,25 +12,6 @@
 
 add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/smokedata.cpp
 	${CMAKE_CURRENT_BINARY_DIR}/x_1.cpp
-                 ${CMAKE_CURRENT_BINARY_DIR}/x_2.cpp 
-                ${CMAKE_CURRENT_BINARY_DIR}/x_3.cpp 
-                ${CMAKE_CURRENT_BINARY_DIR}/x_4.cpp 
-                ${CMAKE_CURRENT_BINARY_DIR}/x_5.cpp 
-                ${CMAKE_CURRENT_BINARY_DIR}/x_6.cpp 
-                ${CMAKE_CURRENT_BINARY_DIR}/x_7.cpp 
-                ${CMAKE_CURRENT_BINARY_DIR}/x_8.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_9.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_10.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_11.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_12.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_13.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_14.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_15.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_16.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_17.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_18.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_19.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_20.cpp
 
 	COMMAND ${PERL_EXECUTABLE} ARGS ${CMAKE_CURRENT_BINARY_DIR}/generate.pl
 	DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/generate.pl
@@ -43,25 +24,6 @@
 
 set(smokephonon_LIB_SRCS ${CMAKE_CURRENT_BINARY_DIR}/smokedata.cpp
 	${CMAKE_CURRENT_BINARY_DIR}/x_1.cpp
-		${CMAKE_CURRENT_BINARY_DIR}/x_2.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_3.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_4.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_5.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_6.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_7.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_8.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_9.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_10.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_11.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_12.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_13.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_14.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_15.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_16.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_17.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_18.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_19.cpp
-                ${CMAKE_CURRENT_BINARY_DIR}/x_20.cpp
 )
 
 IF(CMAKE_CXX_FLAGS MATCHES "-fvisibility")
