diff --git a/kdebase/workspace/cmake/modules/FindPAM.cmake b/kdebase/workspace/cmake/modules/FindPAM.cmake
index 3499836..cd1c2bb 100644
--- a/kdebase/workspace/cmake/modules/FindPAM.cmake
+++ b/kdebase/workspace/cmake/modules/FindPAM.cmake
@@ -5,6 +5,8 @@
 #  PAM_INCLUDE_DIR - the pam include directory
 #  PAM_LIBRARIES - libpam library
 
+if (WITH_PAM)
+
 if (PAM_INCLUDE_DIR AND PAM_LIBRARY)
 	# Already in cache, be silent
 	set(PAM_FIND_QUIETLY TRUE)
@@ -72,3 +74,5 @@ else (PAM_FOUND)
 endif (PAM_FOUND)
 
 mark_as_advanced(PAM_INCLUDE_DIR PAM_LIBRARY DL_LIBRARY PAM_MESSAGE_CONST)
+
+endif (WITH_PAM)
diff --git a/kdebase/workspace/cmake/modules/UnixAuth.cmake b/kdebase/workspace/cmake/modules/UnixAuth.cmake
index ed8ab9e..ce3bcc2 100644
--- a/kdebase/workspace/cmake/modules/UnixAuth.cmake
+++ b/kdebase/workspace/cmake/modules/UnixAuth.cmake
@@ -1,4 +1,4 @@
-find_package(PAM)
+macro_optional_find_package(PAM)
 
 set(UNIXAUTH_LIBRARIES)
 set(UNIXAUTH_INCLUDE_DIRS)
