diff --git a/kdiff3/src-QT4/difftextwindow.cpp b/kdiff3/src-QT4/difftextwindow.cpp
index c2c0779..ee104a3 100644
--- a/kdiff3/src-QT4/difftextwindow.cpp
+++ b/kdiff3/src-QT4/difftextwindow.cpp
@@ -33,6 +33,8 @@
 #include <QUrl>
 #include <QMenu>
 #include <QTextLayout>
+#include <QRunnable>
+#include <QThreadPool>
 
 
 
diff --git a/kdiff3/src-QT4/progress.cpp b/kdiff3/src-QT4/progress.cpp
index 9e585a3..88947db 100644
--- a/kdiff3/src-QT4/progress.cpp
+++ b/kdiff3/src-QT4/progress.cpp
@@ -15,6 +15,7 @@
 #include <QPushButton>
 #include <QLabel>
 #include <QApplication>
+#include <QThread>
 #include <kio/job.h>
 
 #include <klocale.h>
