From d5220258e7e652a6544cf80b2375c3fa185b1152 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Wed, 12 Sep 2018 18:12:14 +0200
Subject: [PATCH] Fix dependencies: add implicit, drop unused

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0221b84..d195569 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,18 +7,19 @@ set (KF5_MIN_VERSION "5.30.0")
 find_package(ECM ${KF5_MIN_VERSION} REQUIRED CONFIG)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 
-find_package(Qt5 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Widgets )
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Gui Widgets)
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS
+    Completion
     Config
     ConfigWidgets
     CoreAddons
     Crash
     DBusAddons
+    DocTools
     I18n
     KDELibs4Support
     KIO
     WidgetsAddons
-    WindowSystem
     XmlGui
 )
 
-- 
2.19.0

