diff --git a/gwenview/app/main.cpp b/gwenview/app/main.cpp
index 8bdf041..74dd694 100644
--- a/gwenview/app/main.cpp
+++ b/gwenview/app/main.cpp
@@ -39,7 +39,7 @@ int main(int argc, char *argv[]) {
 		"gwenview",        /* appname */
 		0,                 /* catalogName */
 		ki18n("Gwenview"), /* programName */
-		"2.2.0");          /* version */
+		"2.2.1");          /* version */
 	aboutData.setShortDescription(ki18n("An Image Viewer"));
 	aboutData.setLicense(KAboutData::License_GPL);
 	aboutData.setCopyrightStatement(ki18n("Copyright 2000-2008 Aurélien Gâteau"));
