--- gwenview-4.3.90/gwenview/app/mainwindow.cpp.orig	2010-01-10 09:32:02.000000000 -0400
+++ gwenview-4.3.90/gwenview/app/mainwindow.cpp	2010-01-10 09:37:48.000000000 -0400
@@ -419,8 +419,8 @@
 		InfoContextManagerItem* infoItem = new InfoContextManagerItem(mContextManager);
 		mContextManager->addItem(infoItem);
 
-		SemanticInfoContextManagerItem* semanticInfoItem = 0;
 		#ifndef GWENVIEW_SEMANTICINFO_BACKEND_NONE
+		SemanticInfoContextManagerItem* semanticInfoItem = 0;
 		// When using Nepomuk, create a SemanticInfoContextManagerItem instance
 		// only if Nepomuk is available
 		// When using Fake backend always create it
@@ -431,10 +431,10 @@
 		#ifdef GWENVIEW_SEMANTICINFO_BACKEND_NEPOMUK
 		}
 		#endif
-		#endif
 		if (semanticInfoItem) {
 			mContextManager->addItem(semanticInfoItem);
 		}
+		#endif
 
 		ImageOpsContextManagerItem* imageOpsItem =
 			new ImageOpsContextManagerItem(mContextManager, mWindow);
@@ -454,9 +454,11 @@
 		page = new SideBarPage(i18n("Information"));
 		page->setObjectName("information");
 		page->addWidget(infoItem->widget());
+		#ifndef GWENVIEW_SEMANTICINFO_BACKEND_NONE
 		if (semanticInfoItem) {
 			page->addWidget(semanticInfoItem->widget());
 		}
+		#endif
 		page->addStretch();
 		mSideBar->addPage(page);
 