diff -r -u basket-1.80-vanilla/CMakeLists.txt basket-1.80/CMakeLists.txt
--- basket-1.80-vanilla/CMakeLists.txt	2010-03-20 21:20:57.000000000 +0100
+++ basket-1.80/CMakeLists.txt	2010-03-28 22:04:43.000000000 +0200
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 2.4)
 
+OPTION(ENABLE_CRYPT "build encryption support" ON)
+
 find_package(KDE4 REQUIRED)
 find_package(QImageBlitz REQUIRED)
 find_package(Gpgme)
@@ -15,13 +17,14 @@
 
 include(MacroLibrary)
 
-IF(GPGME_FOUND)
+IF(GPGME_FOUND AND ENABLE_CRYPT)
   SET(HAVE_LIBGPGME 1)
   SET(LARGEFILE_SOURCE_1)
   ADD_DEFINITIONS(-D_FILE_OFFSET_BITS=64)
-ELSE(GPGME_FOUND)
-  MESSAGE("GPG not found, configuring without")
-ENDIF(GPGME_FOUND)
+  MESSAGE("Configuring with GPG support")
+ELSE(GPGME_FOUND AND ENABLE_CRYPT)
+  MESSAGE("Configuring without GPG support")
+ENDIF(GPGME_FOUND AND ENABLE_CRYPT)
 
 #TODO: find meinproc
 
