diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 6084f2f..7529b41 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -222,8 +222,6 @@ file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/CLuceneConfig.cmake" "
     set(CLUCENE_VERSION ${CLUCENE_VERSION})
     set(CLUCENE_SOVERSION ${CLUCENE_SOVERSION})
 ")
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/CLuceneConfig.cmake"
-        DESTINATION ${LIB_DESTINATION}/CLuceneConfig.cmake)
 
 # install pkg-config file
 IF(NOT WIN32)
@@ -238,16 +236,10 @@ IF ( LUCENE_SYS_INCLUDES )
     install(FILES ${clucene-shared_BINARY_DIR}/CLucene/clucene-config.h 
             DESTINATION ${LUCENE_SYS_INCLUDES}/CLucene  
             COMPONENT development)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/CLuceneConfig.cmake 
-            DESTINATION ${LUCENE_SYS_INCLUDES}/CLucene  
-            COMPONENT development)
 ELSE ( LUCENE_SYS_INCLUDES )
     install(FILES ${clucene-shared_BINARY_DIR}/CLucene/clucene-config.h 
             DESTINATION include/CLucene
             COMPONENT development)
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/CLuceneConfig.cmake 
-            DESTINATION include/CLucene
-            COMPONENT development)
 ENDIF ( LUCENE_SYS_INCLUDES )
 
 
