Only in .: build
diff -ru ../libdbusmenu-qt-0.3.2/cmake/modules/FindQJSON.cmake ./cmake/modules/FindQJSON.cmake
--- ../libdbusmenu-qt-0.3.2/cmake/modules/FindQJSON.cmake	2010-04-27 15:20:49.187784398 +0200
+++ ./cmake/modules/FindQJSON.cmake	2010-04-27 15:21:06.515858660 +0200
@@ -17,14 +17,14 @@
     # use pkg-config to get the values of QJSON_INCLUDE_DIRS
     # and QJSON_LIBRARY_DIRS to add as hints to the find commands.
     include (FindPkgConfig)
-    pkg_check_modules (QJSON QJson>=0.5)
+    pkg_check_modules (PC_QJSON QJson>=0.5)
   endif (NOT WIN32)
 
   find_library (QJSON_LIBRARIES
     NAMES
     qjson
     PATHS
-    ${QJSON_LIBRARY_DIRS}
+    ${PC_QJSON_LIBRARY_DIRS}
     ${LIB_INSTALL_DIR}
     ${KDE4_LIB_DIR}
   )
@@ -35,7 +35,7 @@
     PATH_SUFFIXES
     qjson
     PATHS
-    ${QJSON_INCLUDE_DIRS}
+    ${PC_QJSON_INCLUDE_DIRS}
     ${INCLUDE_INSTALL_DIR}
     ${KDE4_INCLUDE_DIR}
   )
