--- a/CMakeLists.txt	2015-01-19 01:06:41.388214548 +0100
+++ b/CMakeLists.txt	2015-01-19 01:13:14.851202822 +0100
@@ -26,7 +26,7 @@
 ecm_setup_version(${Konsole_VERSION} VARIABLE_PREFIX KONSOLEPRIVATE
                   SOVERSION ${Konsole_VERSION_MAJOR}
 )
-find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core DBus Widgets Script Test)
+find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core DBus Widgets Script)
 
 find_package(KF5 REQUIRED
     Bookmarks Completion Config ConfigWidgets
--- a/src/CMakeLists.txt	2015-01-19 01:06:41.386214548 +0100
+++ b/src/CMakeLists.txt	2015-01-19 01:15:51.963198139 +0100
@@ -22,8 +22,11 @@
               ${CMAKE_CURRENT_BINARY_DIR}/config-konsole.h)
 
 ### Tests
-add_subdirectory(autotests)
-add_subdirectory(tests)
+if(BUILD_TESTING)
+  find_package(Qt5Test ${QT_MIN_VERSION} CONFIG REQUIRED)
+  add_subdirectory(autotests)
+  add_subdirectory(tests)
+endif()
 
 ### Font Embedder and LineFont.h
 option(KONSOLE_BUILD_FONTEMBEDDER "Konsole: build fontembedder executable" OFF)
