diff -Naur a/telepathy-qt4-0.7.1/TelepathyQt4/CMakeLists.txt b/telepathy-qt4-0.7.1/TelepathyQt4/CMakeLists.txt
--- a/telepathy-qt4-0.7.1/TelepathyQt4/CMakeLists.txt	2011-06-14 23:26:48.708045080 +0200
+++ b/telepathy-qt4-0.7.1/TelepathyQt4/CMakeLists.txt	2011-06-14 23:27:05.997827718 +0200
@@ -603,8 +603,6 @@
 tpqt4_future_client_generator(channel-dispatcher TpFuture::Client --mainiface=Tp::Client::ChannelDispatcherInterface DEPENDS channel-dispatcher-future-xincludator)
 tpqt4_future_client_generator(misc TpFuture::Client DEPENDS misc-future-xincludator)
 
-add_dependencies(doxygen-doc all-generated-sources)
-
 # Create the library
 if (ENABLE_COMPILER_COVERAGE)
     add_library(telepathy-qt4 STATIC ${telepathy_qt4_SRCS})
