*** eyesaver.original/eyesaver-0.1/CMakeLists.txt	2009-02-17 23:10:01.538216659 +0300
--- eyesaver/eyesaver-0.1/CMakeLists.txt	2009-02-17 23:10:40.097216054 +0300
***************
*** 4,11 ****
  # Find the required Libaries
  find_package(KDE4 REQUIRED)
  include(KDE4Defaults)
! find_package(Plasma REQUIRED)
!  
  add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
  include_directories(
     ${CMAKE_SOURCE_DIR}
--- 4,15 ----
  # Find the required Libaries
  find_package(KDE4 REQUIRED)
  include(KDE4Defaults)
! 
! if ( ${KDE_VERSION} VERSION_LESS "4.1.90" )
! else ( ${KDE_VERSION} VERSION_LESS "4.1.90" )
! set( PLASMA_LIBS ${KDE4_PLASMA_LIBS} )
! endif ( ${KDE_VERSION} VERSION_LESS "4.1.90" )
! 
  add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
  include_directories(
     ${CMAKE_SOURCE_DIR}
***************
*** 28,31 ****
          DESTINATION ${SERVICES_INSTALL_DIR}) 
  
  install(FILES images/occhio2.svg
!         DESTINATION ${DATA_INSTALL_DIR}/desktoptheme/default/widgets/)
\ No newline at end of file
--- 32,35 ----
          DESTINATION ${SERVICES_INSTALL_DIR}) 
  
  install(FILES images/occhio2.svg
!         DESTINATION ${DATA_INSTALL_DIR}/desktoptheme/default/widgets/)
