
Index: D:/WaterWATCH/dev/opencv/modules/highgui/src/cap_ffmpeg.cpp
===================================================================
--- D:/WaterWATCH/dev/opencv/modules/highgui/src/cap_ffmpeg.cpp	(revision 2121)
+++ D:/WaterWATCH/dev/opencv/modules/highgui/src/cap_ffmpeg.cpp	(revision 2122)
@@ -815,24 +815,25 @@
 #endif
 };
 
-static const char * icvFFMPEGErrStr(int err)
-{
-    switch(err) {
-    case AVERROR_NUMEXPECTED:
-		return "Incorrect filename syntax";
-    case AVERROR_INVALIDDATA:
-		return "Invalid data in header";
-    case AVERROR_NOFMT:
-		return "Unknown format";
-    case AVERROR_IO:
-		return "I/O error occurred";
-    case AVERROR_NOMEM:
-		return "Memory allocation error";
-    default:
-		break;
-    }
-  	return "Unspecified error";
-}
+//Deprecated Errors, should be using AVERROR(EINVAL) to return error strings
+//static const char * icvFFMPEGErrStr(int err)
+//{
+//    switch(err) {
+//    case AVERROR_NUMEXPECTED:
+//		return "Incorrect filename syntax";
+//    case AVERROR_INVALIDDATA:
+//		return "Invalid data in header";
+//    case AVERROR_NOFMT:
+//		return "Unknown format";
+//    case AVERROR_IO:
+//		return "I/O error occurred";
+//    case AVERROR_NOMEM:
+//		return "Memory allocation error";
+//    default:
+//		break;
+//    }
+//  	return "Unspecified error";
+//}
 
 /* function internal to FFMPEG (libavformat/riff.c) to lookup codec id by fourcc tag*/
 extern "C" {
@@ -1322,7 +1323,7 @@
     /* open the codec */
     if ( (err=avcodec_open(c, codec)) < 0) {
 		char errtext[256];
-		sprintf(errtext, "Could not open codec '%s': %s", codec->name, icvFFMPEGErrStr(err));
+		sprintf(errtext, "Could not open codec '%s': %s", codec->name, AVERROR(EINVAL));
 		CV_Error(CV_StsBadArg, errtext);
     }
 
