
Index: /trunk/opencv/modules/highgui/src/grfmt_png.cpp
===================================================================
--- /trunk/opencv/modules/highgui/src/grfmt_png.cpp	(revision 4337)
+++ /trunk/opencv/modules/highgui/src/grfmt_png.cpp	(revision 4945)
@@ -108,5 +108,5 @@
 {
     png_structp png_ptr = (png_structp)_png_ptr;
-    PngDecoder* decoder = (PngDecoder*)(png_ptr->io_ptr);
+    PngDecoder* decoder = (PngDecoder*)(png_get_io_ptr(png_ptr));
     CV_Assert( decoder );
     const Mat& buf = decoder->m_buf;
@@ -139,5 +139,5 @@
         if( info_ptr && end_info )
         {
-            if( setjmp( png_ptr->jmpbuf ) == 0 )
+            if( setjmp( png_jmpbuf( png_ptr ) ) == 0 )
             {
                 if( !m_buf.empty() )
@@ -201,5 +201,5 @@
         png_infop end_info = (png_infop)m_end_info;
 
-        if( setjmp(png_ptr->jmpbuf) == 0 )
+        if( setjmp( png_jmpbuf ( png_ptr ) ) == 0 )
         {
             int y;
@@ -285,5 +285,5 @@
         return;
     png_structp png_ptr = (png_structp)_png_ptr;
-    PngEncoder* encoder = (PngEncoder*)(png_ptr->io_ptr);
+    PngEncoder* encoder = (PngEncoder*)(png_get_io_ptr(png_ptr));
     CV_Assert( encoder && encoder->m_buf );
     size_t cursz = encoder->m_buf->size();
@@ -327,5 +327,5 @@
         if( info_ptr )
         {
-            if( setjmp( png_ptr->jmpbuf ) == 0 )
+            if( setjmp( png_jmpbuf ( png_ptr ) ) == 0 )
             {
                 if( m_buf )
