diff -up OpenCV-2.2.0/CMakeLists.txt.numpy OpenCV-2.2.0/CMakeLists.txt
--- OpenCV-2.2.0/CMakeLists.txt.numpy	2010-12-07 17:54:56.369070005 +0100
+++ OpenCV-2.2.0/CMakeLists.txt	2010-12-07 17:58:31.987070004 +0100
@@ -532,7 +532,7 @@ if(BUILD_NEW_PYTHON_SUPPORT)
                     OUTPUT_STRIP_TRAILING_WHITESPACE)
     if(PYTHON_NUMPY_PROCESS EQUAL 0)
       set(PYTHON_USE_NUMPY 1)
-      add_definitions(-D PYTHON_USE_NUMPY=1)
+      add_definitions(-DPYTHON_USE_NUMPY=1)
       include_directories(AFTER ${PYTHON_NUMPY_INCLUDE_DIRS})
       message(STATUS "    Use INCLUDE: ${PYTHON_NUMPY_INCLUDE_DIRS}")
     else()
