--- a/CMakeLists.txt	2015-01-21 21:44:44.786490427 +0100
+++ b/CMakeLists.txt	2015-01-21 22:07:15.833138021 +0100
@@ -30,7 +30,11 @@
 include(KDECMakeSettings)
 include(KDECompilerSettings)
 
-find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core DBus Widgets Script Sql Test)
+find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core DBus Widgets Script Sql)
+
+if(BUILD_TESTING)
+  find_package(Qt5Test ${QT_MIN_VERSION} CONFIG REQUIRED)
+endif()
 
 # Load the frameworks we need
 find_package(KF5 REQUIRED COMPONENTS
