--- k3b-1.66.0.orig/cmake/modules/FindMuse.cmake
+++ k3b-1.66.0/cmake/modules/FindMuse.cmake
@@ -20,8 +20,8 @@ FIND_PATH(MUSE_INCLUDE_DIR NAMES mpcdec/
 )
 
 if(MUSE_INCLUDE_DIR)
-  FIND_LIBRARY(MUSE_LIBRARIES NAMES mpcdec)
-  set(MPC_HEADER_FILE "<mpcdec/mpcdec.h>")
+  FIND_LIBRARY(MUSE_LIBRARIES NAMES mpcdecsv7 )
+  set(MPC_HEADER_FILE "<mpcdecsv7/mpcdec.h>")
 else(MUSE_INCLUDE_DIR)
   FIND_PATH(MUSE_INCLUDE_DIR NAMES musepack/musepack.h)
   set(MPC_HEADER_FILE "<musepack/musepack.h>")
