diff -ru ../virtuoso-opensource-6.1.0/appsrc/Makefile.am ./appsrc/Makefile.am
--- ../virtuoso-opensource-6.1.0/appsrc/Makefile.am	2007-05-07 15:23:31.000000000 +0200
+++ ./appsrc/Makefile.am	2010-02-04 03:29:07.355245690 +0100
@@ -19,17 +19,4 @@
 #  
 #  
 
-SUBDIRS = \
-	ODS-Framework \
-	ODS-Addressbook \
-	ODS-Blog \
-	ODS-Bookmark \
-	ODS-Briefcase \
-	ODS-Calendar \
-	ODS-Community \
-	ODS-Discussion \
-	ODS-FeedManager \
-	ODS-Gallery \
-	ODS-Polls \
-	ODS-WebMail \
-	ODS-Wiki
+SUBDIRS = ODS-Framework ODS-Addressbook ODS-Blog ODS-Bookmark ODS-Briefcase ODS-Calendar ODS-Community ODS-Discussion ODS-FeedManager ODS-Gallery ODS-Polls ODS-WebMail ODS-Wiki
diff -ru ../virtuoso-opensource-6.1.0/configure.in ./configure.in
--- ../virtuoso-opensource-6.1.0/configure.in	2010-02-03 21:49:18.000000000 +0100
+++ ./configure.in	2010-02-04 03:29:07.406282543 +0100
@@ -608,18 +608,6 @@
 ##									##
 ##########################################################################
 
-AC_ARG_WITH(debug, [dnl
-AC_HELP_STRING([--with-debug], [debugging code])dnl
-], [ with_debug="$withval" ],
-   [ with_debug="no" ])
-if test "x$with_debug" = "xyes"; then
-    CCOPT="-g"
-    AM_LDFLAGS="-g"
-else
-    CCOPT=""
-    AM_LDFLAGS="-s"
-fi
-
 AC_CHECKING([default compiler values])
 if eval "${CC-cc} $CFLAGS $AM_LDFLAGS -o binsrc/config/ccdefs binsrc/config/ccdefs.c" ; then
     eval `binsrc/config/ccdefs`
diff -ru ../virtuoso-opensource-6.1.0/Makefile.am ./Makefile.am
--- ../virtuoso-opensource-6.1.0/Makefile.am	2010-02-03 17:09:01.000000000 +0100
+++ ./Makefile.am	2010-02-04 03:29:07.414281441 +0100
@@ -25,14 +25,6 @@
 
 SUBDIRS = . docsrc libsrc binsrc appsrc
 
-dist_doc_DATA = \
-	AUTHORS \
-	COPYING \
-	CREDITS \
-	INSTALL \
-	LICENSE \
-	NEWS \
-	README*
 
 # ----------------------------------------------------------------------
 #
