--- themedesigner/CMakeLists.txt.orig	2009-01-20 00:29:10.000000000 +0100
+++ themedesigner/CMakeLists.txt	2009-01-20 00:29:29.000000000 +0100
@@ -85,6 +85,9 @@
 TARGET_LINK_LIBRARIES(digikamthemedesigner 
                       digikamcore 
                       digikamdatabase 
+                      ${KDE4_PHONON_LIBS}
+                      ${KDE4_KPARTS_LIBS}
+                      ${KDE4_KHTML_LIBS}
                       ${KDE4_KUTILS_LIBS}
                       ${KIPI_LIBRARIES}
                       ${GPHOTO2_LIBRARIES}
