diff -r -u digikam-1.2.0.orig/CMakeLists.txt digikam-1.2.0/CMakeLists.txt
--- digikam-1.2.0.orig/CMakeLists.txt	2010-05-30 21:38:14.189946492 +0200
+++ digikam-1.2.0/CMakeLists.txt	2010-05-30 21:39:46.679946458 +0200
@@ -3,6 +3,8 @@
 #OPTION(ENABLE_THEMEDESIGNER "Build digiKam theme designer (default=OFF)" OFF)
 OPTION(ENABLE_THUMBS_DB "Build digiKam with thumbnails database support (default=ON)" ON)
 
+ENABLE_LANGUAGE(Fortran)
+
 # ==================================================================================================
 # Information to update before to release this package.
 
@@ -123,6 +125,9 @@
 # This function call the script cmake/modules/FindPGF.cmake
 FIND_PACKAGE(PGF)
 
+# Included to unbundle lapack that is in portage tree.
+FIND_PACKAGE(LAPACK)
+
 FIND_PACKAGE(Jasper)
 FIND_PACKAGE(Kdcraw)
 FIND_PACKAGE(Kexiv2)
@@ -320,13 +325,13 @@
     MESSAGE(STATUS "")
 ENDIF(DOXYGEN_FOUND)
 
-IF(LCMS_FOUND AND JPEG_FOUND AND PNG_FOUND AND TIFF_FOUND AND PGF_FOUND AND JASPER_FOUND AND KDCRAW_FOR_DIGIKAM AND KEXIV2_FOUND AND KIPI_FOUND)
+IF(LCMS_FOUND AND JPEG_FOUND AND PNG_FOUND AND TIFF_FOUND AND PGF_FOUND AND JASPER_FOUND AND KDCRAW_FOR_DIGIKAM AND KEXIV2_FOUND AND KIPI_FOUND AND LAPACK_FOUND)
     MESSAGE(STATUS " digiKam will be compiled................. YES")
     SET(DIGIKAM_CAN_BE_COMPILED true)
-ELSE(LCMS_FOUND AND JPEG_FOUND AND PNG_FOUND AND TIFF_FOUND AND PGF_FOUND AND JASPER_FOUND AND KDCRAW_FOR_DIGIKAM AND KEXIV2_FOUND AND KIPI_FOUND)
+ELSE(LCMS_FOUND AND JPEG_FOUND AND PNG_FOUND AND TIFF_FOUND AND PGF_FOUND AND JASPER_FOUND AND KDCRAW_FOR_DIGIKAM AND KEXIV2_FOUND AND KIPI_FOUND AND LAPACK_FOUND)
     MESSAGE(FATAL_ERROR " digiKam will be compiled................. NO  (Look README file for more details about dependencies)")
     SET(DIGIKAM_CAN_BE_COMPILED false)
-ENDIF(LCMS_FOUND AND JPEG_FOUND AND PNG_FOUND AND TIFF_FOUND AND PGF_FOUND AND JASPER_FOUND AND KDCRAW_FOR_DIGIKAM AND KEXIV2_FOUND AND KIPI_FOUND)
+ENDIF(LCMS_FOUND AND JPEG_FOUND AND PNG_FOUND AND TIFF_FOUND AND PGF_FOUND AND JASPER_FOUND AND KDCRAW_FOR_DIGIKAM AND KEXIV2_FOUND AND KIPI_FOUND AND LAPACK_FOUND)
 
 MESSAGE(STATUS "----------------------------------------------------------------------------------")
 MESSAGE(STATUS "")
@@ -484,39 +489,7 @@
         ${CMAKE_CURRENT_SOURCE_DIR}/libs/dimg/loaders/pgfsettings.cpp
        )
 
-    SET(libclapack_SRCS
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/abort_.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dgesv.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dlaswp.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/endfile.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/idamax.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/open.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/sig_die.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/wref.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/close.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dgetf2.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dscal.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/err.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/ieeeck.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/s_cmp.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/s_stop.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/wrtfmt.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dgemm.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dgetrf.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dswap.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/fmt.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/ilaenv.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/s_copy.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/wsfe.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dger.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dgetrs.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dtrsm.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/fmtlib.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/lsame.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/sfe.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/util.c
-        ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/xerbla.c
-       )
+    INCLUDE_DIRECTORIES(${LAPACK_INCLUDE_DIRS})
 
     SET(libdimgfilters_SRCS
         ${libclapack_SRCS}
Nur in digikam-1.2.0: CMakeLists.txt.orig.
diff -r -u digikam-1.2.0.orig/digikam/CMakeLists.txt digikam-1.2.0/digikam/CMakeLists.txt
--- digikam-1.2.0.orig/digikam/CMakeLists.txt	2010-05-30 21:38:14.191892532 +0200
+++ digikam-1.2.0/digikam/CMakeLists.txt	2010-05-30 21:39:46.680979177 +0200
@@ -89,6 +89,7 @@
                      ${PNG_PNG_INCLUDE_DIR}
                      ${TIFF_INCLUDE_DIR}
 		     ${PGF_INCLUDE_DIRS}
+		     ${LAPACK_INCLUDE_DIRS}
                    )
 
 IF(LENSFUN_LIBRARY)
@@ -172,6 +173,7 @@
 		      ${PGF_LIBRARIES}
                       ${LCMS_LIBRARIES}
                       ${JASPER_LIBRARIES}
+		      ${LAPACK_LIBRARIES}
 
                       # Require by CImg library which use pthread internally.
                       ${CMAKE_THREAD_LIBS_INIT}
@@ -237,6 +239,7 @@
                       ${QT_QTGUI_LIBRARY}
                       ${QT_QTSQL_LIBRARY}
 		      ${PGF_LIBRARIES}
+		      ${LAPACK_LIBRARIES}
                      )
 
 SET_TARGET_PROPERTIES(digikamdatabase PROPERTIES VERSION 1.0.0 SOVERSION 1 )
Nur in digikam-1.2.0/digikam: CMakeLists.txt.orig.
diff -r -u digikam-1.2.0.orig/libs/CMakeLists.txt digikam-1.2.0/libs/CMakeLists.txt
--- digikam-1.2.0.orig/libs/CMakeLists.txt	2010-05-30 21:38:14.191892532 +0200
+++ digikam-1.2.0/libs/CMakeLists.txt	2010-05-30 21:41:20.226778777 +0200
@@ -14,12 +14,12 @@
                      imageproperties
                      dialogs
                      3rdparty/cimg
-                     3rdparty/clapack
                      ${JPEG_INCLUDE_DIR}
                      ${TIFF_INCLUDE_DIR}
                      ${PNG_INCLUDE_DIR}
                      ${PGF_INCLUDE_DIRS}
                      ${JASPER_INCLUDE_DIR}
+                     ${LAPACK_INCLUDE_DIRS}
                    )
 ADD_SUBDIRECTORY( histogram )
 ADD_SUBDIRECTORY( dmetadata )
