diff -r -u -N digikam-1.4.0.1pgf/CMakeLists.txt digikam-1.4.0/CMakeLists.txt
--- digikam-1.4.0.1pgf/CMakeLists.txt	2010-08-31 23:39:20.000000000 +0200
+++ digikam-1.4.0/CMakeLists.txt	2010-08-31 23:39:47.000000000 +0200
@@ -141,6 +141,7 @@
 FIND_PACKAGE(Kdcraw)
 FIND_PACKAGE(Kexiv2)
 FIND_PACKAGE(Kipi)
+FIND_PACKAGE(CLAPACK)
 
 # Extract version of libjpeg so that we can use the appropriate dir
 # See bug #227313, #228483
@@ -278,7 +279,13 @@
     MESSAGE(STATUS " libpgf library found..................... YES (optional)")
 ELSE(PGF_FOUND)
     MESSAGE(STATUS " libpgf library found..................... NO  (optional - internal version used instead)")
-ENDIF(GPHOTO2_FOUND)
+ENDIF(PGF_FOUND)
+
+IF(CLAPACK_FOUND)
+    MESSAGE(STATUS " libclapack library found................. YES (optional)")
+ELSE(CLAPACK_FOUND)
+    MESSAGE(STATUS " libclapack library found................. NO  (optional - internal version used instead)")
+ENDIF(CLAPACK_FOUND)
 
 
 # ================================================================
@@ -377,13 +384,19 @@
 ENDIF(DOXYGEN_FOUND)
 
 IF(LCMS_FOUND AND JPEG_FOUND AND PNG_FOUND AND TIFF_FOUND AND JASPER_FOUND AND KDCRAW_FOR_DIGIKAM AND KEXIV2_FOR_DIGIKAM AND KIPI_FOUND)
-    MESSAGE(STATUS " digiKam will be compiled................. YES")
+    MESSAGE(STATUS " digiKam can be compiled.................. YES")
     SET(DIGIKAM_CAN_BE_COMPILED true)
 ELSE(LCMS_FOUND AND JPEG_FOUND AND PNG_FOUND AND TIFF_FOUND AND AND JASPER_FOUND AND KDCRAW_FOR_DIGIKAM AND KEXIV2_FOR_DIGIKAM AND KIPI_FOUND)
     MESSAGE(FATAL_ERROR " digiKam will be compiled................. NO  (Look README file for more details about dependencies)")
     SET(DIGIKAM_CAN_BE_COMPILED false)
 ENDIF(LCMS_FOUND AND JPEG_FOUND AND PNG_FOUND AND TIFF_FOUND AND JASPER_FOUND AND KDCRAW_FOR_DIGIKAM AND KEXIV2_FOR_DIGIKAM AND KIPI_FOUND)
 
+IF(DIGIKAM_CAN_BE_COMPILED AND FORCED_UNBUNDLE AND ((NOT PGF_FOUND) OR (NOT CLAPACK_FOUND)))
+    MESSAGE(STATUS "")
+    MESSAGE(FATAL_ERROR " BUNDLED LIBRARIES HAVE BEEN FORCE-DISABLED IN THIS BUILD. Aborting.")
+    SET(DIGIKAM_CAN_BE_COMPILED false)
+ENDIF(DIGIKAM_CAN_BE_COMPILED AND FORCED_UNBUNDLE AND ((NOT PGF_FOUND) OR (NOT CLAPACK_FOUND)))
+
 MESSAGE(STATUS "----------------------------------------------------------------------------------")
 MESSAGE(STATUS "")
 
@@ -541,7 +554,8 @@
         ${CMAKE_CURRENT_SOURCE_DIR}/libs/dimg/loaders/pgfsettings.cpp
        )
 
-    SET(libclapack_SRCS
+    if(NOT CLAPACK_FOUND)
+      SET(libclapack_SRCS
         ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/abort_.c
         ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dgesv.c
         ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/dlaswp.c
@@ -574,6 +588,7 @@
         ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/util.c
         ${CMAKE_CURRENT_SOURCE_DIR}/libs/3rdparty/clapack/xerbla.c
        )
+    endif(NOT CLAPACK_FOUND)
 
     SET(libdimgfilters_SRCS
         ${libclapack_SRCS}
diff -r -u -N digikam-1.4.0.1pgf/cmake/modules/FindCLAPACK.cmake digikam-1.4.0/cmake/modules/FindCLAPACK.cmake
--- digikam-1.4.0.1pgf/cmake/modules/FindCLAPACK.cmake	1970-01-01 01:00:00.000000000 +0100
+++ digikam-1.4.0/cmake/modules/FindCLAPACK.cmake	2010-08-31 23:43:20.000000000 +0200
@@ -0,0 +1,44 @@
+# - Try to find clapack library
+# Once done this will define
+#
+#  CLAPACK_FOUND - system has CLAPACK library
+#  CLAPACK_INCLUDE_DIR - the CLAPACK include directory
+#  CLAPACK_LIBRARY - the marblewidget library
+#
+#  copyright 2008 by Patrick Spendrin <ps_ml@gmx.de>
+#  copyright 2010 by Andreas K. Huettel <mail@akhuettel.de>
+#  use this file as you like
+#
+
+if(CLAPACK_INCLUDE_DIR AND CLAPACK_LIBRARY)
+
+  # Already in cache
+  set(CLAPACK_FOUND TRUE)
+
+else(CLAPACK_INCLUDE_DIR AND CLAPACK_LIBRARY)
+
+  find_path(CLAPACK_INCLUDE_DIR clapack.h PATHS /usr/include/clapack NO_DEFAULT_PATH)
+  if(CLAPACK_INCLUDE_DIR)
+     message(STATUS "Found clapack includes: ${CLAPACK_INCLUDE_DIR}")
+  else(CLAPACK_INCLUDE_DIR)
+     find_path(CLAPACK_INCLUDE_DIR clapack.h PATHS /usr/include/clapack)
+     if(CLAPACK_INCLUDE_DIR)
+        message(STATUS "Found clapack includes: ${CLAPACK_INCLUDE_DIR}")
+     endif(CLAPACK_INCLUDE_DIR)
+  endif(CLAPACK_INCLUDE_DIR)
+
+  find_library(CLAPACK_LIBRARY clapack)
+  if(CLAPACK_LIBRARY)
+      message(STATUS "Found clapack library: ${CLAPACK_LIBRARY}")
+  endif(CLAPACK_LIBRARY)
+
+  if(CLAPACK_INCLUDE_DIR AND CLAPACK_LIBRARY)
+    set(CLAPACK_FOUND TRUE)
+  endif(CLAPACK_INCLUDE_DIR AND CLAPACK_LIBRARY)
+
+  if(NOT CLAPACK_FOUND)
+      message(STATUS "Could NOT find any working clapack installation")
+  endif(NOT CLAPACK_FOUND)
+
+  mark_as_advanced(CLAPACK_INCLUDE_DIR CLAPACK_LIBRARY)
+endif(CLAPACK_INCLUDE_DIR AND CLAPACK_LIBRARY)
diff -r -u -N digikam-1.4.0.1pgf/digikam/CMakeLists.txt digikam-1.4.0/digikam/CMakeLists.txt
--- digikam-1.4.0.1pgf/digikam/CMakeLists.txt	2010-08-31 23:39:20.000000000 +0200
+++ digikam-1.4.0/digikam/CMakeLists.txt	2010-08-31 23:39:47.000000000 +0200
@@ -53,7 +53,6 @@
                      # specific libs and utilities
                      ${CMAKE_CURRENT_SOURCE_DIR}/../libs/3rdparty/lprof
                      ${CMAKE_CURRENT_SOURCE_DIR}/../libs/3rdparty/sqlite2
-                     ${CMAKE_CURRENT_SOURCE_DIR}/../libs/3rdparty/clapack
                      ${CMAKE_CURRENT_SOURCE_DIR}/../libs/3rdparty/cimg
                      ${CMAKE_CURRENT_SOURCE_DIR}/../libs/themeengine
                      ${CMAKE_CURRENT_SOURCE_DIR}/../utilities/kipiiface
@@ -103,10 +102,18 @@
    INCLUDE_DIRECTORIES(${LENSFUN_INCLUDE_DIR})
 ENDIF(LENSFUN_LIBRARY)
 
+IF(CLAPACK_FOUND)
+   INCLUDE_DIRECTORIES(${CLAPACK_INCLUDE_DIR})
+ELSE(CLAPACK_FOUND)
+   INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/../libs/3rdparty/clapack)
+ENDIF(CLAPACK_FOUND)
+
 # Shared libdigikamcore ################################################################
 
-# Disable Clapack C code warnings.
-SET_SOURCE_FILES_PROPERTIES(${libclapack_SRCS} PROPERTIES COMPILE_FLAGS "-w")
+IF(NOT CLAPACK_FOUND)
+   # Disable Clapack C code warnings.
+   SET_SOURCE_FILES_PROPERTIES(${libclapack_SRCS} PROPERTIES COMPILE_FLAGS "-w")
+ENDIF(NOT CLAPACK_FOUND)
 
 IF(NOT PGF_FOUND)
    # Disable libpgf warnings.
@@ -203,6 +210,10 @@
     TARGET_LINK_LIBRARIES(digikamcore ${LIBMARBLEWIDGET_LIBRARY})
 ENDIF(MARBLEWIDGET_FOUND)
 
+IF(CLAPACK_FOUND)
+   TARGET_LINK_LIBRARIES(digikamcore ${CLAPACK_LIBRARY})
+ENDIF(CLAPACK_FOUND)
+
 IF(Q_WS_X11)
     TARGET_LINK_LIBRARIES(digikamcore ${X11_LIBRARIES})
 ENDIF(Q_WS_X11)
@@ -359,6 +370,10 @@
   TARGET_LINK_LIBRARIES(digikam ${LIBMARBLEWIDGET_LIBRARY})
 ENDIF(MARBLEWIDGET_FOUND)
 
+IF(CLAPACK_FOUND)
+   TARGET_LINK_LIBRARIES(digikam ${CLAPACK_LIBRARY})
+ENDIF(CLAPACK_FOUND)
+
 INSTALL(TARGETS digikam ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 # Others Files to install #######################################################
diff -r -u -N digikam-1.4.0.1pgf/showfoto/CMakeLists.txt digikam-1.4.0/showfoto/CMakeLists.txt
--- digikam-1.4.0.1pgf/showfoto/CMakeLists.txt	2010-08-22 10:53:39.000000000 +0200
+++ digikam-1.4.0/showfoto/CMakeLists.txt	2010-08-31 23:39:47.000000000 +0200
@@ -65,6 +65,10 @@
 
 KDE4_ADD_EXECUTABLE(showfoto ${showfoto_SRCS})
 
+IF(CLAPACK_FOUND)
+   TARGET_LINK_LIBRARIES(showfoto ${CLAPACK_LIBRARY})
+ENDIF(CLAPACK_FOUND)
+
 TARGET_LINK_LIBRARIES(showfoto
                       digikamcore
                       ${KEXIV2_LIBRARIES}
