diff -ruN digikam-2.0.0-beta5.orig/core/CMakeLists.txt digikam-2.0.0-beta5/core/CMakeLists.txt
--- digikam-2.0.0-beta5.orig/core/CMakeLists.txt	2011-05-04 10:52:35.000000000 +0200
+++ digikam-2.0.0-beta5/core/CMakeLists.txt	2011-05-11 22:40:30.000000000 +0200
@@ -1560,8 +1560,15 @@
 # ==================================================================================================
 # API documentation generation
 
+# Checks if user ask to build documentation.
+IF(BUILD_DOC)
+
 IF(DOXYGEN_FOUND)
 
+    MESSAGE(STATUS "")
+    MESSAGE(STATUS "API documentation will be installed")
+    MESSAGE(STATUS "")
+
     SET(API_DIR ${CMAKE_BINARY_DIR}/api)
     SET(SOURCE_DIR ${CMAKE_SOURCE_DIR})
     SET(DOXYFILE ${CMAKE_BINARY_DIR}/Doxyfile)
@@ -1569,7 +1576,21 @@
 
     CONFIGURE_FILE(Doxyfile.cmake ${DOXYFILE})
 
-    ADD_CUSTOM_TARGET(doc ${DOXYGEN_EXECUTABLE} ${DOXYFILE}
+    ADD_CUSTOM_TARGET(doc ALL ${DOXYGEN_EXECUTABLE} ${DOXYFILE}
                       WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+
+ELSE(DOXYGEN_FOUND)
+
+    MESSAGE(FATAL_ERROR "API documentation requested but doxygen not found")
+
 ENDIF(DOXYGEN_FOUND)
 
+ELSE(BUILD_DOC)
+
+
+    MESSAGE(STATUS "")
+    MESSAGE(STATUS "Documentation will NOT be installed, because")
+    MESSAGE(STATUS "the option -DBUILD_DOC=ON wasn't passed to cmake")
+    MESSAGE(STATUS "")
+
+ENDIF(BUILD_DOC)
