commit 1f80f35b68702136ebd3d315fe4baf7d75b3c142
Author: Gilles Caulier <caulier.gilles@gmail.com>
Date:   Sun Jul 3 10:57:21 2011 +0200

    compile with libpgf 6.09.44
    BUGS: 276969

diff --git a/libs/threadimageio/pgfutils.cpp b/libs/threadimageio/pgfutils.cpp
index 2b5829f..758f12c 100644
--- a/libs/threadimageio/pgfutils.cpp
+++ b/libs/threadimageio/pgfutils.cpp
@@ -151,7 +151,14 @@ bool writePGFImageData(const QImage& img, QByteArray& data, int quality)
         // TODO : optimize memory allocation...
         CPGFMemoryStream stream(256000);
         UINT32 nWrittenBytes = 0;
+
+#ifdef PGFCodecVersionID
+#   if PGFCodecVersionID >= 0x061124
         pgfImg.Write(&stream, &nWrittenBytes);
+#   endif
+#else
+        pgfImg.Write(&stream, 0, 0, &nWrittenBytes);
+#endif
 
         data = QByteArray((const char*)stream.GetBuffer(), nWrittenBytes);
 
