diff -ruN kscreensaver-4.6.4.orig/ConfigureChecks.cmake kscreensaver-4.6.4/ConfigureChecks.cmake
--- kscreensaver-4.6.4.orig/ConfigureChecks.cmake	2011-02-25 22:54:48.000000000 +0100
+++ kscreensaver-4.6.4/ConfigureChecks.cmake	2011-06-13 22:34:10.713760997 +0200
@@ -73,7 +73,7 @@
 macro_bool_to_01(X11_Xinerama_FOUND HAVE_XINERAMA)
 macro_bool_to_01(X11_Xrandr_FOUND HAVE_XRANDR) # kwin
 macro_bool_to_01(X11_Xrender_FOUND HAVE_XRENDER) # kcontrol/style, kicker
-macro_bool_to_01(X11_xf86misc_FOUND HAVE_XF86MISC) # kdesktop and kcontrol/lock
+macro_bool_to_01(FALSE HAVE_XF86MISC) # kdesktop and kcontrol/lock
 macro_bool_to_01(X11_dpms_FOUND HAVE_DPMS) # kdesktop
 macro_bool_to_01(X11_XSync_FOUND HAVE_XSYNC) # kwin
 
