diff -urN gtk-kde4.old/CMakeLists.txt gtk-kde4/CMakeLists.txt
--- gtk-kde4.old/CMakeLists.txt	2009-04-20 19:11:55.000000000 +0200
+++ gtk-kde4/CMakeLists.txt	2009-05-01 16:43:49.000000000 +0200
@@ -8,10 +8,9 @@
 
 ADD_SUBDIRECTORY(kcm)
 
-INSTALL(FILES lib/kcm_gk4.so DESTINATION ${KDE4_LIB_DIR}/kde4)
-INSTALL(FILES kcm/kcm_gk4.desktop DESTINATION ${KDE4_SERVICES_INSTALL_DIR} )
-INSTALL(FILES kcm/gk4icon.png DESTINATION ${KDE4_ICON_INSTALL_DIR})
-INSTALL(DIRECTORY qt4 DESTINATION /usr/share/themes)
+INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/kcm/kcm_gk4.desktop DESTINATION ${KDE4_SERVICES_INSTALL_DIR} )
+INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/kcm/gk4icon.png DESTINATION ${KDE4_ICON_INSTALL_DIR})
+INSTALL(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/qt4 DESTINATION /usr/share/themes)
 
 
 PROJECT (gtk-kde4)
@@ -23,5 +22,4 @@
 
 ADD_SUBDIRECTORY(daemon)
 
-INSTALL(FILES daemon/gtk-kde4 DESTINATION ${KDE4_BIN_INSTALL_DIR} PERMISSIONS WORLD_EXECUTE)
-INSTALL(FILES daemon/gtk-kde4.desktop DESTINATION ${AUTOSTART_INSTALL_DIR})
+INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/daemon/gtk-kde4.desktop DESTINATION ${AUTOSTART_INSTALL_DIR})
diff -urN gtk-kde4.old/daemon/CMakeLists.txt gtk-kde4/daemon/CMakeLists.txt
--- gtk-kde4.old/daemon/CMakeLists.txt	2009-04-20 16:37:41.000000000 +0200
+++ gtk-kde4/daemon/CMakeLists.txt	2009-05-01 16:40:43.000000000 +0200
@@ -1,4 +1,3 @@
-FIND_PACKAGE(KDE4 REQUIRED)
 SET(gtk-kde4-sources main.cpp widg.cpp mainwin.cpp)
 
 KDE4_ADD_EXECUTABLE(gtk-kde4 ${gtk-kde4-sources})
@@ -8,8 +7,10 @@
 
 TARGET_LINK_LIBRARIES(gtk-kde4 ${KDE4_KDECORE_LIBRARY} ${KDE4_KDEUI_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTSVG_LIBRARY})
 
-IF(EXISTS /etc/gtk-2.0/gtkrc)
-ADD_CUSTOM_COMMAND(TARGET gtk-kde4 POST_BUILD COMMAND sudo ARGS rm /etc/gtk-2.0/gtkrc)
-ADD_CUSTOM_COMMAND(TARGET gtk-kde4 POST_BUILD COMMAND sudo ARGS ln -s `kde4-config --localprefix`/share/config/gtkrc /etc/gtk-2.0)
+INSTALL(TARGETS gtk-kde4 RUNTIME DESTINATION ${KDE4_BIN_INSTALL_DIR} LIBRARY DESTINATION ${KDE4_LIB_DIR}/kde4)
 
-ENDIF()
\ Chybí znak konce řádku na konci souboru
+# really dangerous approach, rather disable
+#IF(EXISTS /etc/gtk-2.0/gtkrc)
+#	ADD_CUSTOM_COMMAND(TARGET gtk-kde4 POST_BUILD COMMAND sudo ARGS rm /etc/gtk-2.0/gtkrc)
+#	ADD_CUSTOM_COMMAND(TARGET gtk-kde4 POST_BUILD COMMAND sudo ARGS ln -s `kde4-config --localprefix`/share/config/gtkrc /etc/gtk-2.0)
+#ENDIF()
diff -urN gtk-kde4.old/daemon/gtk-kde4.desktop gtk-kde4/daemon/gtk-kde4.desktop
--- gtk-kde4.old/daemon/gtk-kde4.desktop	2009-04-20 16:09:26.000000000 +0200
+++ gtk-kde4/daemon/gtk-kde4.desktop	2009-05-01 16:23:00.000000000 +0200
@@ -2,7 +2,7 @@
 Comment[uk]=
 Comment=
 Comment[ru]=
-Exec=if test -e /home/apocalypse/.kde/share/config/gk4lock; then kstart --iconify gtk-kde4; fi
+Exec=[[ -e `kde4-config --localprefix`/share/config/gk4lock ]] && kstart --iconify gtk-kde4
 GenericName[uk]=
 GenericName=
 GenericName[ru]=
diff -urN gtk-kde4.old/kcm/CMakeLists.txt gtk-kde4/kcm/CMakeLists.txt
--- gtk-kde4.old/kcm/CMakeLists.txt	2009-01-02 20:03:49.000000000 +0100
+++ gtk-kde4/kcm/CMakeLists.txt	2009-05-01 16:40:32.000000000 +0200
@@ -1,4 +1,3 @@
-FIND_PACKAGE(KDE4 REQUIRED)
 SET(kcm_gk4-sources main.cpp form1.cpp)
 SET(kcm_gk4-uifile form.ui)
 
@@ -9,3 +8,5 @@
 LINK_DIRECTORIES(${KDE4_LIB_DIR})
 
 TARGET_LINK_LIBRARIES(kcm_gk4 ${KDE4_KDECORE_LIBRARY} ${KDE4_KDEUI_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTSVG_LIBRARY})
+
+INSTALL(TARGETS kcm_gk4 RUNTIME DESTINATION ${KDE4_BIN_INSTALL_DIR} LIBRARY DESTINATION ${KDE4_LIB_DIR}/kde4)
