--- kdesdk-4.5.4/lokalize/src/project/projectmodel.cpp.orig	2010-12-26 18:42:44.132000006 +0100
+++ kdesdk-4.5.4/lokalize/src/project/projectmodel.cpp	2010-12-26 18:42:44.132000006 +0100
@@ -1233,21 +1233,6 @@
 {
 }
 
-static void initDb(QSqlDatabase& db)
-{
-    QSqlQuery queryMain(db);
-    queryMain.exec("PRAGMA encoding = \"UTF-8\"");
-    queryMain.exec("CREATE TABLE IF NOT EXISTS metainfo ("
-                   "filepath INTEGER PRIMARY KEY ON CONFLICT REPLACE, "// AUTOINCREMENT,"
-                   //"filepath TEXT UNIQUE ON CONFLICT REPLACE, "
-                   "metainfo BLOB, "//XLIFF markup info, see catalog/catalogstring.h catalog/xliff/*
-                   "changedate INTEGER"
-                   ")");
-
-    //queryMain.exec("CREATE INDEX IF NOT EXISTS filepath_index ON metainfo ("filepath)");
-}
-
-
 static KFileMetaInfo cachedMetaInfo(const KFileItem& file)
 {
     QString dbName="metainfocache";
@@ -1257,7 +1242,16 @@
         db.setDatabaseName(KStandardDirs::locateLocal("appdata", dbName+".sqlite"));
         if (KDE_ISUNLIKELY( !db.open() ))
             return KFileMetaInfo(file.url());
-        initDb(db);
+        QSqlQuery queryMain(db);
+        queryMain.exec("PRAGMA encoding = \"UTF-8\"");
+        queryMain.exec("CREATE TABLE IF NOT EXISTS metainfo ("
+                       "filepath INTEGER PRIMARY KEY ON CONFLICT REPLACE, "// AUTOINCREMENT,"
+                       //"filepath TEXT UNIQUE ON CONFLICT REPLACE, "
+                       "metainfo BLOB, "//XLIFF markup info, see catalog/catalogstring.h catalog/xliff/*
+                       "changedate INTEGER"
+                       ")");
+
+        //queryMain.exec("CREATE INDEX IF NOT EXISTS filepath_index ON metainfo ("filepath)");
     }
     QSqlDatabase db=QSqlDatabase::database(dbName);
     if (!db.isOpen())
