Index: kdepim/korganizer/koprefsdialog.cpp
===================================================================
--- kdepim/korganizer/koprefsdialog.cpp	(revision 1133219)
+++ kdepim/korganizer/koprefsdialog.cpp	(revision 1133220)
@@ -342,7 +342,7 @@
 
       mHolidayCombo->setWhatsThis( KOPrefs::instance()->holidaysItem()->whatsThis() );
 
-      QStringList regions = HolidayRegion::regions();
+      QStringList regions = HolidayRegion::regionCodes();
       QMap<QString, QString> regionsMap;
 
       foreach ( const QString & regionCode, regions ) {
