--- khotkeys/update/CMakeLists.txt.old	2009-06-02 23:06:15.604322396 +0300
+++ khotkeys/update/CMakeLists.txt	2009-06-02 23:06:05.516480905 +0300
@@ -12,6 +12,8 @@
 
 kde4_add_executable( khotkeys_update ${khotkeys_update_SRCS})
 
+add_dependencies( khotkeys_update kded_khotkeys )
+
 target_link_libraries( khotkeys_update khotkeysprivate kworkspace ${KDE4_KDE3SUPPORT_LIBS} )
 if(X11_XTest_FOUND)
   target_link_libraries( khotkeys_update ${X11_XTest_LIB} )
--- khotkeys/kcm_hotkeys/CMakeLists.txt.old	2009-06-02 23:05:32.685403364 +0300
+++ khotkeys/kcm_hotkeys/CMakeLists.txt	2009-06-02 23:05:15.410616052 +0300
@@ -50,6 +50,8 @@
     ${kcm_hotkeys_PART_SRCS}
     )
 
+add_dependencies(kcm_hotkeys kded_khotkeys)
+
 target_link_libraries(
     kcm_hotkeys
     khotkeysprivate
